diff --git a/mobile/android/android-components/components/browser/engine-gecko/build.gradle b/mobile/android/android-components/components/browser/engine-gecko/build.gradle
index 8b57d01c4f..1c3d209c78 100644
--- a/mobile/android/android-components/components/browser/engine-gecko/build.gradle
+++ b/mobile/android/android-components/components/browser/engine-gecko/build.gradle
@@ -135,7 +135,7 @@ def getVersionNumber() {
     def parts = appVersion.split('\\.')
     def version = parts[0] + "." + parts[1] + "." + getBuildId()
 
-    if (!gradle.mozconfig.substs.MOZILLA_OFFICIAL && !gradle.mozconfig.substs.MOZ_ANDROID_FAT_AAR_ARCHITECTURES) {
+    if (!gradle.mozconfig.substs.MOZ_ANDROID_FAT_AAR_ARCHITECTURES) {
         // Use -SNAPSHOT versions locally to enable the local GeckoView substitution flow.
         version += "-SNAPSHOT"
     }
