diff --git a/mobile/android/android-components/components/feature/fxsuggest/build.gradle b/mobile/android/android-components/components/feature/fxsuggest/build.gradle
index 1fdc583cf7..933b4ec8d0 100644
--- a/mobile/android/android-components/components/feature/fxsuggest/build.gradle
+++ b/mobile/android/android-components/components/feature/fxsuggest/build.gradle
@@ -6,9 +6,7 @@ import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 
 buildscript {
     repositories {
-        maven {
-            url = "https://maven.mozilla.org/maven2"
-        }
+        mavenLocal()
     }
 
     dependencies {
