diff --git a/mobile/android/android-components/components/feature/fxsuggest/src/main/java/mozilla/components/feature/fxsuggest/FxSuggestStorage.kt b/mobile/android/android-components/components/feature/fxsuggest/src/main/java/mozilla/components/feature/fxsuggest/FxSuggestStorage.kt
index 4832f0356a..d7cea215d7 100644
--- a/mobile/android/android-components/components/feature/fxsuggest/src/main/java/mozilla/components/feature/fxsuggest/FxSuggestStorage.kt
+++ b/mobile/android/android-components/components/feature/fxsuggest/src/main/java/mozilla/components/feature/fxsuggest/FxSuggestStorage.kt
@@ -20,7 +20,7 @@ import mozilla.appservices.suggest.SuggestionQuery
 import mozilla.components.feature.fxsuggest.facts.emitSuggestionQueryCountFact
 import mozilla.components.support.base.log.logger.Logger
 import mozilla.components.support.remotesettings.RemoteSettingsService
-import mozilla.components.support.rusterrors.reportRustError
+// import mozilla.components.support.rusterrors.reportRustError
 import mozilla.appservices.suggest.InternalException as UniffiInternalException
 
 /**
@@ -126,7 +126,7 @@ class FxSuggestStorage(
             default
         } catch (e: UniffiInternalException) {
             Logger.error(e.toString())
-            reportRustError("suggest-internal-error", e.toString())
+//            reportRustError("suggest-internal-error", e.toString())
             default
         }
     }
diff --git a/mobile/android/android-components/components/lib/crash/src/main/java/mozilla/components/lib/crash/CrashReporter.kt b/mobile/android/android-components/components/lib/crash/src/main/java/mozilla/components/lib/crash/CrashReporter.kt
index 810eb70491..f11b2767f1 100644
--- a/mobile/android/android-components/components/lib/crash/src/main/java/mozilla/components/lib/crash/CrashReporter.kt
+++ b/mobile/android/android-components/components/lib/crash/src/main/java/mozilla/components/lib/crash/CrashReporter.kt
@@ -157,8 +157,8 @@ class CrashReporter internal constructor(
     fun install(
         applicationContext: Context,
         handleCaughtExceptionSideEffects: (() -> Unit)? = null,
-    ): CrashReporter {
-        instance = this
+    ) {
+/*        instance = this
 
         val defaultHandler = Thread.getDefaultUncaughtExceptionHandler()
         val handler = ExceptionHandler(
@@ -169,7 +169,7 @@ class CrashReporter internal constructor(
         )
         Thread.setDefaultUncaughtExceptionHandler(handler)
 
-        return this
+        return this*/
     }
 
     /**
diff --git a/mobile/android/android-components/components/support/appservices/src/main/java/mozilla/components/support/AppServicesInitializer.kt b/mobile/android/android-components/components/support/appservices/src/main/java/mozilla/components/support/AppServicesInitializer.kt
index 93ec365f24..4c7e01d3e9 100644
--- a/mobile/android/android-components/components/support/appservices/src/main/java/mozilla/components/support/AppServicesInitializer.kt
+++ b/mobile/android/android-components/components/support/appservices/src/main/java/mozilla/components/support/AppServicesInitializer.kt
@@ -5,9 +5,9 @@
 package mozilla.components.support
 
 import mozilla.appservices.RustComponentsInitializer
-import mozilla.components.concept.base.crash.CrashReporting
+// import mozilla.components.concept.base.crash.CrashReporting
 import mozilla.components.support.base.log.Log
-import mozilla.components.support.rusterrors.initializeRustErrors
+// import mozilla.components.support.rusterrors.initializeRustErrors
 import mozilla.components.support.rustlog.RustLog
 
 /**
@@ -27,9 +27,9 @@ object AppServicesInitializer {
         // Rust components must be initialized at the very beginning, before any other Rust call, ...
         RustComponentsInitializer.init()
 
-        if (config.crashReporting != null) {
-            initializeRustErrors(config.crashReporting)
-        }
+//        if (config.crashReporting != null) {
+//            initializeRustErrors(config.crashReporting)
+//        }
 
         // ... but RustHttpConfig.setClient() and RustLog.enable() can be called later.
         RustLog.apply {
@@ -46,7 +46,7 @@ object AppServicesInitializer {
      */
     data class Config(
         // Do not add a crash reporting default. We want to be explicit if we make this choice.
-        val crashReporting: CrashReporting?,
+//        val crashReporting: CrashReporting?,
         val logLevel: Log.Priority = Log.Priority.INFO,
     )
 }
diff --git a/mobile/android/android-components/components/support/remotesettings/src/main/java/mozilla/components/support/remotesettings/RemoteSettingsClient.kt b/mobile/android/android-components/components/support/remotesettings/src/main/java/mozilla/components/support/remotesettings/RemoteSettingsClient.kt
index f746b3f709..08208c4cc5 100644
--- a/mobile/android/android-components/components/support/remotesettings/src/main/java/mozilla/components/support/remotesettings/RemoteSettingsClient.kt
+++ b/mobile/android/android-components/components/support/remotesettings/src/main/java/mozilla/components/support/remotesettings/RemoteSettingsClient.kt
@@ -25,7 +25,7 @@ import mozilla.appservices.remotesettings.RemoteSettingsRecord
 import mozilla.appservices.remotesettings.RemoteSettingsResponse
 import mozilla.components.support.base.log.logger.Logger
 import mozilla.components.support.ktx.util.writeString
-import mozilla.components.support.rusterrors.reportRustError
+// import mozilla.components.support.rusterrors.reportRustError
 import org.json.JSONObject
 import java.io.File
 import java.io.FileNotFoundException
@@ -79,7 +79,7 @@ class RemoteSettingsClient(
             RemoteSettingsResult.NetworkFailure(e)
         } catch (e: UniffiInternalException) {
             Logger.error(e.toString())
-            reportRustError("remote-settings-internal-error", e.toString())
+//            reportRustError("remote-settings-internal-error", e.toString())
             RemoteSettingsResult.NetworkFailure(e)
         }
     }
