diff --git a/mobile/android/android-components/components/support/appservices/build.gradle b/mobile/android/android-components/components/support/appservices/build.gradle
index d9af2a9b78e8..b525b51fd39e 100644
--- a/mobile/android/android-components/components/support/appservices/build.gradle
+++ b/mobile/android/android-components/components/support/appservices/build.gradle
@@ -13,7 +13,7 @@ dependencies {
     api project(':components:concept-fetch') // Needed for rusthttp
     api project(':components:support-base') // Log.Priority is in the public api.
 
-    implementation ComponentsDependencies.mozilla_appservices_errorsupport
+//    implementation ComponentsDependencies.mozilla_appservices_errorsupport
     implementation ComponentsDependencies.mozilla_appservices_fxrelay
     implementation (ComponentsDependencies.mozilla_appservices_httpconfig) {
         // Override the version of concept-fetch that A-S depends on,
