diff --git a/mobile/android/android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt b/mobile/android/android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt
index 5233b8a8dd..b64d994704 100644
--- a/mobile/android/android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt
+++ b/mobile/android/android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt
@@ -1655,6 +1655,142 @@ class GeckoEngine(
         override var lnaBlockingEnabled: Boolean
             get() = runtime.settings.lnaBlockingEnabled
             set(value) { runtime.settings.setLnaBlockingEnabled(value) }
+
+        override var accessibilityEnabled: Boolean
+            get() = runtime.settings.accessibilityEnabled
+            set(value) { runtime.settings.accessibilityEnabled = value }
+
+        override var addressAutofillEnabled: Boolean
+            get() = runtime.settings.addressAutofillEnabled
+            set(value) { runtime.settings.addressAutofillEnabled = value }
+
+        override var cacheEnabled: Boolean
+            get() = runtime.settings.cacheEnabled
+            set(value) { runtime.settings.cacheEnabled = value }
+
+        override var cardAutofillEnabled: Boolean
+            get() = runtime.settings.cardAutofillEnabled
+            set(value) { runtime.settings.cardAutofillEnabled = value }
+
+        override var emeEnabled: Boolean
+            get() = runtime.settings.emeEnabled
+            set(value) { runtime.settings.emeEnabled = value }
+
+        override var fppOverridesIronFoxEnabled: Boolean
+            get() = runtime.settings.fppOverridesIronFoxEnabled
+            set(value) { runtime.settings.fppOverridesIronFoxEnabled = value }
+
+        override var fppOverridesIronFoxTimezoneEnabled: Boolean
+            get() = runtime.settings.fppOverridesIronFoxTimezoneEnabled
+            set(value) { runtime.settings.fppOverridesIronFoxTimezoneEnabled = value }
+
+        override var fppOverridesIronFoxWebGLEnabled: Boolean
+            get() = runtime.settings.fppOverridesIronFoxWebGLEnabled
+            set(value) { runtime.settings.fppOverridesIronFoxWebGLEnabled = value }
+
+        override var fppOverridesMozillaEnabled: Boolean
+            get() = runtime.settings.fppOverridesMozillaEnabled
+            set(value) { runtime.settings.fppOverridesMozillaEnabled = value }
+
+        override var ipv6Enabled: Boolean
+            get() = runtime.settings.ipv6Enabled
+            set(value) { runtime.settings.ipv6Enabled = value }
+
+        override var javascriptJitBaselineEnabled: Boolean
+            get() = runtime.settings.javaScriptJitBaselineEnabled
+            set(value) { runtime.settings.javaScriptJitBaselineEnabled = value }
+
+        override var javascriptJitHintsEnabled: Boolean
+            get() = runtime.settings.javaScriptJitHintsEnabled
+            set(value) { runtime.settings.javaScriptJitHintsEnabled = value }
+
+        override var javascriptJitIonEnabled: Boolean
+            get() = runtime.settings.javaScriptJitIonEnabled
+            set(value) { runtime.settings.javaScriptJitIonEnabled = value }
+
+        override var javascriptJitIonWasmEnabled: Boolean
+            get() = runtime.settings.javaScriptJitIonWasmEnabled
+            set(value) { runtime.settings.javaScriptJitIonWasmEnabled = value }
+
+        override var javascriptJitNativeRegexpEnabled: Boolean
+            get() = runtime.settings.javaScriptJitNativeRegexpEnabled
+            set(value) { runtime.settings.javaScriptJitNativeRegexpEnabled = value }
+
+        override var javascriptJitTrustedPrincipalsEnabled: Boolean
+            get() = runtime.settings.javaScriptJitTrustedPrincipalsEnabled
+            set(value) { runtime.settings.javaScriptJitTrustedPrincipalsEnabled = value }
+
+        override var passwordManagerEnabled: Boolean
+            get() = runtime.settings.passwordManagerEnabled
+            set(value) { runtime.settings.passwordManagerEnabled = value }
+
+        override var pdfjsDisabled: Boolean
+            get() = runtime.settings.pdfjsDisabled
+            set(value) { runtime.settings.pdfjsDisabled = value }
+
+        override var printEnabled: Boolean
+            get() = runtime.settings.printEnabled
+            set(value) { runtime.settings.printEnabled = value }
+
+        override var refererXOriginPolicy: Engine.RefererXOriginPolicy
+            get() = when (runtime.settings.refererXOriginPolicy) {
+                GeckoRuntimeSettings.REFERER_XORIGIN_ALWAYS -> Engine.RefererXOriginPolicy.REFERER_XORIGIN_ALWAYS
+                GeckoRuntimeSettings.REFERER_XORIGIN_IF_BASE_DOMAINS_MATCH -> Engine.RefererXOriginPolicy.REFERER_XORIGIN_IF_BASE_DOMAINS_MATCH
+                GeckoRuntimeSettings.REFERER_XORIGIN_IF_HOSTS_MATCH -> Engine.RefererXOriginPolicy.REFERER_XORIGIN_IF_HOSTS_MATCH
+                else -> throw java.lang.IllegalStateException("Unknown cross-origin referer policy returned by GeckoView")
+            }
+            set(value) {
+                runtime.settings.refererXOriginPolicy = when (value) {
+                    Engine.RefererXOriginPolicy.REFERER_XORIGIN_ALWAYS -> GeckoRuntimeSettings.REFERER_XORIGIN_ALWAYS
+                    Engine.RefererXOriginPolicy.REFERER_XORIGIN_IF_BASE_DOMAINS_MATCH -> GeckoRuntimeSettings.REFERER_XORIGIN_IF_BASE_DOMAINS_MATCH
+                    Engine.RefererXOriginPolicy.REFERER_XORIGIN_IF_HOSTS_MATCH -> GeckoRuntimeSettings.REFERER_XORIGIN_IF_HOSTS_MATCH
+                }
+            }
+
+        override var spoofEnglish: Boolean
+            get() = runtime.settings.spoofEnglish
+            set(value) {
+                value.let {
+                    runtime.settings.spoofEnglish = it
+                    localeUpdater.updateValue()
+                }
+            }
+
+        override var spoofTimezone: Boolean
+            get() = runtime.settings.spoofTimezone
+            set(value) { runtime.settings.spoofTimezone = value }
+
+        override var svgEnabled: Boolean
+            get() = runtime.settings.svgEnabled
+            set(value) { runtime.settings.svgEnabled = value }
+
+        override var translationsEnabled: Boolean
+            get() = runtime.settings.translationsEnabled
+            set(value) { runtime.settings.translationsEnabled = value }
+
+        override var translationsSupported: Boolean
+            get() = runtime.settings.translationsSupported
+            set(value) { runtime.settings.translationsSupported = value }
+
+        override var wasmEnabled: Boolean
+            get() = runtime.settings.wasmEnabled
+            set(value) { runtime.settings.wasmEnabled = value }
+
+        override var webglDisabled: Boolean
+            get() = runtime.settings.webglDisabled
+            set(value) { runtime.settings.webglDisabled = value }
+
+        override var webrtcEnabled: Boolean
+            get() = runtime.settings.webrtcEnabled
+            set(value) { runtime.settings.webrtcEnabled = value }
+
+        override var widevineEnabled: Boolean
+            get() = runtime.settings.widevineEnabled
+            set(value) { runtime.settings.widevineEnabled = value }
+
+        override var xpinstallEnabled: Boolean
+            get() = runtime.settings.xpinstallEnabled
+            set(value) { runtime.settings.xpinstallEnabled = value }
     }.apply {
         defaultSettings?.let {
             this.javascriptEnabled = it.javascriptEnabled
@@ -1702,6 +1838,36 @@ class GeckoEngine(
             this.dohAutoselectEnabled = it.dohAutoselectEnabled
             this.bannedPorts = it.bannedPorts
             this.lnaBlockingEnabled = it.lnaBlockingEnabled
+            this.accessibilityEnabled = it.accessibilityEnabled
+            this.addressAutofillEnabled = it.addressAutofillEnabled
+            this.cacheEnabled = it.cacheEnabled
+            this.cardAutofillEnabled = it.cardAutofillEnabled
+            this.emeEnabled = it.emeEnabled
+            this.fppOverridesIronFoxEnabled = it.fppOverridesIronFoxEnabled
+            this.fppOverridesIronFoxTimezoneEnabled = it.fppOverridesIronFoxTimezoneEnabled
+            this.fppOverridesIronFoxWebGLEnabled = it.fppOverridesIronFoxWebGLEnabled
+            this.fppOverridesMozillaEnabled = it.fppOverridesMozillaEnabled
+            this.ipv6Enabled = it.ipv6Enabled
+            this.javascriptJitBaselineEnabled = it.javascriptJitBaselineEnabled
+            this.javascriptJitHintsEnabled = it.javascriptJitHintsEnabled
+            this.javascriptJitIonEnabled = it.javascriptJitIonEnabled
+            this.javascriptJitIonWasmEnabled = it.javascriptJitIonWasmEnabled
+            this.javascriptJitNativeRegexpEnabled = it.javascriptJitNativeRegexpEnabled
+            this.javascriptJitTrustedPrincipalsEnabled = it.javascriptJitTrustedPrincipalsEnabled
+            this.passwordManagerEnabled = it.passwordManagerEnabled
+            this.pdfjsDisabled = it.pdfjsDisabled
+            this.printEnabled = it.printEnabled
+            this.refererXOriginPolicy = it.refererXOriginPolicy
+            this.spoofEnglish = it.spoofEnglish
+            this.spoofTimezone = it.spoofTimezone
+            this.svgEnabled = it.svgEnabled
+            this.translationsEnabled = it.translationsEnabled
+            this.translationsSupported = it.translationsSupported
+            this.wasmEnabled = it.wasmEnabled
+            this.webglDisabled = it.webglDisabled
+            this.webrtcEnabled = it.webrtcEnabled
+            this.widevineEnabled = it.widevineEnabled
+            this.xpinstallEnabled = it.xpinstallEnabled
         }
     }
 
diff --git a/mobile/android/android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/Settings.kt b/mobile/android/android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/Settings.kt
index f9aa3b43e9..5433fbda47 100644
--- a/mobile/android/android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/Settings.kt
+++ b/mobile/android/android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/Settings.kt
@@ -367,6 +367,37 @@ abstract class Settings {
      * Setting to control blocking of local network & local device (localhost) access
      */
     open var lnaBlockingEnabled: Boolean by UnsupportedSetting()
+
+    open var accessibilityEnabled: Boolean by UnsupportedSetting()
+    open var addressAutofillEnabled: Boolean by UnsupportedSetting()
+    open var cacheEnabled: Boolean by UnsupportedSetting()
+    open var cardAutofillEnabled: Boolean by UnsupportedSetting()
+    open var emeEnabled: Boolean by UnsupportedSetting()
+    open var fppOverridesIronFoxEnabled: Boolean by UnsupportedSetting()
+    open var fppOverridesIronFoxTimezoneEnabled: Boolean by UnsupportedSetting()
+    open var fppOverridesIronFoxWebGLEnabled: Boolean by UnsupportedSetting()
+    open var fppOverridesMozillaEnabled: Boolean by UnsupportedSetting()
+    open var ipv6Enabled: Boolean by UnsupportedSetting()
+    open var javascriptJitBaselineEnabled: Boolean by UnsupportedSetting()
+    open var javascriptJitHintsEnabled: Boolean by UnsupportedSetting()
+    open var javascriptJitIonEnabled: Boolean by UnsupportedSetting()
+    open var javascriptJitIonWasmEnabled: Boolean by UnsupportedSetting()
+    open var javascriptJitNativeRegexpEnabled: Boolean by UnsupportedSetting()
+    open var javascriptJitTrustedPrincipalsEnabled: Boolean by UnsupportedSetting()
+    open var passwordManagerEnabled: Boolean by UnsupportedSetting()
+    open var pdfjsDisabled: Boolean by UnsupportedSetting()
+    open var printEnabled: Boolean by UnsupportedSetting()
+    open var refererXOriginPolicy: Engine.RefererXOriginPolicy by UnsupportedSetting()
+    open var spoofEnglish: Boolean by UnsupportedSetting()
+    open var spoofTimezone: Boolean by UnsupportedSetting()
+    open var svgEnabled: Boolean by UnsupportedSetting()
+    open var translationsEnabled: Boolean by UnsupportedSetting()
+    open var translationsSupported: Boolean by UnsupportedSetting()
+    open var wasmEnabled: Boolean by UnsupportedSetting()
+    open var webglDisabled: Boolean by UnsupportedSetting()
+    open var webrtcEnabled: Boolean by UnsupportedSetting()
+    open var widevineEnabled: Boolean by UnsupportedSetting()
+    open var xpinstallEnabled: Boolean by UnsupportedSetting()
 }
 
 /**
@@ -440,6 +471,36 @@ data class DefaultSettings(
     override var dohAutoselectEnabled: Boolean = false,
     override var bannedPorts: String = "",
     override var lnaBlockingEnabled: Boolean = false,
+    override var accessibilityEnabled: Boolean = false,
+    override var addressAutofillEnabled: Boolean = false,
+    override var cacheEnabled: Boolean = false,
+    override var cardAutofillEnabled: Boolean = false,
+    override var emeEnabled: Boolean = false,
+    override var fppOverridesIronFoxEnabled: Boolean = true,
+    override var fppOverridesIronFoxTimezoneEnabled: Boolean = true,
+    override var fppOverridesIronFoxWebGLEnabled: Boolean = true,
+    override var fppOverridesMozillaEnabled: Boolean = true,
+    override var ipv6Enabled: Boolean = true,
+    override var javascriptJitBaselineEnabled: Boolean = false,
+    override var javascriptJitHintsEnabled: Boolean = false,
+    override var javascriptJitIonEnabled: Boolean = false,
+    override var javascriptJitIonWasmEnabled: Boolean = false,
+    override var javascriptJitNativeRegexpEnabled: Boolean = false,
+    override var javascriptJitTrustedPrincipalsEnabled: Boolean = false,
+    override var passwordManagerEnabled: Boolean = false,
+    override var pdfjsDisabled: Boolean = false,
+    override var printEnabled: Boolean = true,
+    override var refererXOriginPolicy: Engine.RefererXOriginPolicy = Engine.RefererXOriginPolicy.REFERER_XORIGIN_ALWAYS,
+    override var spoofEnglish: Boolean = true,
+    override var spoofTimezone: Boolean = false,
+    override var svgEnabled: Boolean = true,
+    override var translationsEnabled: Boolean = true,
+    override var translationsSupported: Boolean = true,
+    override var wasmEnabled: Boolean = true,
+    override var webglDisabled: Boolean = true,
+    override var webrtcEnabled: Boolean = true,
+    override var widevineEnabled: Boolean = false,
+    override var xpinstallEnabled: Boolean = false,
 ) : Settings() {
     override val desktopModeEnabled: Boolean
         get() = getDesktopMode()
