diff --git a/mobile/android/android-components/components/feature/fxsuggest/src/main/java/mozilla/components/feature/fxsuggest/FxSuggestStorage.kt b/mobile/android/android-components/components/feature/fxsuggest/src/main/java/mozilla/components/feature/fxsuggest/FxSuggestStorage.kt
index 4832f0356a..d7cea215d7 100644
--- a/mobile/android/android-components/components/feature/fxsuggest/src/main/java/mozilla/components/feature/fxsuggest/FxSuggestStorage.kt
+++ b/mobile/android/android-components/components/feature/fxsuggest/src/main/java/mozilla/components/feature/fxsuggest/FxSuggestStorage.kt
@@ -20,7 +20,7 @@ import mozilla.appservices.suggest.SuggestionQuery
 import mozilla.components.feature.fxsuggest.facts.emitSuggestionQueryCountFact
 import mozilla.components.support.base.log.logger.Logger
 import mozilla.components.support.remotesettings.RemoteSettingsService
-import mozilla.components.support.rusterrors.reportRustError
+// import mozilla.components.support.rusterrors.reportRustError
 import mozilla.appservices.suggest.InternalException as UniffiInternalException
 
 /**
@@ -126,7 +126,7 @@ class FxSuggestStorage(
             default
         } catch (e: UniffiInternalException) {
             Logger.error(e.toString())
-            reportRustError("suggest-internal-error", e.toString())
+//            reportRustError("suggest-internal-error", e.toString())
             default
         }
     }
diff --git a/mobile/android/android-components/components/lib/crash/src/main/java/mozilla/components/lib/crash/CrashReporter.kt b/mobile/android/android-components/components/lib/crash/src/main/java/mozilla/components/lib/crash/CrashReporter.kt
index 810eb70491..6729254da7 100644
--- a/mobile/android/android-components/components/lib/crash/src/main/java/mozilla/components/lib/crash/CrashReporter.kt
+++ b/mobile/android/android-components/components/lib/crash/src/main/java/mozilla/components/lib/crash/CrashReporter.kt
@@ -27,10 +27,10 @@ import mozilla.components.lib.crash.db.toReportEntity
 import mozilla.components.lib.crash.handler.ExceptionHandler
 import mozilla.components.lib.crash.notification.CrashNotification
 import mozilla.components.lib.crash.prompt.CrashPrompt
-import mozilla.components.lib.crash.service.CrashReporterService
-import mozilla.components.lib.crash.service.CrashTelemetryService
-import mozilla.components.lib.crash.service.SendCrashReportService
-import mozilla.components.lib.crash.service.SendCrashTelemetryService
+// import mozilla.components.lib.crash.service.CrashReporterService
+// import mozilla.components.lib.crash.service.CrashTelemetryService
+// import mozilla.components.lib.crash.service.SendCrashReportService
+// import mozilla.components.lib.crash.service.SendCrashTelemetryService
 import mozilla.components.support.base.log.logger.Logger
 
 /**
@@ -94,8 +94,8 @@ private const val START_OF_144_NIGHTLY_TIMESTAMP = 1755475200000L
  * @param useLegacyReporting Enable/Disable handling crash reporting through a notification or system dialog.
  */
 class CrashReporter internal constructor(
-    private val services: List<CrashReporterService> = emptyList(),
-    private val telemetryServices: List<CrashTelemetryService> = emptyList(),
+    private val services: List<Any> = emptyList(),
+    private val telemetryServices: List<Any> = emptyList(),
     private val shouldPrompt: Prompt = Prompt.NEVER,
     enabled: Boolean = true,
     internal val promptConfiguration: PromptConfiguration = PromptConfiguration(),
@@ -109,8 +109,8 @@ class CrashReporter internal constructor(
 
     constructor(
         context: Context,
-        services: List<CrashReporterService> = emptyList(),
-        telemetryServices: List<CrashTelemetryService> = emptyList(),
+        services: List<Any> = emptyList(),
+        telemetryServices: List<Any> = emptyList(),
         shouldPrompt: Prompt = Prompt.NEVER,
         enabled: Boolean = true,
         promptConfiguration: PromptConfiguration = PromptConfiguration(),
@@ -146,9 +146,9 @@ class CrashReporter internal constructor(
         get() = runtimeTagProviders.fold(emptyMap()) { acc, provider -> acc + provider() }
 
     init {
-        require(services.isNotEmpty() || telemetryServices.isNotEmpty()) {
-            "No crash reporter services defined"
-        }
+//        require(services.isNotEmpty() || telemetryServices.isNotEmpty()) {
+//            "No crash reporter services defined"
+//        }
     }
 
     /**
@@ -157,8 +157,8 @@ class CrashReporter internal constructor(
     fun install(
         applicationContext: Context,
         handleCaughtExceptionSideEffects: (() -> Unit)? = null,
-    ): CrashReporter {
-        instance = this
+    ) {
+/*        instance = this
 
         val defaultHandler = Thread.getDefaultUncaughtExceptionHandler()
         val handler = ExceptionHandler(
@@ -169,7 +169,7 @@ class CrashReporter internal constructor(
         )
         Thread.setDefaultUncaughtExceptionHandler(handler)
 
-        return this
+        return this */
     }
 
     /**
@@ -216,7 +216,7 @@ class CrashReporter internal constructor(
      */
     fun submitReport(crash: Crash, then: () -> Unit = {}): Job {
         return scope.launch {
-            services.forEach { service ->
+/*            services.forEach { service ->
                 val reportId = when (crash) {
                     is Crash.NativeCodeCrash -> service.report(crash)
                     is Crash.UncaughtExceptionCrash -> service.report(crash)
@@ -234,7 +234,7 @@ class CrashReporter internal constructor(
             logger.info("Crash report submitted to ${services.size} services")
             withContext(Dispatchers.Main) {
                 then()
-            }
+            } */
         }
     }
 
@@ -243,7 +243,7 @@ class CrashReporter internal constructor(
      */
     fun submitCrashTelemetry(crash: Crash, then: () -> Unit = {}): Job {
         return scope.launch {
-            telemetryServices.forEach { telemetryService ->
+/*            telemetryServices.forEach { telemetryService ->
                 when (crash) {
                     is Crash.NativeCodeCrash -> telemetryService.record(crash)
                     is Crash.UncaughtExceptionCrash -> telemetryService.record(crash)
@@ -253,7 +253,7 @@ class CrashReporter internal constructor(
             logger.info("Crash report submitted to ${telemetryServices.size} telemetry services")
             withContext(Dispatchers.Main) {
                 then()
-            }
+            } */
         }
     }
 
@@ -265,16 +265,16 @@ class CrashReporter internal constructor(
          * if stacktrace is empty, replace throwable with UnexpectedlyMissingStacktrace exception so
          * we can figure out which module is submitting caught exception reports without a stacktrace.
          */
-        var reportThrowable = throwable
+/*        var reportThrowable = throwable
         if (throwable.stackTrace.isEmpty()) {
             reportThrowable = CrashReporterException.UnexpectedlyMissingStacktrace("Missing Stacktrace", throwable)
         }
 
-        logger.info("Caught Exception report submitted to ${services.size} services")
+        logger.info("Caught Exception report submitted to ${services.size} services") */
         return scope.launch {
-            services.forEach {
+/*            services.forEach {
                 it.report(reportThrowable, crashBreadcrumbsCopy())
-            }
+            } */
         }
     }
 
@@ -288,7 +288,7 @@ class CrashReporter internal constructor(
      * ```
      */
     override fun recordCrashBreadcrumb(breadcrumb: Breadcrumb) {
-        crashBreadcrumbs.add(breadcrumb)
+//        crashBreadcrumbs.add(breadcrumb)
     }
 
     /**
@@ -383,12 +383,12 @@ class CrashReporter internal constructor(
 
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
     internal fun sendCrashReport(context: Context, crash: Crash) {
-        ContextCompat.startForegroundService(context, SendCrashReportService.createReportIntent(context, crash))
+//        ContextCompat.startForegroundService(context, SendCrashReportService.createReportIntent(context, crash))
     }
 
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
     internal fun sendCrashTelemetry(context: Context, crash: Crash) {
-        ContextCompat.startForegroundService(context, SendCrashTelemetryService.createReportIntent(context, crash))
+//        ContextCompat.startForegroundService(context, SendCrashTelemetryService.createReportIntent(context, crash))
     }
 
     @VisibleForTesting
@@ -411,8 +411,8 @@ class CrashReporter internal constructor(
         }
     }
 
-    internal fun getCrashReporterServiceById(id: String): CrashReporterService? {
-        return services.firstOrNull { it.id == id }
+    internal fun getCrashReporterServiceById(id: String) {
+//        return services.firstOrNull { it.id == id }
     }
 
     enum class Prompt {
diff --git a/mobile/android/android-components/components/lib/crash/src/main/java/mozilla/components/lib/crash/ui/DisplayableCrash.kt b/mobile/android/android-components/components/lib/crash/src/main/java/mozilla/components/lib/crash/ui/DisplayableCrash.kt
index 8c4943f230..7ef2293767 100644
--- a/mobile/android/android-components/components/lib/crash/src/main/java/mozilla/components/lib/crash/ui/DisplayableCrash.kt
+++ b/mobile/android/android-components/components/lib/crash/src/main/java/mozilla/components/lib/crash/ui/DisplayableCrash.kt
@@ -60,7 +60,7 @@ internal fun CrashWithReports.toCrash(reporter: CrashReporter) = DisplayableCras
 
 internal fun ReportEntity.toReport(reporter: CrashReporter): DisplayableCrash.Report {
     val service = reporter.getCrashReporterServiceById(serviceId)
-    val name = service?.name ?: serviceId
-    val url = service?.createCrashReportUrl(reportId)
+    val name = ""
+    val url = ""
     return DisplayableCrash.Report(name, url)
 }
diff --git a/mobile/android/android-components/components/support/appservices/src/main/java/mozilla/components/support/AppServicesInitializer.kt b/mobile/android/android-components/components/support/appservices/src/main/java/mozilla/components/support/AppServicesInitializer.kt
index 93ec365f24..4c7e01d3e9 100644
--- a/mobile/android/android-components/components/support/appservices/src/main/java/mozilla/components/support/AppServicesInitializer.kt
+++ b/mobile/android/android-components/components/support/appservices/src/main/java/mozilla/components/support/AppServicesInitializer.kt
@@ -5,9 +5,9 @@
 package mozilla.components.support
 
 import mozilla.appservices.RustComponentsInitializer
-import mozilla.components.concept.base.crash.CrashReporting
+// import mozilla.components.concept.base.crash.CrashReporting
 import mozilla.components.support.base.log.Log
-import mozilla.components.support.rusterrors.initializeRustErrors
+// import mozilla.components.support.rusterrors.initializeRustErrors
 import mozilla.components.support.rustlog.RustLog
 
 /**
@@ -27,9 +27,9 @@ object AppServicesInitializer {
         // Rust components must be initialized at the very beginning, before any other Rust call, ...
         RustComponentsInitializer.init()
 
-        if (config.crashReporting != null) {
-            initializeRustErrors(config.crashReporting)
-        }
+//        if (config.crashReporting != null) {
+//            initializeRustErrors(config.crashReporting)
+//        }
 
         // ... but RustHttpConfig.setClient() and RustLog.enable() can be called later.
         RustLog.apply {
@@ -46,7 +46,7 @@ object AppServicesInitializer {
      */
     data class Config(
         // Do not add a crash reporting default. We want to be explicit if we make this choice.
-        val crashReporting: CrashReporting?,
+//        val crashReporting: CrashReporting?,
         val logLevel: Log.Priority = Log.Priority.INFO,
     )
 }
diff --git a/mobile/android/android-components/components/support/remotesettings/src/main/java/mozilla/components/support/remotesettings/RemoteSettingsClient.kt b/mobile/android/android-components/components/support/remotesettings/src/main/java/mozilla/components/support/remotesettings/RemoteSettingsClient.kt
index f746b3f709..08208c4cc5 100644
--- a/mobile/android/android-components/components/support/remotesettings/src/main/java/mozilla/components/support/remotesettings/RemoteSettingsClient.kt
+++ b/mobile/android/android-components/components/support/remotesettings/src/main/java/mozilla/components/support/remotesettings/RemoteSettingsClient.kt
@@ -25,7 +25,7 @@ import mozilla.appservices.remotesettings.RemoteSettingsRecord
 import mozilla.appservices.remotesettings.RemoteSettingsResponse
 import mozilla.components.support.base.log.logger.Logger
 import mozilla.components.support.ktx.util.writeString
-import mozilla.components.support.rusterrors.reportRustError
+// import mozilla.components.support.rusterrors.reportRustError
 import org.json.JSONObject
 import java.io.File
 import java.io.FileNotFoundException
@@ -79,7 +79,7 @@ class RemoteSettingsClient(
             RemoteSettingsResult.NetworkFailure(e)
         } catch (e: UniffiInternalException) {
             Logger.error(e.toString())
-            reportRustError("remote-settings-internal-error", e.toString())
+//            reportRustError("remote-settings-internal-error", e.toString())
             RemoteSettingsResult.NetworkFailure(e)
         }
     }
