diff --git a/mobile/android/android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/PermissionsDialogFragment.kt b/mobile/android/android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/PermissionsDialogFragment.kt
index be8a66d479..4a8b332484 100644
--- a/mobile/android/android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/PermissionsDialogFragment.kt
+++ b/mobile/android/android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/PermissionsDialogFragment.kt
@@ -288,7 +288,7 @@ class PermissionsDialogFragment : AddonDialogFragment() {
             allowedInPrivateBrowsing.isVisible = false
         }
 
-        if (dataCollectionPermissions.contains(TECHNICAL_AND_INTERACTION_PERM) && !forOptionalPermissions) {
+        if (false) {
             optionalsSettingsTitle.isVisible = true
             technicalAndInteraction.isVisible = true
             // This is an opt-out setting.
