diff --git a/settings.gradle b/settings.gradle
index 04bd9e969..fbba69c16 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -31,14 +31,7 @@ buildscript {
         gradle.ext.mozconfig = gradle.root.mozconfig
 
         repositories {
-            gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-                maven {
-                    url = repository
-                    if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                        allowInsecureProtocol = true
-                    }
-                }
-            }
+            mavenLocal()
         }
     }
 }
diff --git a/tools/nimbus-gradle-plugin/settings.gradle b/tools/nimbus-gradle-plugin/settings.gradle
index 7d907f057..8679108f7 100644
--- a/tools/nimbus-gradle-plugin/settings.gradle
+++ b/tools/nimbus-gradle-plugin/settings.gradle
@@ -21,14 +21,7 @@ buildscript {
         gradle.ext.mozconfig = gradle.root.mozconfig
 
         repositories {
-            gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-                maven {
-                    url = repository
-                    if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                        allowInsecureProtocol = true
-                    }
-                }
-            }
+            mavenLocal()
         }
     }
 }
