From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sun, 05 Jul 2026 20:19:20 +0000
Subject: [PATCH] Application Services - Remote Settings Blocker

Prevents Remote Settings from downloading unwanted collections.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/components/remote_settings/src/client.rs b/components/remote_settings/src/client.rs
index b951d45c0..f7106e898 100644
--- a/components/remote_settings/src/client.rs
+++ b/components/remote_settings/src/client.rs
@@ -203,6 +203,13 @@ impl<C: ApiClient> RemoteSettingsClient<C> {
         jexl_filter: JexlFilter,
         api_client: C,
     ) -> Self {
+        let collection_name = if collection_name == String::from("quicksuggest-other") || collection_name == String::from("search-config-icons") ||
+         collection_name == String::from("search-config-v2") || collection_name == String::from("translations-models") ||
+         collection_name == String::from("translations-wasm") {
+            collection_name
+        } else {
+            String::from("")
+        };
         Self {
             collection_name,
             inner: Mutex::new(RemoteSettingsClientInner {
diff --git a/components/remote_settings/src/service.rs b/components/remote_settings/src/service.rs
index b19942604..f378f7897 100644
--- a/components/remote_settings/src/service.rs
+++ b/components/remote_settings/src/service.rs
@@ -64,17 +64,24 @@ impl RemoteSettingsService {
 
     pub fn make_client(&self, collection_name: String) -> Arc<RemoteSettingsClient> {
         let mut inner = self.inner.lock();
+
+        let overridden_collection_name = if collection_name == String::from("quicksuggest-other") || collection_name == String::from("search-config-icons") || collection_name == String::from("search-config-v2") || collection_name == String::from("translations-models") || collection_name == String::from("translations-wasm") {
+            collection_name
+        } else {
+            String::from("")
+        };
+
         // Allow using in-memory databases for testing of external crates.
         let storage = if inner.storage_dir == ":memory:" {
             Storage::new(inner.storage_dir.clone())
         } else {
-            Storage::new(inner.storage_dir.join(format!("{collection_name}.sql")))
+            Storage::new(inner.storage_dir.join(format!("{overridden_collection_name}.sql")))
         };
 
         let client = Arc::new(RemoteSettingsClient::new(
             inner.base_url.clone(),
             inner.bucket_name.clone(),
-            collection_name.clone(),
+            overridden_collection_name.clone(),
             inner.app_context.clone(),
             storage,
         ));
@@ -108,7 +115,7 @@ impl RemoteSettingsService {
                     }
                 }
             }
-            if synced_collections.insert(collection_name.to_string()) {
+            if (collection_name == String::from("quicksuggest-other") || collection_name == String::from("translations-models") || collection_name == String::from("translations-wasm")) && synced_collections.insert(collection_name.to_string()) {
                 trace!("syncing collection: {collection_name}");
                 client.sync()?;
             }
-- 
