diff --git a/components/remote_settings/src/client.rs b/components/remote_settings/src/client.rs
index 84a720d3e..1c4a2b433 100644
--- a/components/remote_settings/src/client.rs
+++ b/components/remote_settings/src/client.rs
@@ -108,10 +108,10 @@ impl<C: ApiClient> RemoteSettingsClient<C> {
     // and check for the folder + name in
     // `remote_settings/dumps/{bucket}/attachments/{collection}/{filename}
     packaged_attachments! {
-        ("main", "regions") => [
+/*        ("main", "regions") => [
             "world",
             "world-buffered",
-        ],
+        ], */
         ("main", "search-config-icons") => [
             "001500a9-1a6c-3f5a-ba15-a5f5a075d256",
             "06cf7432-efd7-f244-927b-5e423005e1ea",
