diff --git a/components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/Nimbus.kt b/components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/Nimbus.kt
index f77d82a..dbc4ce5 100644
--- a/components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/Nimbus.kt
+++ b/components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/Nimbus.kt
@@ -160,11 +160,11 @@ open class Nimbus(
         }
 
     override var globalUserParticipation: Boolean
-        get() = nimbusClient.getGlobalUserParticipation()
+        get() = false
         set(active) {
             dbScope.launch {
-                setExperimentParticipationOnThisThread(active)
-                setRolloutParticipationOnThisThread(active)
+//                setExperimentParticipationOnThisThread(active)
+//                setRolloutParticipationOnThisThread(active)
             }
         }
 
@@ -295,7 +295,7 @@ open class Nimbus(
 
     override fun isFetchEnabled() = withCatchAll("isFetchEnabled") {
         nimbusClient.isFetchEnabled()
-    } ?: true
+    } ?: false
 
     @WorkerThread
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
