diff --git a/components/remote_settings/src/client.rs b/components/remote_settings/src/client.rs
index f3e54d3..95154eb 100644
--- a/components/remote_settings/src/client.rs
+++ b/components/remote_settings/src/client.rs
@@ -106,10 +106,10 @@ impl<C: ApiClient> RemoteSettingsClient<C> {
     // and check for the folder + name in
     // `remote_settings/dumps/{bucket}/attachments/{collection}/{filename}
     packaged_attachments! {
-        ("main", "regions") => [
-            "world",
-            "world-buffered",
-        ],
+//        ("main", "regions") => [
+//            "world",
+//            "world-buffered",
+//        ],
         ("main", "search-config-icons") => [
             "001500a9-1a6c-3f5a-ba15-a5f5a075d256",
             "06cf7432-efd7-f244-927b-5e423005e1ea",
