diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 2842a8204c..5a81d93037 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -654,20 +654,17 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         true,
     )
 
-    var shouldUseCookieBannerPrivateMode by lazyFeatureFlagPreference(
+    var shouldUseCookieBannerPrivateMode by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_cookie_banner_private_mode),
-        featureFlag = true,
-        default = { shouldUseCookieBannerPrivateModeDefaultValue },
+        default = true,
     )
 
-    val shouldUseCookieBannerPrivateModeDefaultValue: Boolean
-        get() = cookieBannersSection[CookieBannersSection.FEATURE_SETTING_VALUE_PBM] == 1
+    val shouldUseCookieBannerPrivateModeDefaultValue: Boolean = true
 
     val shouldUseCookieBanner: Boolean
         get() = cookieBannersSection[CookieBannersSection.FEATURE_SETTING_VALUE] == 1
 
-    val shouldShowCookieBannerUI: Boolean
-        get() = cookieBannersSection[CookieBannersSection.FEATURE_UI] == 1
+    val shouldShowCookieBannerUI: Boolean = true
 
     val shouldEnableCookieBannerDetectOnly: Boolean
         get() = cookieBannersSection[CookieBannersSection.FEATURE_SETTING_DETECT_ONLY] == 1
diff --git a/mobile/android/fenix/app/src/main/res/xml/preferences.xml b/mobile/android/fenix/app/src/main/res/xml/preferences.xml
index c03abd3..2abf25c 100644
--- a/mobile/android/fenix/app/src/main/res/xml/preferences.xml
+++ b/mobile/android/fenix/app/src/main/res/xml/preferences.xml
@@ -111,7 +111,7 @@
         <org.mozilla.fenix.settings.cookiebannerhandling.CustomCBHSwitchPreference
             android:key="@string/pref_key_cookie_banner_private_mode"
             app:iconSpaceReserved="false"
-            app:isPreferenceVisible="false"
+            app:isPreferenceVisible="true"
             android:title="@string/preferences_cookie_banner_reduction_private_mode" />
 
         <androidx.preference.Preference
