diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingScreen.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingScreen.kt
index 732a80cb03..7a2fdae0c3 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingScreen.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingScreen.kt
@@ -135,6 +135,8 @@ fun OnboardingScreen(
         if (!settings.shouldShowMarketingOnboarding && isNotPartnershipDistribution) {
             val marketingPage = pagesToDisplay.find { it.type == OnboardingPageUiData.Type.MARKETING_DATA }
             marketingPage?.let { pagesToDisplay.remove(it) }
+            val syncPage = pagesToDisplay.find { it.type == OnboardingPageUiData.Type.SYNC_SIGN_IN }
+            syncPage?.let { pagesToDisplay.remove(it) }
         }
 
         onDispose {
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/about/AboutFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/about/AboutFragment.kt
index e43dec84c6..e462b4c0fe 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/about/AboutFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/about/AboutFragment.kt
@@ -208,11 +208,11 @@ class AboutFragment(
             AboutPageItem(
                 AboutItem.ExternalLink(
                     SUPPORT,
-                    SupportUtils.getSumoURLForTopic(context, SupportUtils.SumoTopic.HELP),
+                    "https://gitlab.com/ironfox-oss/IronFox/-/issues",
                 ),
                 getString(R.string.about_support),
             ),
-            AboutPageItem(
+/*            AboutPageItem(
                 AboutItem.Crashes,
                 getString(R.string.about_crashes),
             ),
@@ -229,7 +229,7 @@ class AboutFragment(
                     SupportUtils.getSumoURLForTopic(context, SupportUtils.SumoTopic.YOUR_RIGHTS),
                 ),
                 getString(R.string.about_know_your_rights),
-            ),
+            ),*/
             AboutPageItem(
                 AboutItem.ExternalLink(LICENSING_INFO, ABOUT_LICENSE_URL),
                 getString(R.string.about_licensing_information),
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt
index c13ecb6c86..cbcdfcb59e 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt
@@ -26,9 +26,9 @@ object SupportUtils {
     const val WIKIPEDIA_URL = "https://www.wikipedia.org/"
     const val FENIX_PLAY_STORE_URL = "https://play.google.com/store/apps/details?id=${BuildConfig.APPLICATION_ID}"
     const val GOOGLE_URL = "https://www.google.com/"
-    const val GOOGLE_US_URL = "https://www.google.com/webhp?client=firefox-b-1-m&channel=ts"
-    const val GOOGLE_XX_URL = "https://www.google.com/webhp?client=firefox-b-m&channel=ts"
-    const val WHATS_NEW_URL = "https://www.mozilla.org/firefox/android/notes"
+    const val GOOGLE_US_URL = ""
+    const val GOOGLE_XX_URL = ""
+    const val WHATS_NEW_URL = "https://gitlab.com/ironfox-oss/IronFox/-/releases"
     const val FXACCOUNT_SUMO_URL = "https://support.mozilla.org/kb/access-mozilla-services-firefox-account"
 
     // This is locale-less on purpose so that the content negotiation happens on the AMO side because the current
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt
index e5ab956ac3..8f88bf456e 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt
@@ -82,7 +82,7 @@ class OnboardingFragment : Fragment() {
             pagesToDisplay(
                 showDefaultBrowserPage = isNotDefaultBrowser(this) && !isDefaultBrowserPromptSupported(),
                 showNotificationPage = canShowNotificationPage(),
-                showAddWidgetPage = canShowAddSearchWidgetPrompt(AppWidgetManager.getInstance(activity)),
+                showAddWidgetPage = false,
             ).toMutableList()
         }
     }
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingMapper.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingMapper.kt
index 795fcd7b7e..1b4d2eef78 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingMapper.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingMapper.kt
@@ -32,7 +32,7 @@ internal fun Collection<OnboardingCardData>.toPageUiData(
         .sortedBy { it.ordering }
         .mapIndexed { index, onboardingCardData ->
             // only first onboarding card shows privacy caption
-            onboardingCardData.toPageUiData(if (index == 0) privacyCaption else null)
+            onboardingCardData.toPageUiData(null)
         }
 }
 
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/TermsOfServiceOnboardingPage.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/TermsOfServiceOnboardingPage.kt
index 10488889d7..f916d94cc9 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/TermsOfServiceOnboardingPage.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/TermsOfServiceOnboardingPage.kt
@@ -135,22 +135,22 @@ private fun BodyText(
             modifier = Modifier.padding(horizontal = 10.dp),
         ) {
             val lineOneState = LinkTextState(
-                text = it.lineOneLinkText,
+                text = "",
                 url = it.lineOneLinkUrl,
-                onClick = eventHandler::onTermsOfServiceLinkClicked,
+                onClick = {},
             )
             val lineTwoState = LinkTextState(
-                text = it.lineTwoLinkText,
+                text = "",
                 url = it.lineTwoLinkUrl,
-                onClick = eventHandler::onPrivacyNoticeLinkClicked,
+                onClick = {},
             )
             val lineThreeState = LinkTextState(
-                text = it.lineThreeLinkText,
+                text = "",
                 url = "",
-                onClick = { _ -> eventHandler.onManagePrivacyPreferencesLinkClicked() },
+                onClick = {},
             )
             LinkText(
-                text = it.lineOneText.updateFirstPlaceholder(it.lineOneLinkText),
+                text = "",
                 linkTextStates = listOf(
                     lineOneState,
                 ),
@@ -162,7 +162,7 @@ private fun BodyText(
             )
 
             LinkText(
-                text = it.lineTwoText.updateFirstPlaceholder(it.lineTwoLinkText),
+                text = "",
                 linkTextStates = listOf(
                     lineTwoState,
                 ),
@@ -174,7 +174,7 @@ private fun BodyText(
             )
 
             LinkText(
-                text = it.lineThreeText.updateFirstPlaceholder(it.lineThreeLinkText),
+                text = "",
                 linkTextStates = listOf(
                     lineThreeState,
                 ),
diff --git a/mobile/android/moz.configure b/mobile/android/moz.configure
index 0e2a6efdc9..3d7ff4a88f 100644
--- a/mobile/android/moz.configure
+++ b/mobile/android/moz.configure
@@ -116,7 +116,7 @@ imply_option("MOZ_SERVICES_HEALTHREPORT", True)
 imply_option("MOZ_GECKOVIEW_HISTORY", True)
 
 imply_option("MOZ_APP_UA_NAME", "Firefox")
-imply_option("MOZ_APP_VENDOR", "Mozilla")
+imply_option("MOZ_APP_VENDOR", "IronFox OSS")
 imply_option("MOZ_APP_ID", "{aa3c5121-dab2-40e2-81ca-7ea25febc110}")
 imply_option("BROWSER_CHROME_URL", "chrome://geckoview/content/geckoview.xhtml")
 
@@ -153,3 +153,5 @@ set_config(
     "MOZ_ANDROID_FAT_AAR_ARCHITECTURES",
     depends("MOZ_ANDROID_FAT_AAR_ARCHITECTURES")(lambda x: x),
 )
+
+include("ironfox.configure")
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 9197b274db..652a7a4486 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -438,7 +438,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
 
     var shouldShowPrivacyPopWindow by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_privacy_pop_window),
-        default = true,
+        default = false,
     )
 
     var shouldUseLightTheme by booleanPreference(
