diff --git a/mobile/android/fenix/app/build.gradle b/mobile/android/fenix/app/build.gradle
index bdbc2336e3..abb8d71f2a 100644
--- a/mobile/android/fenix/app/build.gradle
+++ b/mobile/android/fenix/app/build.gradle
@@ -68,8 +68,8 @@ android {
         buildConfigField "String", "VCS_HASH", "\"\"" // see override in release builds for why it's blank.
         // This should be the "public" base URL of AMO.
         buildConfigField "String", "AMO_BASE_URL", "\"https://addons.mozilla.org\""
-        buildConfigField "String", "AMO_COLLECTION_NAME", "\"Extensions-for-Android\""
-        buildConfigField "String", "AMO_COLLECTION_USER", "\"mozilla\""
+        buildConfigField "String", "AMO_COLLECTION_NAME", "\"IronFox\""
+        buildConfigField "String", "AMO_COLLECTION_USER", "\"18588170\""
         // This should be the base URL used to call the AMO API.
         buildConfigField "String", "AMO_SERVER_URL", "\"https://services.addons.mozilla.org\""
 
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt
index c13ecb6c86..75b9f05e51 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt
@@ -21,14 +21,14 @@ import java.net.URLEncoder
 import java.util.Locale
 
 object SupportUtils {
-    const val RATE_APP_URL = "market://details?id=" + BuildConfig.APPLICATION_ID
-    const val POCKET_TRENDING_URL = "https://getpocket.com/fenix-top-articles"
-    const val WIKIPEDIA_URL = "https://www.wikipedia.org/"
-    const val FENIX_PLAY_STORE_URL = "https://play.google.com/store/apps/details?id=${BuildConfig.APPLICATION_ID}"
-    const val GOOGLE_URL = "https://www.google.com/"
-    const val GOOGLE_US_URL = "https://www.google.com/webhp?client=firefox-b-1-m&channel=ts"
-    const val GOOGLE_XX_URL = "https://www.google.com/webhp?client=firefox-b-m&channel=ts"
-    const val WHATS_NEW_URL = "https://www.mozilla.org/firefox/android/notes"
+    const val RATE_APP_URL = ""
+    const val POCKET_TRENDING_URL = ""
+    const val WIKIPEDIA_URL = ""
+    const val FENIX_PLAY_STORE_URL = ""
+    const val GOOGLE_URL = ""
+    const val GOOGLE_US_URL = ""
+    const val GOOGLE_XX_URL = ""
+    const val WHATS_NEW_URL = "https://gitlab.com/ironfox-oss/IronFox/-/releases"
     const val FXACCOUNT_SUMO_URL = "https://support.mozilla.org/kb/access-mozilla-services-firefox-account"
 
     // This is locale-less on purpose so that the content negotiation happens on the AMO side because the current
@@ -38,34 +38,34 @@ object SupportUtils {
 
     enum class SumoTopic(internal val topicStr: String) {
         HELP("faq-android"),
-        PRIVATE_BROWSING_MYTHS("common-myths-about-private-browsing"),
-        YOUR_RIGHTS("your-rights"),
-        TRACKING_PROTECTION("tracking-protection-firefox-android"),
-        TOTAL_COOKIE_PROTECTION("enhanced-tracking-protection-android"),
-        OPT_OUT_STUDIES("how-opt-out-studies-firefox-android"),
-        SEND_TABS("send-tab-preview"),
-        SET_AS_DEFAULT_BROWSER("make-firefox-default-browser-android"),
-        SEARCH_SUGGESTION("how-search-firefox-preview"),
-        CUSTOM_SEARCH_ENGINES("custom-search-engines"),
-        SYNC_SETUP("how-set-firefox-sync-firefox-android"),
-        QR_CAMERA_ACCESS("qr-camera-access"),
-        SMARTBLOCK("smartblock-enhanced-tracking-protection"),
-        SPONSOR_PRIVACY("sponsor-privacy"),
-        HTTPS_ONLY_MODE("https-only-mode-firefox-android"),
-        DNS_OVER_HTTPS("https-only-mode-firefox-android"), // FIXME
-        DNS_OVER_HTTPS_LOCAL_PROVIDER("https-only-mode-firefox-android"), // FIXME
-        DNS_OVER_HTTPS_NETWORK("https-only-mode-firefox-android"), // FIXME
-        UNSIGNED_ADDONS("unsigned-addons"),
-        REVIEW_QUALITY_CHECK("review_checker_mobile"),
-        FX_SUGGEST("search-suggestions-firefox"),
-        TRANSLATIONS("android-translation"),
-        MANAGE_OPTIONAL_EXTENSION_PERMISSIONS("manage-optional-permissions-android-extensions"),
-        EXTENSION_PERMISSIONS("extension-permissions"),
-        FIND_INSTALL_ADDONS("add-ons-firefox-android"),
-        CRASH_REPORTS("mobile-crash-reports"),
-        TECHNICAL_AND_INTERACTION_DATA("mobile-technical-and-interaction-data"),
-        USAGE_PING_SETTINGS("usage-ping-settings-mobile"),
-        MARKETING_DATA("mobile-marketing-data"),
+        PRIVATE_BROWSING_MYTHS("kb/common-myths-about-private-browsing"),
+        YOUR_RIGHTS(""),
+        TRACKING_PROTECTION("kb/tracking-protection-firefox-android"),
+        TOTAL_COOKIE_PROTECTION("kb/enhanced-tracking-protection-android"),
+        OPT_OUT_STUDIES(""),
+        SEND_TABS("kb/send-tab-preview"),
+        SET_AS_DEFAULT_BROWSER("kb/make-firefox-default-browser-android"),
+        SEARCH_SUGGESTION("kb/how-search-firefox-preview"),
+        CUSTOM_SEARCH_ENGINES("kb/custom-search-engines"),
+        SYNC_SETUP("kb/how-set-firefox-sync-firefox-android"),
+        QR_CAMERA_ACCESS("kb/qr-camera-access"),
+        SMARTBLOCK("kb/smartblock-enhanced-tracking-protection"),
+        SPONSOR_PRIVACY(""),
+        HTTPS_ONLY_MODE("kb/https-only-mode-firefox-android"),
+        DNS_OVER_HTTPS("kb/https-only-mode-firefox-android"), // FIXME
+        DNS_OVER_HTTPS_LOCAL_PROVIDER("kb/https-only-mode-firefox-android"), // FIXME
+        DNS_OVER_HTTPS_NETWORK("kb/https-only-mode-firefox-android"), // FIXME
+        UNSIGNED_ADDONS("kb/unsigned-addons"),
+        REVIEW_QUALITY_CHECK(""),
+        FX_SUGGEST(""),
+        TRANSLATIONS("kb/android-translation"),
+        MANAGE_OPTIONAL_EXTENSION_PERMISSIONS("kb/manage-optional-permissions-android-extensions"),
+        EXTENSION_PERMISSIONS("kb/extension-permissions"),
+        FIND_INSTALL_ADDONS("kb/add-ons-firefox-android"),
+        CRASH_REPORTS(""),
+        TECHNICAL_AND_INTERACTION_DATA(""),
+        USAGE_PING_SETTINGS(""),
+        MARKETING_DATA(""),
     }
 
     enum class MozillaPage(internal val path: String) {
@@ -87,7 +87,7 @@ object SupportUtils {
         val appVersion = context.appVersionName.replace(" ", "")
         val osTarget = "Android"
         val langTag = getLanguageTag(locale)
-        return "https://support.mozilla.org/1/mobile/$appVersion/$osTarget/$langTag/$escapedTopic"
+        return "https://support.mozilla.org/$langTag/$escapedTopic"
     }
 
     /**
@@ -97,13 +97,13 @@ object SupportUtils {
     fun getGenericSumoURLForTopic(topic: SumoTopic, locale: Locale = Locale.getDefault()): String {
         val escapedTopic = getEncodedTopicUTF8(topic.topicStr)
         val langTag = getLanguageTag(locale)
-        return "https://support.mozilla.org/$langTag/kb/$escapedTopic"
+        return "https://support.mozilla.org/kb/$escapedTopic"
     }
 
     fun getMozillaPageUrl(page: MozillaPage, locale: Locale = Locale.getDefault()): String {
         val path = page.path
         val langTag = getLanguageTag(locale)
-        return "https://www.mozilla.org/$langTag/$path"
+        return "https://www.mozilla.org/$path"
     }
 
     fun createCustomTabIntent(context: Context, url: String): Intent = CustomTabsIntent.Builder()
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt
index 827dd48501..02d13b7b83 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt
@@ -83,7 +83,7 @@ class OnboardingFragment : Fragment() {
             pagesToDisplay(
                 showDefaultBrowserPage = isNotDefaultBrowser(this) && !isDefaultBrowserPromptSupported(),
                 showNotificationPage = canShowNotificationPage(),
-                showAddWidgetPage = canShowAddSearchWidgetPrompt(AppWidgetManager.getInstance(activity)),
+                showAddWidgetPage = false,
             ).toMutableList()
         }
     }
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingMapper.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingMapper.kt
index 1c717bf900..f33154cc20 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingMapper.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingMapper.kt
@@ -34,7 +34,7 @@ internal fun Collection<OnboardingCardData>.toPageUiData(
         .mapIndexed {
                 index, onboardingCardData ->
             // only first onboarding card shows privacy caption
-            onboardingCardData.toPageUiData(if (index == 0) privacyCaption else null)
+            onboardingCardData.toPageUiData(null)
         }
 }
 
@@ -114,7 +114,6 @@ private fun OnboardingCardData.toPageUiData(privacyCaption: Caption?) = Onboardi
     themeOptions = extraData?.customizationThemeData
         ?.takeIf { it.isNotEmpty() }
         ?.toOnboardingThemeOptions(),
-    termsOfService = extraData?.termOfServiceData?.toOnboardingTermsOfService(),
     marketingData = extraData?.marketingData?.toOnboardingMarketingData(),
 )
 
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/TermsOfServiceOnboardingPage.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/TermsOfServiceOnboardingPage.kt
index d093b9a370..5c454a258c 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/TermsOfServiceOnboardingPage.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/TermsOfServiceOnboardingPage.kt
@@ -134,22 +134,22 @@ private fun BodyText(
             modifier = Modifier.padding(horizontal = 10.dp),
         ) {
             val lineOneState = LinkTextState(
-                text = it.lineOneLinkText,
-                url = it.lineOneLinkUrl,
-                onClick = eventHandler::onTermsOfServiceLinkClicked,
+                text = "",
+                url = "",
+                onClick = {},
             )
             val lineTwoState = LinkTextState(
-                text = it.lineTwoLinkText,
-                url = it.lineTwoLinkUrl,
-                onClick = eventHandler::onPrivacyNoticeLinkClicked,
+                text = "",
+                url = "",
+                onClick = {},
             )
             val lineThreeState = LinkTextState(
-                text = it.lineThreeLinkText,
+                text = "",
                 url = "",
-                onClick = { _ -> eventHandler.onManagePrivacyPreferencesLinkClicked() },
+                onClick = {},
             )
             LinkText(
-                text = it.lineOneText.updateFirstPlaceholder(it.lineOneLinkText),
+                text = "",
                 linkTextStates = listOf(
                     lineOneState,
                 ),
@@ -161,7 +161,7 @@ private fun BodyText(
             )
 
             LinkText(
-                text = it.lineTwoText.updateFirstPlaceholder(it.lineTwoLinkText),
+                text = "",
                 linkTextStates = listOf(
                     lineTwoState,
                 ),
@@ -173,7 +173,7 @@ private fun BodyText(
             )
 
             LinkText(
-                text = it.lineThreeText.updateFirstPlaceholder(it.lineThreeLinkText),
+                text = "",
                 linkTextStates = listOf(
                     lineThreeState,
                 ),
@@ -200,14 +200,14 @@ private fun OnboardingPagePreview() {
                 title = stringResource(id = R.string.onboarding_welcome_to_firefox),
                 description = stringResource(id = R.string.onboarding_terms_of_service_sub_header_2),
                 termsOfService = OnboardingTermsOfService(
-                    lineOneText = stringResource(id = R.string.onboarding_term_of_service_line_one_2),
-                    lineOneLinkText = stringResource(id = R.string.onboarding_term_of_service_line_one_link_text_2),
-                    lineOneLinkUrl = "URL",
-                    lineTwoText = stringResource(id = R.string.onboarding_term_of_service_line_two_2),
-                    lineTwoLinkText = stringResource(id = R.string.onboarding_term_of_service_line_two_link_text),
-                    lineTwoLinkUrl = "URL",
-                    lineThreeText = stringResource(id = R.string.onboarding_term_of_service_line_three),
-                    lineThreeLinkText = stringResource(id = R.string.onboarding_term_of_service_line_three_link_text),
+                    lineOneText = "",
+                    lineOneLinkText = "",
+                    lineOneLinkUrl = "",
+                    lineTwoText = "",
+                    lineTwoLinkText = "",
+                    lineTwoLinkUrl = "",
+                    lineThreeText = "",
+                    lineThreeLinkText = "",
                 ),
                 imageRes = R.drawable.ic_firefox,
                 primaryButton = Action(
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 60daeacc7e..698fadc834 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -2485,26 +2485,17 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     /**
      * Indicates if the user has completed the setup step for choosing the toolbar location
      */
-    var hasCompletedSetupStepToolbar by booleanPreference(
-        key = appContext.getPreferenceKey(R.string.pref_key_setup_step_toolbar),
-        default = false,
-    )
+    var hasCompletedSetupStepToolbar = true
 
     /**
      * Indicates if the user has completed the setup step for choosing the theme
      */
-    var hasCompletedSetupStepTheme by booleanPreference(
-        key = appContext.getPreferenceKey(R.string.pref_key_setup_step_theme),
-        default = false,
-    )
+    var hasCompletedSetupStepTheme = true
 
     /**
      * Indicates if the user has completed the setup step for exploring extensions
      */
-    var hasCompletedSetupStepExtensions by booleanPreference(
-        key = appContext.getPreferenceKey(R.string.pref_key_setup_step_extensions),
-        default = false,
-    )
+    var hasCompletedSetupStepExtensions = true
 
     /**
      * Indicates if this is the default browser.
@@ -2517,12 +2508,5 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     /**
      * Indicates whether or not to show the checklist feature.
      */
-    var showSetupChecklist by lazyFeatureFlagPreference(
-        key = appContext.getPreferenceKey(R.string.pref_key_setup_checklist_complete),
-        default = {
-            FxNimbus.features.setupChecklist.value().enabled &&
-                canShowAddSearchWidgetPrompt(AppWidgetManager.getInstance(appContext))
-        },
-        featureFlag = true,
-    )
+    var showSetupChecklist = false
 }
diff --git a/mobile/android/fenix/app/src/main/res/drawable/ic_firefox.xml b/mobile/android/fenix/app/src/main/res/drawable/ic_firefox.xml
index 8e369db27c..b1e6090f14 100644
--- a/mobile/android/fenix/app/src/main/res/drawable/ic_firefox.xml
+++ b/mobile/android/fenix/app/src/main/res/drawable/ic_firefox.xml
@@ -1,141 +1,32 @@
-<!-- This Source Code Form is subject to the terms of the Mozilla Public
-   - License, v. 2.0. If a copy of the MPL was not distributed with this
-   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
-
-<vector android:height="112.7dp" android:viewportHeight="54.28"
-    android:viewportWidth="52.03" android:width="108.02839dp"
-    xmlns:aapt="http://schemas.android.com/aapt" xmlns:android="http://schemas.android.com/apk/res/android">
-    <path android:pathData="M50.15,18.05c-1.13,-2.72 -3.43,-5.66 -5.23,-6.59a26.79,26.79 0,0 1,2.64 7.91c-3,-7.34 -7.94,-10.3 -12,-16.74 -0.2,-0.33 -0.41,-0.65 -0.61,-1a5.94,5.94 0,0 1,-0.29 -0.54,4.92 4.92,0 0,1 -0.39,-1 0.07,0.07 0,0 0,-0.05 -0.07h-0.07a19,19 0,0 0,-9 14.46A12.92,12.92 0,0 0,18.02 17.26a10.031,10.031 0,0 0,-0.68 -0.51,12 12,0 0,1 -0.07,-6.36A19.21,19.21 0,0 0,11.02 15.23c-1,-1.31 -1,-5.62 -0.9,-6.52a4.39,4.39 0,0 0,-0.87 0.46,19.37 19.37,0 0,0 -2.54,2.18 22.23,22.23 0,0 0,-2.43 2.92,21.88 21.88,0 0,0 -3.5,7.88v0.18a39.159,39.159 0,0 0,-0.26 1.62,0.13 0.13,0 0,1 0,0.06 26.47,26.47 0,0 0,-0.5 3.59L0.02,27.77a26,26 0,0 0,51.7 4.41l0.12,-1a26.9,26.9 0,0 0,-1.69 -13.13zM20.15,38.43l0.36,0.18zM26.15,22.73zM47.62,19.42z">
-        <aapt:attr name="android:fillColor">
-            <gradient android:endX="4.825955" android:endY="50.964695"
-                android:startX="45.6871" android:startY="8.501661" android:type="linear">
-                <item android:color="#FFFFF44F" android:offset="0.05"/>
-                <item android:color="#FFFF980E" android:offset="0.37"/>
-                <item android:color="#FFFF3647" android:offset="0.53"/>
-                <item android:color="#FFE31587" android:offset="0.7"/>
-            </gradient>
-        </aapt:attr>
-    </path>
-    <path android:pathData="M50.15,18.05c-1.13,-2.72 -3.43,-5.66 -5.23,-6.59a26.79,26.79 0,0 1,2.64 7.91,23.55 23.55,0 0,1 -0.81,17.57c-3,6.4 -10.21,13 -21.51,12.65 -12.22,-0.35 -23,-9.41 -25,-21.29a11.22,11.22 0,0 1,0.18 -4.34,19.61 19.61,0 0,0 -0.4,3.64L0.02,27.77a26,26 0,0 0,51.7 4.41l0.12,-1a26.9,26.9 0,0 0,-1.69 -13.13z">
-        <aapt:attr name="android:fillColor">
-            <gradient android:centerX="45.955353"
-                android:centerY="11.459999"
-                android:gradientRadius="53.746708" android:type="radial">
-                <item android:color="#FFFFBD4F" android:offset="0.13"/>
-                <item android:color="#FFFF980E" android:offset="0.28"/>
-                <item android:color="#FFFF3750" android:offset="0.47"/>
-                <item android:color="#FFEB0878" android:offset="0.78"/>
-                <item android:color="#FFE50080" android:offset="0.86"/>
-            </gradient>
-        </aapt:attr>
-    </path>
-    <path android:pathData="M50.15,18.05c-1.13,-2.72 -3.43,-5.66 -5.23,-6.59a26.79,26.79 0,0 1,2.64 7.91,23.55 23.55,0 0,1 -0.81,17.57c-3,6.4 -10.21,13 -21.51,12.65 -12.22,-0.35 -23,-9.41 -25,-21.29a11.22,11.22 0,0 1,0.18 -4.34,19.61 19.61,0 0,0 -0.4,3.64L0.02,27.77a26,26 0,0 0,51.7 4.41l0.12,-1a26.9,26.9 0,0 0,-1.69 -13.13z">
-        <aapt:attr name="android:fillColor">
-            <gradient android:centerX="25.630222"
-                android:centerY="28.77216"
-                android:gradientRadius="55.090374" android:type="radial">
-                <item android:color="#FF960E18" android:offset="0.3"/>
-                <item android:color="#BCB11927" android:offset="0.35"/>
-                <item android:color="#56DB293D" android:offset="0.43"/>
-                <item android:color="#16F5334B" android:offset="0.5"/>
-                <item android:color="#00FF3750" android:offset="0.53"/>
-            </gradient>
-        </aapt:attr>
-    </path>
-    <path android:pathData="M37.49,21.11l0.17,0.12a14.35,14.35 0,0 0,-2.42 -3.15c-8.09,-8.09 -2.12,-17.55 -1.12,-18a19,19 0,0 0,-9 14.46h0.92a13.13,13.13 0,0 1,11.45 6.57z">
-        <aapt:attr name="android:fillColor">
-            <gradient android:centerX="31.1392"
-                android:centerY="-2.457998"
-                android:gradientRadius="29.609999" android:type="radial">
-                <item android:color="#FFFFF44F" android:offset="0.13"/>
-                <item android:color="#FFFF980E" android:offset="0.53"/>
-            </gradient>
-        </aapt:attr>
-    </path>
-    <path android:pathData="M26.1,22.77c0,0.65 -2.33,2.88 -3.13,2.88 -7.4,0 -8.6,4.48 -8.6,4.48a11,11 0,0 0,6.13 8.52l0.44,0.21a12.13,12.13 0,0 0,0.76 0.31,11.91 11.91,0 0,0 3.39,0.66c13,0.61 15.49,-15.52 6.13,-20.2a9,9 0,0 1,6.27 1.52,13.13 13.13,0 0,0 -11.4,-6.66h-0.92A12.92,12.92 0,0 0,18.02 17.26c0.39,0.34 0.84,0.78 1.79,1.71 1.75,1.8 6.28,3.55 6.29,3.8z">
-        <aapt:attr name="android:fillColor">
-            <gradient android:centerX="20.335398"
-                android:centerY="43.06355"
-                android:gradientRadius="26.210001" android:type="radial">
-                <item android:color="#FF3A8EE6" android:offset="0.35"/>
-                <item android:color="#FF9059FF" android:offset="0.67"/>
-                <item android:color="#FFC139E6" android:offset="1"/>
-            </gradient>
-        </aapt:attr>
-    </path>
-    <path android:pathData="M26.1,22.77c0,0.65 -2.33,2.88 -3.13,2.88 -7.4,0 -8.6,4.48 -8.6,4.48a11,11 0,0 0,6.13 8.52l0.44,0.21a12.13,12.13 0,0 0,0.76 0.31,11.91 11.91,0 0,0 3.39,0.66c13,0.61 15.49,-15.52 6.13,-20.2a9,9 0,0 1,6.27 1.52,13.13 13.13,0 0,0 -11.4,-6.66h-0.92A12.92,12.92 0,0 0,18.02 17.26c0.39,0.34 0.84,0.78 1.79,1.71 1.75,1.8 6.28,3.55 6.29,3.8z">
-        <aapt:attr name="android:fillColor">
-            <gradient android:centerX="25.19928"
-                android:centerY="23.053501"
-                android:gradientRadius="13.91751" android:type="radial">
-                <item android:color="#009059FF" android:offset="0.21"/>
-                <item android:color="#996E008B" android:offset="0.97"/>
-            </gradient>
-        </aapt:attr>
-    </path>
-    <path android:pathData="M16.79,16.39l0.54,0.36a12,12 0,0 1,-0.07 -6.36A19.21,19.21 0,0 0,11.02 15.23c0.1,0 3.87,-0.07 5.77,1.16z">
-        <aapt:attr name="android:fillColor">
-            <gradient android:centerX="24.271004"
-                android:centerY="4.029999"
-                android:gradientRadius="18.444002" android:type="radial">
-                <item android:color="#FFFFE226" android:offset="0.1"/>
-                <item android:color="#FFFF7139" android:offset="0.79"/>
-            </gradient>
-        </aapt:attr>
-    </path>
-    <path android:pathData="M0.26,28.35c2,11.88 12.77,20.94 25,21.29 11.3,0.32 18.52,-6.25 21.51,-12.65a23.55,23.55 0,0 0,0.81 -17.57c0.92,6 -2.15,11.87 -6.94,15.82 -9.34,7.61 -18.28,4.59 -20.09,3.35l-0.38,-0.19c-5.45,-2.6 -7.7,-7.56 -7.22,-11.82A6.67,6.67 0,0 1,6.76 22.77a9.8,9.8 0,0 1,9.57 -0.38A13,13 0,0 0,26.1 22.77c0,-0.21 -4.54,-2 -6.3,-3.76 -1,-0.93 -1.4,-1.37 -1.79,-1.71a10.031,10.031 0,0 0,-0.68 -0.51l-0.54,-0.36c-1.9,-1.23 -5.67,-1.16 -5.8,-1.16 -1,-1.31 -1,-5.62 -0.9,-6.52a4.39,4.39 0,0 0,-0.87 0.46,19.37 19.37,0 0,0 -2.54,2.18 22.23,22.23 0,0 0,-2.43 2.92,21.88 21.88,0 0,0 -3.5,7.88c-0.02,0.02 -0.95,4.07 -0.49,6.16z">
-        <aapt:attr name="android:fillColor">
-            <gradient android:centerX="41.62241"
-                android:centerY="-8.146102"
-                android:gradientRadius="89.61946" android:type="radial">
-                <item android:color="#FFFFF44F" android:offset="0.11"/>
-                <item android:color="#FFFF980E" android:offset="0.46"/>
-                <item android:color="#FFFF3647" android:offset="0.72"/>
-                <item android:color="#FFE31587" android:offset="0.9"/>
-            </gradient>
-        </aapt:attr>
-    </path>
-    <path android:pathData="M35.24,18.08a14.35,14.35 0,0 1,2.42 3.15l0.39,0.32c5.9,5.44 2.81,13.13 2.58,13.68 4.79,-4 7.86,-9.79 6.94,-15.82 -3,-7.34 -7.94,-10.3 -12,-16.74 -0.2,-0.33 -0.41,-0.65 -0.61,-1a5.94,5.94 0,0 1,-0.29 -0.54,4.92 4.92,0 0,1 -0.39,-1 0.07,0.07 0,0 0,-0.05 -0.07L34.12,0.06c-1,0.47 -6.97,9.93 1.12,18.02z">
-        <aapt:attr name="android:fillColor">
-            <gradient android:centerX="6.13348"
-                android:centerY="14.259094"
-                android:gradientRadius="95.36752" android:type="radial">
-                <item android:color="#FFFFF44F" android:offset="0"/>
-                <item android:color="#FFFF980E" android:offset="0.3"/>
-                <item android:color="#FFFF3647" android:offset="0.57"/>
-                <item android:color="#FFE31587" android:offset="0.74"/>
-            </gradient>
-        </aapt:attr>
-    </path>
-    <path android:pathData="M38.02,21.55l-0.39,-0.32 -0.17,-0.12a9,9 0,0 0,-6.27 -1.52c9.36,4.68 6.85,20.81 -6.13,20.2a11.91,11.91 0,0 1,-3.39 -0.66,12.13 12.13,0 0,1 -0.76,-0.31l-0.44,-0.21c1.81,1.24 10.75,4.26 20.09,-3.35 0.3,-0.58 3.39,-8.27 -2.54,-13.71z">
-        <aapt:attr name="android:fillColor">
-            <gradient android:centerX="24.90772"
-                android:centerY="9.289151"
-                android:gradientRadius="56.180176" android:type="radial">
-                <item android:color="#FFFFF44F" android:offset="0.14"/>
-                <item android:color="#FFFF980E" android:offset="0.48"/>
-                <item android:color="#FFFF3647" android:offset="0.66"/>
-                <item android:color="#FFE31587" android:offset="0.9"/>
-            </gradient>
-        </aapt:attr>
-    </path>
-    <path android:pathData="M14.37,30.09s1.2,-4.48 8.6,-4.48c0.8,0 3.09,-2.23 3.13,-2.88a13,13 0,0 1,-9.77 -0.38,9.8 9.8,0 0,0 -9.57,0.38 6.67,6.67 0,0 0,6.16 3.88c-0.48,4.26 1.77,9.22 7.22,11.82l0.36,0.18a11,11 0,0 1,-6.13 -8.52z">
-        <aapt:attr name="android:fillColor">
-            <gradient android:centerX="37.568623"
-                android:centerY="13.450078"
-                android:gradientRadius="60.55354" android:type="radial">
-                <item android:color="#FFFFF44F" android:offset="0.09"/>
-                <item android:color="#FFFF980E" android:offset="0.63"/>
-            </gradient>
-        </aapt:attr>
-    </path>
-    <path android:pathData="M50.15,18.05c-1.13,-2.72 -3.43,-5.66 -5.23,-6.59a26.79,26.79 0,0 1,2.64 7.91c-3,-7.34 -7.94,-10.3 -12,-16.74 -0.2,-0.33 -0.41,-0.65 -0.61,-1a5.94,5.94 0,0 1,-0.29 -0.54,4.92 4.92,0 0,1 -0.39,-1 0.07,0.07 0,0 0,-0.05 -0.07h-0.07a19,19 0,0 0,-9 14.46h0.92a13.13,13.13 0,0 1,11.4 6.66,9 9,0 0,0 -6.27,-1.52c9.36,4.68 6.85,20.81 -6.13,20.2a11.91,11.91 0,0 1,-3.39 -0.66,12.13 12.13,0 0,1 -0.76,-0.31l-0.44,-0.21 -0.38,-0.19 0.36,0.18a11,11 0,0 1,-6.13 -8.52s1.2,-4.48 8.6,-4.48c0.8,0 3.09,-2.23 3.13,-2.88 0,-0.21 -4.54,-2 -6.3,-3.76 -1,-0.93 -1.4,-1.37 -1.79,-1.71A10.031,10.031 0,0 0,17.29 16.77a12,12 0,0 1,-0.07 -6.36,19.21 19.21,0 0,0 -6.2,4.82c-1,-1.31 -1,-5.62 -0.9,-6.52a4.39,4.39 0,0 0,-0.87 0.46,19.37 19.37,0 0,0 -2.54,2.18 22.23,22.23 0,0 0,-2.43 2.92,21.88 21.88,0 0,0 -3.5,7.88v0.18a39.815,39.815 0,0 0,-0.3 1.64A31.77,31.77 0,0 0,0.02 27.6L0.02,27.77a26,26 0,0 0,51.7 4.41l0.12,-1a26.9,26.9 0,0 0,-1.69 -13.13zM47.56,19.39z">
-        <aapt:attr name="android:fillColor">
-            <gradient android:endX="9.339838" android:endY="45.985138"
-                android:startX="41.88179" android:startY="7.6808558" android:type="linear">
-                <item android:color="#CCFFF44F" android:offset="0.17"/>
-                <item android:color="#00FFF44F" android:offset="0.6"/>
-            </gradient>
-        </aapt:attr>
-    </path>
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:aapt="http://schemas.android.com/aapt"
+    android:width="108dp"
+    android:height="108dp"
+    android:viewportWidth="32"
+    android:viewportHeight="32">
+  <group android:scaleX="0.64166665"
+      android:scaleY="0.64166665"
+      android:translateX="5.733333"
+      android:translateY="5.733333">
+    <path
+        android:pathData="M10.562,5.498C10.516,5.479 10.462,5.49 10.425,5.525l-4.89,4.803c-0.118,0.116 -0.214,0.344 -0.213,0.51 0.085,9 5.054,11.624 10.426,15.6 0.133,0.098 0.349,0.098 0.482,0 5.476,-3.998 10.332,-6.331 10.447,-15.521 0.002,-0.165 -0.093,-0.395 -0.211,-0.512l-4.931,-4.88c-0.08,-0.08 -0.216,-0.023 -0.215,0.09l-0.029,4.701c-0.001,0.166 -0.135,0.299 -0.301,0.299L10.959,10.615c-0.166,0 -0.3,-0.133 -0.301,-0.299L10.641,5.615c-0,-0.051 -0.031,-0.098 -0.079,-0.117z">
+      <aapt:attr name="android:fillColor">
+        <gradient 
+            android:startX="5.322"
+            android:startY="5.488"
+            android:endX="26.678"
+            android:endY="26.512"
+            android:type="linear">
+          <item android:offset="0" android:color="#AEFF0E00"/>
+          <item android:offset="1" android:color="#FFFFD500"/>
+        </gradient>
+      </aapt:attr>
+    </path>
+    <path
+        android:pathData="m21.32,15.949c-1.995,3.338 -2.095,3.239 -3.345,1.988"
+        android:fillColor="#ffffff"/>
+    <path
+        android:pathData="m10.627,15.949c1.995,3.338 2.095,3.239 3.345,1.988"
+        android:fillColor="#ffffff"/>
+  </group>
 </vector>
diff --git a/mobile/android/moz.configure b/mobile/android/moz.configure
index 0e2a6efdc9..083c9cac2d 100644
--- a/mobile/android/moz.configure
+++ b/mobile/android/moz.configure
@@ -116,7 +116,7 @@ imply_option("MOZ_SERVICES_HEALTHREPORT", True)
 imply_option("MOZ_GECKOVIEW_HISTORY", True)
 
 imply_option("MOZ_APP_UA_NAME", "Firefox")
-imply_option("MOZ_APP_VENDOR", "Mozilla")
+imply_option("MOZ_APP_VENDOR", "IronFox OSS")
 imply_option("MOZ_APP_ID", "{aa3c5121-dab2-40e2-81ca-7ea25febc110}")
 imply_option("BROWSER_CHROME_URL", "chrome://geckoview/content/geckoview.xhtml")
 
