diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/about/AboutFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/about/AboutFragment.kt
index e43dec84c6..4ddd7a6678 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/about/AboutFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/about/AboutFragment.kt
@@ -208,28 +208,28 @@ class AboutFragment(
             AboutPageItem(
                 AboutItem.ExternalLink(
                     SUPPORT,
-                    SupportUtils.getSumoURLForTopic(context, SupportUtils.SumoTopic.HELP),
+                    SupportUtils.IF_SUPPORT_URL,
                 ),
                 getString(R.string.about_support),
             ),
-            AboutPageItem(
-                AboutItem.Crashes,
-                getString(R.string.about_crashes),
-            ),
-            AboutPageItem(
-                AboutItem.ExternalLink(
-                    PRIVACY_NOTICE,
-                    SupportUtils.getMozillaPageUrl(SupportUtils.MozillaPage.PRIVATE_NOTICE),
-                ),
-                getString(R.string.about_privacy_notice),
-            ),
-            AboutPageItem(
-                AboutItem.ExternalLink(
-                    RIGHTS,
-                    SupportUtils.getSumoURLForTopic(context, SupportUtils.SumoTopic.YOUR_RIGHTS),
-                ),
-                getString(R.string.about_know_your_rights),
-            ),
+//            AboutPageItem(
+//                AboutItem.Crashes,
+//                getString(R.string.about_crashes),
+//            ),
+//            AboutPageItem(
+//                AboutItem.ExternalLink(
+//                    PRIVACY_NOTICE,
+//                    SupportUtils.getMozillaPageUrl(SupportUtils.MozillaPage.PRIVATE_NOTICE),
+//                ),
+//                getString(R.string.about_privacy_notice),
+//            ),
+//            AboutPageItem(
+//                AboutItem.ExternalLink(
+//                    RIGHTS,
+//                    SupportUtils.getSumoURLForTopic(context, SupportUtils.SumoTopic.YOUR_RIGHTS),
+//                ),
+//                getString(R.string.about_know_your_rights),
+//            ),
             AboutPageItem(
                 AboutItem.ExternalLink(LICENSING_INFO, ABOUT_LICENSE_URL),
                 getString(R.string.about_licensing_information),
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt
index c13ecb6c86..a9e364d69a 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt
@@ -21,15 +21,16 @@ import java.net.URLEncoder
 import java.util.Locale
 
 object SupportUtils {
-    const val RATE_APP_URL = "market://details?id=" + BuildConfig.APPLICATION_ID
-    const val POCKET_TRENDING_URL = "https://getpocket.com/fenix-top-articles"
-    const val WIKIPEDIA_URL = "https://www.wikipedia.org/"
-    const val FENIX_PLAY_STORE_URL = "https://play.google.com/store/apps/details?id=${BuildConfig.APPLICATION_ID}"
-    const val GOOGLE_URL = "https://www.google.com/"
-    const val GOOGLE_US_URL = "https://www.google.com/webhp?client=firefox-b-1-m&channel=ts"
-    const val GOOGLE_XX_URL = "https://www.google.com/webhp?client=firefox-b-m&channel=ts"
-    const val WHATS_NEW_URL = "https://www.mozilla.org/firefox/android/notes"
+    const val RATE_APP_URL = ""
+    const val POCKET_TRENDING_URL = ""
+    const val WIKIPEDIA_URL = ""
+    const val FENIX_PLAY_STORE_URL = ""
+    const val GOOGLE_URL = ""
+    const val GOOGLE_US_URL = ""
+    const val GOOGLE_XX_URL = ""
+    const val WHATS_NEW_URL = "https://gitlab.com/ironfox-oss/IronFox/-/releases"
     const val FXACCOUNT_SUMO_URL = "https://support.mozilla.org/kb/access-mozilla-services-firefox-account"
+    const val IF_SUPPORT_URL = "https://gitlab.com/ironfox-oss/IronFox/-/issues"
 
     // This is locale-less on purpose so that the content negotiation happens on the AMO side because the current
     // user language might not be supported by AMO and/or the language might not be exactly what AMO is expecting
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt
index e5ab956ac3..8f88bf456e 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt
@@ -82,7 +82,7 @@ class OnboardingFragment : Fragment() {
             pagesToDisplay(
                 showDefaultBrowserPage = isNotDefaultBrowser(this) && !isDefaultBrowserPromptSupported(),
                 showNotificationPage = canShowNotificationPage(),
-                showAddWidgetPage = canShowAddSearchWidgetPrompt(AppWidgetManager.getInstance(activity)),
+                showAddWidgetPage = false,
             ).toMutableList()
         }
     }
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingMapper.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingMapper.kt
index 795fcd7b7e..5c0cb8796e 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingMapper.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingMapper.kt
@@ -32,7 +32,7 @@ internal fun Collection<OnboardingCardData>.toPageUiData(
         .sortedBy { it.ordering }
         .mapIndexed { index, onboardingCardData ->
             // only first onboarding card shows privacy caption
-            onboardingCardData.toPageUiData(if (index == 0) privacyCaption else null)
+            onboardingCardData.toPageUiData(null)
         }
 }
 
@@ -112,7 +112,6 @@ private fun OnboardingCardData.toPageUiData(privacyCaption: Caption?) = Onboardi
     themeOptions = extraData?.customizationThemeData
         ?.takeIf { it.isNotEmpty() }
         ?.toOnboardingThemeOptions(),
-    termsOfService = extraData?.termOfServiceData?.toOnboardingTermsOfService(),
     marketingData = extraData?.marketingData?.toOnboardingMarketingData(),
 )
 
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/TermsOfServiceOnboardingPage.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/TermsOfServiceOnboardingPage.kt
index 10488889d7..02572c03a1 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/TermsOfServiceOnboardingPage.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/TermsOfServiceOnboardingPage.kt
@@ -135,22 +135,22 @@ private fun BodyText(
             modifier = Modifier.padding(horizontal = 10.dp),
         ) {
             val lineOneState = LinkTextState(
-                text = it.lineOneLinkText,
-                url = it.lineOneLinkUrl,
-                onClick = eventHandler::onTermsOfServiceLinkClicked,
+                text = "",
+                url = "",
+                onClick = {},
             )
             val lineTwoState = LinkTextState(
-                text = it.lineTwoLinkText,
-                url = it.lineTwoLinkUrl,
-                onClick = eventHandler::onPrivacyNoticeLinkClicked,
+                text = "",
+                url = "",
+                onClick = {},
             )
             val lineThreeState = LinkTextState(
-                text = it.lineThreeLinkText,
+                text = "",
                 url = "",
-                onClick = { _ -> eventHandler.onManagePrivacyPreferencesLinkClicked() },
+                onClick = {},
             )
             LinkText(
-                text = it.lineOneText.updateFirstPlaceholder(it.lineOneLinkText),
+                text = "",
                 linkTextStates = listOf(
                     lineOneState,
                 ),
@@ -162,7 +162,7 @@ private fun BodyText(
             )
 
             LinkText(
-                text = it.lineTwoText.updateFirstPlaceholder(it.lineTwoLinkText),
+                text = "",
                 linkTextStates = listOf(
                     lineTwoState,
                 ),
@@ -174,7 +174,7 @@ private fun BodyText(
             )
 
             LinkText(
-                text = it.lineThreeText.updateFirstPlaceholder(it.lineThreeLinkText),
+                text = "",
                 linkTextStates = listOf(
                     lineThreeState,
                 ),
@@ -201,14 +201,14 @@ private fun OnboardingPagePreview() {
                 title = stringResource(id = R.string.onboarding_welcome_to_firefox),
                 description = stringResource(id = R.string.onboarding_terms_of_service_sub_header_2),
                 termsOfService = OnboardingTermsOfService(
-                    lineOneText = stringResource(id = R.string.onboarding_term_of_service_line_one_2),
-                    lineOneLinkText = stringResource(id = R.string.onboarding_term_of_service_line_one_link_text_2),
-                    lineOneLinkUrl = "URL",
-                    lineTwoText = stringResource(id = R.string.onboarding_term_of_service_line_two_2),
-                    lineTwoLinkText = stringResource(id = R.string.onboarding_term_of_service_line_two_link_text),
-                    lineTwoLinkUrl = "URL",
-                    lineThreeText = stringResource(id = R.string.onboarding_term_of_service_line_three),
-                    lineThreeLinkText = stringResource(id = R.string.onboarding_term_of_service_line_three_link_text),
+                    lineOneText = "",
+                    lineOneLinkText = "",
+                    lineOneLinkUrl = "",
+                    lineTwoText = "",
+                    lineTwoLinkText = "",
+                    lineTwoLinkUrl = "",
+                    lineThreeText = "",
+                    lineThreeLinkText = "",
                 ),
                 imageRes = R.drawable.ic_firefox,
                 primaryButton = Action(
diff --git a/mobile/android/moz.configure b/mobile/android/moz.configure
index 0e2a6efdc9..3d7ff4a88f 100644
--- a/mobile/android/moz.configure
+++ b/mobile/android/moz.configure
@@ -116,7 +116,7 @@ imply_option("MOZ_SERVICES_HEALTHREPORT", True)
 imply_option("MOZ_GECKOVIEW_HISTORY", True)
 
 imply_option("MOZ_APP_UA_NAME", "Firefox")
-imply_option("MOZ_APP_VENDOR", "Mozilla")
+imply_option("MOZ_APP_VENDOR", "IronFox OSS")
 imply_option("MOZ_APP_ID", "{aa3c5121-dab2-40e2-81ca-7ea25febc110}")
 imply_option("BROWSER_CHROME_URL", "chrome://geckoview/content/geckoview.xhtml")
 
@@ -153,3 +153,5 @@ set_config(
     "MOZ_ANDROID_FAT_AAR_ARCHITECTURES",
     depends("MOZ_ANDROID_FAT_AAR_ARCHITECTURES")(lambda x: x),
 )
+
+include("ironfox.configure")
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 9197b274db..2d791fa7b7 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -395,10 +395,10 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         default = false,
     )
 
-    var hasAcceptedTermsOfService by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_terms_accepted),
-        default = false,
-    )
+    var hasAcceptedTermsOfService = true
+//        appContext.getPreferenceKey(R.string.pref_key_terms_accepted),
+//        default = false,
+//    )
 
     /**
      * The daily usage ping is not normally tied to normal telemetry.  We set the default value to
@@ -436,10 +436,10 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     val shouldShowSecurityPinWarning: Boolean
         get() = secureWarningCount.underMaxCount()
 
-    var shouldShowPrivacyPopWindow by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_privacy_pop_window),
-        default = true,
-    )
+    var shouldShowPrivacyPopWindow = false
+//        appContext.getPreferenceKey(R.string.pref_key_privacy_pop_window),
+//        default = true,
+//    )
 
     var shouldUseLightTheme by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_light_theme),
