diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/doh/DohSettingsProvider.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/doh/DohSettingsProvider.kt
index 028be832d6..50bf4971ff 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/doh/DohSettingsProvider.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/doh/DohSettingsProvider.kt
@@ -37,17 +37,92 @@ internal class DefaultDohSettingsProvider(
 
     private val cloudflare = Provider.BuiltIn(
         url = cloudflareUri,
-        name = "Cloudflare",
-        default = dohDefaultProviderUrl.isBlank() || dohDefaultProviderUrl == cloudflareUri,
+        name = "Cloudflare - Unfiltered",
+        default = dohDefaultProviderUrl == cloudflareUri,
     )
     private val nextDns = Provider.BuiltIn(
         url = nextDnsUri,
-        name = "NextDNS",
+        name = "NextDNS (Public) - Unfiltered",
         default = dohDefaultProviderUrl == nextDnsUri,
     )
+    private val quad9 = Provider.BuiltIn(
+        url = quad9Uri,
+        name = "Quad9",
+        default = dohDefaultProviderUrl.isNullOrBlank() || dohDefaultProviderUrl == quad9Uri,
+     )
+    private val dns0Zero = Provider.BuiltIn(
+        url = dns0ZeroUri,
+        name = "DNS0 (ZERO)",
+        default = dohDefaultProviderUrl == dns0ZeroUri,
+    )
+    private val dns0Standard = Provider.BuiltIn(
+        url = dns0StandardUri,
+        name = "DNS0",
+        default = dohDefaultProviderUrl == dns0StandardUri,
+    )
+    private val mullvadBase = Provider.BuiltIn(
+        url = mullvadBaseUri,
+        name = "Mullvad - Base",
+        default = dohDefaultProviderUrl == mullvadBaseUri,
+    )
+    private val adGuardStandard = Provider.BuiltIn(
+        url = adGuardStandardUri,
+        name = "AdGuard (Public)",
+        default = dohDefaultProviderUrl == adGuardStandardUri,
+    )
+    private val mullvadUnfiltered = Provider.BuiltIn(
+        url = mullvadUnfilteredUri,
+        name = "Mullvad - Unfiltered",
+        default = dohDefaultProviderUrl == mullvadUnfilteredUri,
+    )
+    private val wikimedia = Provider.BuiltIn(
+        url = wikimediaUri,
+        name = "Wikimedia - Unfiltered",
+        default = dohDefaultProviderUrl == wikimediaUri,
+     )
+    private val adGuardUnfiltered = Provider.BuiltIn(
+        url = adGuardUnfilteredUri,
+        name = "AdGuard (Public) - Unfiltered",
+        default = dohDefaultProviderUrl == adGuardUnfilteredUri,
+    )
+    private val dns0Kids = Provider.BuiltIn(
+        url = dns0KidsUri,
+        name = "DNS0 - Kids",
+        default = dohDefaultProviderUrl == dns0KidsUri,
+    )
+    private val mullvadFamily = Provider.BuiltIn(
+        url = mullvadFamilyUri,
+        name = "Mullvad - Family",
+        default = dohDefaultProviderUrl == mullvadFamilyUri,
+    )
+    private val adGuardFamily = Provider.BuiltIn(
+        url = adGuardFamilyUri,
+        name = "AdGuard (Public) - Family",
+        default = dohDefaultProviderUrl == adGuardFamilyUri,
+    )
+    private val mullvadExtended = Provider.BuiltIn(
+        url = mullvadExtendedUri,
+        name = "Mullvad - Extended",
+        default = dohDefaultProviderUrl == mullvadExtendedUri,
+    )
+    private val mullvadAll = Provider.BuiltIn(
+        url = mullvadAllUri,
+        name = "Mullvad - All",
+        default = dohDefaultProviderUrl == mullvadAllUri,
+    )
+    private val cloudflareMalware = Provider.BuiltIn(
+        url = cloudflareMalwareUri,
+        name = "Cloudflare - Malware Protection",
+        default = dohDefaultProviderUrl == cloudflareMalwareUri,
+    )
+    private val cloudflareFamily = Provider.BuiltIn(
+        url = cloudflareFamilyUri,
+        name = "Cloudflare - Family",
+        default = dohDefaultProviderUrl == cloudflareFamilyUri,
+    )
     private val providerUrl = settings.dohProviderUrl
     private val custom = Provider.Custom(
-        url = if (providerUrl != cloudflareUri && providerUrl != nextDnsUri) {
+        url = if (providerUrl != cloudflareUri && providerUrl != nextDnsUri && providerUrl != quad9Uri && providerUrl != dns0ZeroUri && providerUrl != dns0StandardUri && providerUrl != mullvadBaseUri && providerUrl != adGuardStandardUri && providerUrl != mullvadUnfilteredUri && providerUrl != wikimediaUri && providerUrl != adGuardUnfilteredUri && providerUrl != dns0KidsUri && providerUrl != mullvadFamilyUri && providerUrl != adGuardFamilyUri && providerUrl != mullvadExtendedUri && providerUrl != mullvadAllUri && providerUrl != cloudflareMalwareUri && providerUrl != cloudflareFamilyUri) {
             providerUrl
         } else {
             ""
@@ -55,8 +130,23 @@ internal class DefaultDohSettingsProvider(
     )
 
     override fun getDefaultProviders(): List<Provider> = listOf(
-        cloudflare,
+        quad9,
+        dns0Zero,
+        dns0Standard,
+        mullvadBase,
+        adGuardStandard,
+        mullvadUnfiltered,
+        wikimedia,
         nextDns,
+        adGuardUnfiltered,
+        dns0Kids,
+        mullvadFamily,
+        adGuardFamily,
+        mullvadExtended,
+        mullvadAll,
+        cloudflareMalware,
+        cloudflare,
+        cloudflareFamily,
         custom,
     )
 
@@ -79,6 +169,21 @@ internal class DefaultDohSettingsProvider(
                 when (settings.dohProviderUrl) {
                     cloudflareUri -> cloudflare
                     nextDnsUri -> nextDns
+                    quad9Uri -> quad9
+                    dns0ZeroUri -> dns0Zero
+                    dns0StandardUri -> dns0Standard
+                    mullvadBaseUri -> mullvadBase
+                    adGuardStandardUri -> adGuardStandard
+                    mullvadUnfilteredUri -> mullvadUnfiltered
+                    wikimediaUri -> wikimedia
+                    adGuardUnfilteredUri -> adGuardUnfiltered
+                    dns0KidsUri -> dns0Kids
+                    mullvadFamilyUri -> mullvadFamily
+                    adGuardFamilyUri -> adGuardFamily
+                    mullvadExtendedUri -> mullvadExtended
+                    mullvadAllUri -> mullvadAll
+                    cloudflareMalwareUri -> cloudflareMalware
+                    cloudflareFamilyUri -> cloudflareFamily
                     "" -> getDefaultProviders().first()
                     else -> custom
                 }
@@ -122,5 +227,50 @@ internal class DefaultDohSettingsProvider(
 
         @VisibleForTesting
         val nextDnsUri = "https://firefox.dns.nextdns.io/"
+
+        @VisibleForTesting
+        val quad9Uri = "https://dns.quad9.net/dns-query"
+
+        @VisibleForTesting
+        val dns0ZeroUri = "https://zero.dns0.eu"
+
+        @VisibleForTesting
+        val dns0StandardUri = "https://dns0.eu"
+
+        @VisibleForTesting
+        val mullvadBaseUri = "https://base.dns.mullvad.net/dns-query"
+
+        @VisibleForTesting
+        val adGuardStandardUri = "https://dns.adguard-dns.com/dns-query"
+
+        @VisibleForTesting
+        val mullvadUnfilteredUri = "https://dns.mullvad.net/dns-query"
+
+        @VisibleForTesting
+        val wikimediaUri = "https://wikimedia-dns.org/dns-query"
+
+        @VisibleForTesting
+        val adGuardUnfilteredUri = "https://unfiltered.adguard-dns.com/dns-query"
+
+        @VisibleForTesting
+        val dns0KidsUri = "https://kids.dns0.eu"
+
+        @VisibleForTesting
+        val mullvadFamilyUri = "https://family.dns.mullvad.net/dns-query"
+
+        @VisibleForTesting
+        val adGuardFamilyUri = "https://family.adguard-dns.com/dns-query"
+
+        @VisibleForTesting
+        val mullvadExtendedUri = "https://extended.dns.mullvad.net/dns-query"
+
+        @VisibleForTesting
+        val mullvadAllUri = "https://all.dns.mullvad.net/dns-query"
+
+        @VisibleForTesting
+        val cloudflareMalwareUri = "https://security.cloudflare-dns.com/dns-query"
+
+        @VisibleForTesting
+        val cloudflareFamilyUri = "https://family.cloudflare-dns.com/dns-query"
     }
 }
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 9197b274db..f89b2f7ea6 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -2364,7 +2364,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     val showDohEntryPoint by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_doh_settings_enabled),
-        default = { FxNimbus.features.doh.value().showUi },
+        default = { true },
         featureFlag = true,
     )
 
@@ -2377,7 +2377,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     private var trrMode by intPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_doh_settings_mode),
-        default = DOH_SETTINGS_DEFAULT,
+        default = DOH_SETTINGS_MAX,
     )
 
     /**
@@ -2395,7 +2395,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     val dohDefaultProviderUrl by stringPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_doh_default_provider_uri),
-        default = CLOUDFLARE_URI,
+        default = "https://dns.quad9.net/dns-query",
     )
 
     /**
