diff --git a/toolkit/components/pdfjs/PdfJsDefaultPrefs.js b/toolkit/components/pdfjs/PdfJsDefaultPrefs.js
index 396cc493c8..8a9fb457ff 100644
--- a/toolkit/components/pdfjs/PdfJsDefaultPrefs.js
+++ b/toolkit/components/pdfjs/PdfJsDefaultPrefs.js
@@ -37,11 +37,11 @@ pref("pdfjs.enableHighlightFloatingButton", false);
 pref("pdfjs.enableNewAltTextWhenAddingImage", true);
 pref("pdfjs.enablePermissions", false);
 pref("pdfjs.enablePrintAutoRotate", true);
-pref("pdfjs.enableScripting", true);
-pref("pdfjs.enableSignatureEditor", false);
+pref("pdfjs.enableScripting", false);
+pref("pdfjs.enableSignatureEditor", true);
 pref("pdfjs.enableUpdatedAddImage", false);
-pref("pdfjs.enableXfa", true);
-pref("pdfjs.externalLinkTarget", 0);
+pref("pdfjs.enableXfa", false);
+pref("pdfjs.externalLinkTarget", 2);
 pref("pdfjs.forcePageColors", false);
 pref("pdfjs.highlightEditorColors", "yellow=#FFFF98,green=#53FFBC,blue=#80EBFF,pink=#FFCBE6,red=#FF4F5F");
 pref("pdfjs.historyUpdateUrl", false);
@@ -50,7 +50,7 @@ pref("pdfjs.pageColorsBackground", "Canvas");
 pref("pdfjs.pageColorsForeground", "CanvasText");
 pref("pdfjs.pdfBugEnabled", false);
 pref("pdfjs.scrollModeOnLoad", -1);
-pref("pdfjs.sidebarViewOnLoad", -1);
+pref("pdfjs.sidebarViewOnLoad", 2);
 pref("pdfjs.spreadModeOnLoad", -1);
 pref("pdfjs.textLayerMode", 1);
 pref("pdfjs.viewOnLoad", 0);
diff --git a/toolkit/components/pdfjs/PdfJsOverridePrefs.js b/toolkit/components/pdfjs/PdfJsOverridePrefs.js
index 0961800f40..690614d91b 100644
--- a/toolkit/components/pdfjs/PdfJsOverridePrefs.js
+++ b/toolkit/components/pdfjs/PdfJsOverridePrefs.js
@@ -29,7 +29,7 @@
     pref("pdfjs.enableSignatureEditor", true);
   #else
     pref("pdfjs.enableHighlightFloatingButton", false);
-    pref("pdfjs.enableSignatureEditor", false);
+    pref("pdfjs.enableSignatureEditor", true);
   #endif
 
   pref("pdfjs.enableAltTextForEnglish", false);
