diff --git a/mobile/android/android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/Engine.kt b/mobile/android/android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/Engine.kt
index 6088ad065c..2c553a6c18 100644
--- a/mobile/android/android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/Engine.kt
+++ b/mobile/android/android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/Engine.kt
@@ -109,6 +109,12 @@ interface Engine : WebExtensionRuntime, TranslationsRuntime, DataCleanable {
         OFF,
     }
 
+    enum class RefererXOriginPolicy {
+        REFERER_XORIGIN_ALWAYS,
+        REFERER_XORIGIN_IF_BASE_DOMAINS_MATCH,
+        REFERER_XORIGIN_IF_HOSTS_MATCH,
+    }
+
     /**
      * Makes sure all required engine initialization logic is executed. The
      * details are specific to individual implementations, but the following must be true:
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/RefererSettingsFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/RefererSettingsFragment.kt
new file mode 100644
index 0000000000..c0947e5d21
--- /dev/null
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/RefererSettingsFragment.kt
@@ -0,0 +1,77 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings
+
+import android.os.Bundle
+import androidx.preference.Preference
+import androidx.preference.PreferenceFragmentCompat
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.ext.requireComponents
+import org.mozilla.fenix.ext.settings
+import org.mozilla.fenix.ext.showToolbar
+import org.mozilla.fenix.R
+import org.mozilla.fenix.utils.view.addToRadioGroup
+
+/**
+ * Lets the user configure the cross-origin referer policy.
+ */
+
+@Suppress("TooManyFunctions")
+class RefererSettingsFragment : PreferenceFragmentCompat() {
+    private lateinit var radioRefererXOriginAlways: RadioButtonPreference
+    private lateinit var radioRefererXOriginBaseDomainsMatch: RadioButtonPreference
+    private lateinit var radioRefererXOriginHostsMatch: RadioButtonPreference
+
+    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
+        setPreferencesFromResource(R.xml.referer_preferences, rootKey)
+        setupPreferences()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        showToolbar(getString(R.string.preference_referer_policy))
+    }
+
+    private fun setupPreferences() {
+        bindRefererXOriginAlways()
+        bindRefererXOriginBaseDomainsMatch()
+        bindRefererXOriginHostsMatch()
+        setupRadioGroups()
+    }
+
+    private fun setupRadioGroups() {
+        addToRadioGroup(
+            radioRefererXOriginAlways,
+            radioRefererXOriginBaseDomainsMatch,
+            radioRefererXOriginHostsMatch
+        )
+    }
+
+    private fun bindRefererXOriginAlways() {
+        radioRefererXOriginAlways = requirePreference(R.string.pref_key_referer_policy_always)
+        radioRefererXOriginAlways.onClickListener {
+            updateEngineRefererXOriginPolicy()
+        }
+    }
+
+    private fun bindRefererXOriginBaseDomainsMatch() {
+        radioRefererXOriginBaseDomainsMatch = requirePreference(R.string.pref_key_referer_policy_base_domains_match)
+        radioRefererXOriginBaseDomainsMatch.onClickListener {
+            updateEngineRefererXOriginPolicy()
+        }
+    }
+
+    private fun bindRefererXOriginHostsMatch() {
+        radioRefererXOriginHostsMatch = requirePreference(R.string.pref_key_referer_policy_hosts_match)
+        radioRefererXOriginHostsMatch.onClickListener {
+            updateEngineRefererXOriginPolicy()
+        }
+    }
+
+    private fun updateEngineRefererXOriginPolicy() {
+        requireContext().components.core.engine.settings.refererXOriginPolicy =
+            requireContext().settings().getRefererXOriginPolicy()
+    }
+}
diff --git a/mobile/android/fenix/app/src/main/res/navigation/nav_graph.xml b/mobile/android/fenix/app/src/main/res/navigation/nav_graph.xml
index 87f445940d..c5a68eb685 100644
--- a/mobile/android/fenix/app/src/main/res/navigation/nav_graph.xml
+++ b/mobile/android/fenix/app/src/main/res/navigation/nav_graph.xml
@@ -651,6 +651,13 @@
             app:exitAnim="@anim/slide_out_left"
             app:popEnterAnim="@anim/slide_in_left"
             app:popExitAnim="@anim/slide_out_right" />
+        <action
+            android:id="@+id/action_settingsFragment_to_refererSettingsFragment"
+            app:destination="@id/refererSettingsFragment"
+            app:enterAnim="@anim/slide_in_right"
+            app:exitAnim="@anim/slide_out_left"
+            app:popEnterAnim="@anim/slide_in_left"
+            app:popExitAnim="@anim/slide_out_right" />
         <action
             android:id="@+id/action_settingsFragment_to_trackingProtectionFragment"
             app:destination="@id/trackingProtectionFragment"
@@ -908,6 +915,10 @@
         android:id="@+id/sponsoredStoriesSettings"
         android:name="org.mozilla.fenix.settings.SponsoredStoriesSettingsFragment"
         android:label="@string/preferences_debug_settings_custom_sponsored_stories_parameters" />
+    <fragment
+        android:id="@+id/refererSettingsFragment"
+        android:name="org.mozilla.fenix.settings.RefererSettingsFragment"
+        android:label="@string/preference_referer_policy" />
     <fragment
         android:id="@+id/trackingProtectionFragment"
         android:name="org.mozilla.fenix.settings.TrackingProtectionFragment">
diff --git a/mobile/android/fenix/app/src/main/res/xml/referer_preferences.xml b/mobile/android/fenix/app/src/main/res/xml/referer_preferences.xml
new file mode 100644
index 0000000000..f6e72872e6
--- /dev/null
+++ b/mobile/android/fenix/app/src/main/res/xml/referer_preferences.xml
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+   - License, v. 2.0. If a copy of the MPL was not distributed with this
+   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+<androidx.preference.PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:key="@string/pref_key_referer_policy_settings" >
+    <androidx.preference.PreferenceCategory
+        android:layout="@layout/preference_cat_style"
+        android:title="@string/preference_referer_policy"
+        app:iconSpaceReserved="false">
+        <org.mozilla.fenix.settings.RadioButtonPreference
+            android:defaultValue="true"
+            android:key="@string/pref_key_referer_policy_always"
+            android:title="@string/preference_referer_policy_always" />
+
+        <org.mozilla.fenix.settings.RadioButtonPreference
+            android:defaultValue="false"
+            android:key="@string/pref_key_referer_policy_base_domains_match"
+            android:title="@string/preference_referer_policy_base_domains_match" />
+
+        <org.mozilla.fenix.settings.RadioButtonPreference
+            android:defaultValue="false"
+            android:key="@string/pref_key_referer_policy_hosts_match"
+            android:title="@string/preference_referer_policy_hosts_match" />
+    </androidx.preference.PreferenceCategory>
+</androidx.preference.PreferenceScreen>
