diff --git a/mobile/android/android-components/components/service/mars/src/main/java/mozilla/components/service/mars/MarsTopSitesProvider.kt b/mobile/android/android-components/components/service/mars/src/main/java/mozilla/components/service/mars/MarsTopSitesProvider.kt
index b8430171ec..63f92e4dde 100644
--- a/mobile/android/android-components/components/service/mars/src/main/java/mozilla/components/service/mars/MarsTopSitesProvider.kt
+++ b/mobile/android/android-components/components/service/mars/src/main/java/mozilla/components/service/mars/MarsTopSitesProvider.kt
@@ -29,14 +29,14 @@ import java.io.File
 import java.io.IOException
 import java.util.Date
 
-private const val MARS_ENDPOINT_URL = "https://ads.mozilla.org/v1/ads"
+private const val MARS_ENDPOINT_URL = ""
 
 private const val REQUEST_BODY_CONTEXT_ID_KEY = "context_id"
 private const val REQUEST_BODY_PLACEMENTS_KEY = "placements"
 private const val REQUEST_BODY_PLACEMENT_KEY = "placement"
 private const val REQUEST_BODY_COUNT_KEY = "count"
 
-internal const val CACHE_FILE_NAME = "mozilla_components_service_mars_tiles.json"
+internal const val CACHE_FILE_NAME = ""
 
 /**
  * Provides access to the MARS API for fetching top sites tile.

diff --git a/mobile/android/android-components/components/service/mars/src/main/java/mozilla/components/service/mars/contile/ContileTopSitesProvider.kt b/mobile/android/android-components/components/service/mars/src/main/java/mozilla/components/service/mars/contile/ContileTopSitesProvider.kt
index 6f1de67d54..5037c18b66 100644
--- a/mobile/android/android-components/components/service/mars/src/main/java/mozilla/components/service/mars/contile/ContileTopSitesProvider.kt
+++ b/mobile/android/android-components/components/service/mars/src/main/java/mozilla/components/service/mars/contile/ContileTopSitesProvider.kt
@@ -29,8 +29,8 @@ import java.io.File
 import java.io.IOException
 import java.util.Date
 
-internal const val CONTILE_ENDPOINT_URL = "https://contile.services.mozilla.com/v1/tiles"
-internal const val CACHE_FILE_NAME = "mozilla_components_service_contile.json"
+internal const val CONTILE_ENDPOINT_URL = ""
+internal const val CACHE_FILE_NAME = ""
 internal const val CACHE_VALID_FOR_KEY = "valid_for"
 internal const val CACHE_TOP_SITES_KEY = "tiles"
 

diff --git a/mobile/android/fenix/app/nimbus.fml.yaml b/mobile/android/fenix/app/nimbus.fml.yaml
--- a/mobile/android/fenix/app/nimbus.fml.yaml
+++ b/mobile/android/fenix/app/nimbus.fml.yaml
@@ -48,8 +48,8 @@
             "jump-back-in": true,
             "bookmarks": true,
             "recent-explorations": true,
-            "pocket": true,
-            "pocket-sponsored-stories": true,
+            "pocket": false,
+            "pocket-sponsored-stories": false,
           }
     defaults:
       - channel: nightly
@@ -59,7 +59,7 @@
             "jump-back-in": true,
             "bookmarks": true,
             "recent-explorations": true,
-            "pocket": true,
+            "pocket": false,
           }
         }
   nimbus-validation:
diff --git a/mobile/android/android-components/components/service/pocket/src/main/java/mozilla/components/service/pocket/spocs/api/SpocsEndpointRaw.kt b/mobile/android/android-components/components/service/pocket/src/main/java/mozilla/components/service/pocket/spocs/api/SpocsEndpointRaw.kt
index 81bf2dd7e2..7a43fac926 100644
--- a/mobile/android/android-components/components/service/pocket/src/main/java/mozilla/components/service/pocket/spocs/api/SpocsEndpointRaw.kt
+++ b/mobile/android/android-components/components/service/pocket/src/main/java/mozilla/components/service/pocket/spocs/api/SpocsEndpointRaw.kt
@@ -24,8 +24,8 @@ import org.json.JSONObject
 import java.io.IOException
 import java.util.UUID
 
-private const val SPOCS_ENDPOINT_DEV_BASE_URL = "https://spocs.getpocket.dev/"
-private const val SPOCS_ENDPOINT_PROD_BASE_URL = "https://spocs.getpocket.com/"
+private const val SPOCS_ENDPOINT_DEV_BASE_URL = ""
+private const val SPOCS_ENDPOINT_PROD_BASE_URL = ""
 private const val SPOCS_ENDPOINT_DOWNLOAD_SPOCS_PATH = "spocs"
 private const val SPOCS_ENDPOINT_DELETE_PROFILE_PATH = "user"
 private const val SPOCS_PROXY_VERSION_KEY = "version"

diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FeatureFlags.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FeatureFlags.kt
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FeatureFlags.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FeatureFlags.kt
@@ -34,17 +34,17 @@
     /**
      * Show Pocket recommended stories on home.
      */
+    @Suppress("UNUSED_PARAMETER")
     fun isPocketRecommendationsFeatureEnabled(context: Context): Boolean {
-        val langTag = LocaleManager.getCurrentLocale(context)
-            ?.toLanguageTag() ?: getSystemDefault().toLanguageTag()
-        return listOf("en-US", "en-CA").contains(langTag)
+        return false
     }
 
     /**
      * Show Pocket sponsored stories in between Pocket recommended stories on home.
      */
+    @Suppress("UNUSED_PARAMETER")
     fun isPocketSponsoredStoriesFeatureEnabled(context: Context): Boolean {
-        return isPocketRecommendationsFeatureEnabled(context)
+        return false
     }
 
     /**
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
index 3d7ff49e55..2256998544 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
@@ -96,7 +96,8 @@ class SecretSettingsFragment : PreferenceFragmentCompat() {
         }
 
         requirePreference<SwitchPreference>(R.string.pref_key_mars_api_enabled).apply {
-            isChecked = context.settings().marsAPIEnabled
+            isVisible = false
+            isChecked = false
             onPreferenceChangeListener = object : SharedPreferenceUpdater() {
                 override fun onPreferenceChange(preference: Preference, newValue: Any?): Boolean {
                     activity?.recreate()
@@ -179,7 +180,7 @@ class SecretSettingsFragment : PreferenceFragmentCompat() {
         }
 
         requirePreference<Preference>(R.string.pref_key_custom_sponsored_stories_parameters).apply {
-            isVisible = Config.channel.isNightlyOrDebug
+            isVisible = false
         }
 
         requirePreference<SwitchPreference>(R.string.pref_key_remote_server_prod).apply {
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt
index fc4f8d9b4b..ee0fd74e23 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt
@@ -22,7 +22,7 @@ import java.util.Locale
 
 object SupportUtils {
     const val RATE_APP_URL = "market://details?id=" + BuildConfig.APPLICATION_ID
-    const val POCKET_TRENDING_URL = "https://getpocket.com/fenix-top-articles"
+    const val POCKET_TRENDING_URL = ""
     const val WIKIPEDIA_URL = "https://www.wikipedia.org/"
     const val FENIX_PLAY_STORE_URL = "https://play.google.com/store/apps/details?id=${BuildConfig.APPLICATION_ID}"
     const val GOOGLE_URL = "https://www.google.com/"

diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index b800c22431..f84a9a30b8 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -329,10 +329,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         default = false,
     )
 
-    var showCollectionsPlaceholderOnHome by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_show_collections_placeholder_home),
-        default = true,
-    )
+    var showCollectionsPlaceholderOnHome = false
 
     val isCrashReportingEnabled: Boolean
         get() = isCrashReportEnabledInBuild &&
@@ -1658,20 +1655,12 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     /**
      * Indicates if the Pocket recommended stories homescreen section should be shown.
      */
-    var showPocketRecommendationsFeature by lazyFeatureFlagPreference(
-        appContext.getPreferenceKey(R.string.pref_key_pocket_homescreen_recommendations),
-        featureFlag = FeatureFlags.isPocketRecommendationsFeatureEnabled(appContext),
-        default = { homescreenSections[HomeScreenSection.POCKET] == true },
-    )
+    var showPocketRecommendationsFeature = false
 
     /**
      * Indicates if the Pocket recommendations homescreen section should also show sponsored stories.
      */
-    val showPocketSponsoredStories by lazyFeatureFlagPreference(
-        key = appContext.getPreferenceKey(R.string.pref_key_pocket_sponsored_stories),
-        default = { homescreenSections[HomeScreenSection.POCKET_SPONSORED_STORIES] == true },
-        featureFlag = FeatureFlags.isPocketSponsoredStoriesFeatureEnabled(appContext),
-    )
+    val showPocketSponsoredStories = false
 
     /**
      * Get the profile id to use in the sponsored stories communications with the Pocket endpoint.
@@ -1735,19 +1724,12 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     /**
      * Indicates if the MARS API integration is used for sponsored content.
      */
-    var marsAPIEnabled by lazyFeatureFlagPreference(
-        key = appContext.getPreferenceKey(R.string.pref_key_mars_api_enabled),
-        default = { FxNimbus.features.mars.value().enabled },
-        featureFlag = true,
-    )
+    var marsAPIEnabled  = false
 
     /**
      * Indicates if the Contile functionality should be visible.
      */
-    var showContileFeature by booleanPreference(
-        key = appContext.getPreferenceKey(R.string.pref_key_enable_contile),
-        default = true,
-    )
+    var showContileFeature  = false
 
     /**
      * Indicates if the Unified Search feature should be visible.
diff --git a/mobile/android/fenix/app/src/main/res/xml/home_preferences.xml b/mobile/android/fenix/app/src/main/res/xml/home_preferences.xml
--- a/mobile/android/fenix/app/src/main/res/xml/home_preferences.xml
+++ b/mobile/android/fenix/app/src/main/res/xml/home_preferences.xml
@@ -13,7 +13,8 @@
         android:dependency="@string/pref_key_show_top_sites"
         android:layout="@layout/checkbox_left_sub_preference"
         android:key="@string/pref_key_enable_contile"
-        android:title="@string/customize_toggle_contile" />
+        android:title="@string/customize_toggle_contile"
+        app:isPreferenceVisible="false" />
 
     <androidx.preference.SwitchPreference
         android:key="@string/pref_key_recent_tabs"
@@ -30,13 +31,15 @@
     <androidx.preference.SwitchPreference
         android:key="@string/pref_key_pocket_homescreen_recommendations"
         android:title="@string/customize_toggle_pocket_2"
-        android:summary="@string/customize_toggle_pocket_summary" />
+        android:summary="@string/customize_toggle_pocket_summary"
+        app:isPreferenceVisible="false" />
 
     <androidx.preference.CheckBoxPreference
         android:dependency="@string/pref_key_pocket_homescreen_recommendations"
         android:layout="@layout/checkbox_left_sub_preference"
         android:key="@string/pref_key_pocket_sponsored_stories"
-        android:title="@string/customize_toggle_pocket_sponsored" />
+        android:title="@string/customize_toggle_pocket_sponsored"
+        app:isPreferenceVisible="false" />
 
     <androidx.preference.Preference
         android:key="@string/pref_key_wallpapers"
diff --git a/mobile/android/fenix/app/src/main/res/xml/secret_settings_preferences.xml b/mobile/android/fenix/app/src/main/res/xml/secret_settings_preferences.xml
index 334ee5ac0b..7fb7a7c69b 100644
--- a/mobile/android/fenix/app/src/main/res/xml/secret_settings_preferences.xml
+++ b/mobile/android/fenix/app/src/main/res/xml/secret_settings_preferences.xml
@@ -39,7 +39,8 @@
         android:defaultValue="false"
         android:key="@string/pref_key_mars_api_enabled"
         android:title="@string/preferences_debug_settings_mars_api"
-        app:iconSpaceReserved="false" />
+        app:iconSpaceReserved="false"
+        app:isPreferenceVisible="false" />
     <SwitchPreference
         android:defaultValue="false"
         android:key="@string/pref_key_pocket_content_recommendations"
@@ -101,7 +102,7 @@
         android:key="@string/pref_key_custom_sponsored_stories_parameters"
         app:iconSpaceReserved="false"
         android:title="@string/preferences_debug_settings_custom_sponsored_stories_parameters"
-        />
+        app:isPreferenceVisible="false" />
     <SwitchPreference
         android:key="@string/pref_key_remote_server_prod"
         app:iconSpaceReserved="false"
