diff --git a/mobile/android/android-components/components/service/location/src/main/java/mozilla/components/service/location/MozillaLocationService.kt b/mobile/android/android-components/components/service/location/src/main/java/mozilla/components/service/location/MozillaLocationService.kt
index d3f59c7a97..c7cef63717 100644
--- a/mobile/android/android-components/components/service/location/src/main/java/mozilla/components/service/location/MozillaLocationService.kt
+++ b/mobile/android/android-components/components/service/location/src/main/java/mozilla/components/service/location/MozillaLocationService.kt
@@ -94,7 +94,7 @@ class MozillaLocationService(
      * Check to see if the cache is still valid.
      */
     private fun isCacheValid(): Boolean {
-        return currentTime() < context.cachedAt() + CACHE_LIFETIME_IN_MS
+        return true
     }
 
     private fun Context.cacheRegion(region: LocationService.Region) {
