diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt
index baadbd7873..87a4d45b15 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt
@@ -110,7 +110,7 @@ class AwesomeBarView(
                 components.core.icons,
                 engineForSpeculativeConnects,
                 showEditSuggestion = false,
-                suggestionsHeader = activity.getString(R.string.firefox_suggest_header),
+//                suggestionsHeader = activity.getString(R.string.firefox_suggest_header),
             )
 
         defaultCombinedHistoryProvider =
@@ -122,7 +122,7 @@ class AwesomeBarView(
                 engine = engineForSpeculativeConnects,
                 maxNumberOfSuggestions = METADATA_SUGGESTION_LIMIT,
                 showEditSuggestion = false,
-                suggestionsHeader = activity.getString(R.string.firefox_suggest_header),
+//                suggestionsHeader = activity.getString(R.string.firefox_suggest_header),
             )
 
         val searchBitmap = getDrawable(activity, R.drawable.ic_search)!!.apply {
@@ -419,7 +419,7 @@ class AwesomeBarView(
                     engine = engineForSpeculativeConnects,
                     maxNumberOfSuggestions = METADATA_SUGGESTION_LIMIT,
                     showEditSuggestion = false,
-                    suggestionsHeader = activity.getString(R.string.firefox_suggest_header),
+//                    suggestionsHeader = activity.getString(R.string.firefox_suggest_header),
                     resultsUriFilter = filter::shouldIncludeUri,
                 )
             } else {
@@ -434,7 +434,7 @@ class AwesomeBarView(
                     engine = engineForSpeculativeConnects,
                     maxNumberOfSuggestions = METADATA_SUGGESTION_LIMIT,
                     showEditSuggestion = false,
-                    suggestionsHeader = activity.getString(R.string.firefox_suggest_header),
+//                    suggestionsHeader = activity.getString(R.string.firefox_suggest_header),
                     resultsUriFilter = filter::shouldIncludeUri,
                 )
             } else {
@@ -556,7 +556,7 @@ class AwesomeBarView(
                 getDrawable(activity, R.drawable.ic_search_results_device_mobile),
                 getDrawable(activity, R.drawable.ic_search_results_device_tablet),
             ),
-            suggestionsHeader = activity.getString(R.string.firefox_suggest_header),
+//            suggestionsHeader = activity.getString(R.string.firefox_suggest_header),
             resultsUrlFilter = filter?.let { it::shouldIncludeUrl },
         )
     }
@@ -579,7 +579,7 @@ class AwesomeBarView(
             components.core.icons,
             getDrawable(activity, R.drawable.ic_search_results_tab),
             excludeSelectedSession = !fromHomeFragment,
-            suggestionsHeader = activity.getString(R.string.firefox_suggest_header),
+//            suggestionsHeader = activity.getString(R.string.firefox_suggest_header),
             resultsUriFilter = filter?.let { it::shouldIncludeUri },
         )
     }
@@ -602,7 +602,7 @@ class AwesomeBarView(
             indicatorIcon = getDrawable(activity, R.drawable.ic_search_results_bookmarks),
             engine = engineForSpeculativeConnects,
             showEditSuggestion = false,
-            suggestionsHeader = activity.getString(R.string.firefox_suggest_header),
+//            suggestionsHeader = activity.getString(R.string.firefox_suggest_header),
             resultsUriFilter = filter?.let { it::shouldIncludeUri },
         )
     }
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt
index ce4b42ef4a..d5f03c9e2a 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt
@@ -39,7 +39,7 @@ class SearchEngineFragment : PreferenceFragmentCompat() {
         )
 
         requirePreference<SwitchPreference>(R.string.pref_key_show_sponsored_suggestions).apply {
-            isVisible = context.settings().enableFxSuggest
+            isVisible = false
         }
         requirePreference<SwitchPreference>(R.string.pref_key_show_nonsponsored_suggestions).apply {
             isVisible = context.settings().enableFxSuggest
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 774d43efab..16acd78c19 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -2146,7 +2146,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var enableFxSuggest by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_enable_fxsuggest),
-        default = { FxNimbus.features.fxSuggest.value().enabled },
+        default = { false },
         featureFlag = FeatureFlags.FX_SUGGEST,
     )
 
@@ -2154,7 +2154,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      * Indicates if boosting AMP/wiki suggestions is enabled.
      */
     val boostAmpWikiSuggestions: Boolean
-        get() = FxNimbus.features.fxSuggest.value().boostAmpWiki
+        get() = true
 
     /**
      * Indicates first time engaging with signup
@@ -2171,7 +2171,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     var showSponsoredSuggestions by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_show_sponsored_suggestions),
         default = { enableFxSuggest },
-        featureFlag = FeatureFlags.FX_SUGGEST,
+        featureFlag = false,
     )
 
     /**
@@ -2181,7 +2181,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var showNonSponsoredSuggestions by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_show_nonsponsored_suggestions),
-        default = { enableFxSuggest },
+        default = { false },
         featureFlag = FeatureFlags.FX_SUGGEST,
     )
 
