diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt
index 3ab8ea1acd..992865d78f 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt
@@ -497,12 +497,7 @@ class Core(
     val bookmarksStorage: PlacesBookmarksStorage get() = lazyBookmarksStorage.value
     val passwordsStorage: SyncableLoginsStorage get() = lazyPasswordsStorage.value
     val autofillStorage: AutofillCreditCardsAddressesStorage get() = lazyAutofillStorage.value
-    val domainsAutocompleteProvider: BaseDomainAutocompleteProvider? get() =
-        if (FxNimbus.features.suggestShippedDomains.value().enabled) {
-            lazyDomainsAutocompleteProvider.value
-        } else {
-            null
-        }
+    val domainsAutocompleteProvider: BaseDomainAutocompleteProvider? get() = null
     val sessionAutocompleteProvider: SessionAutocompleteProvider get() = lazySessionAutocompleteProvider.value
 
     val tabCollectionStorage by lazyMonitored {
