diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
index 3d7ff49e55..43fb1e7181 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
@@ -106,7 +106,8 @@ class SecretSettingsFragment : PreferenceFragmentCompat() {
         }
 
         requirePreference<SwitchPreference>(R.string.pref_key_pocket_content_recommendations).apply {
-            isChecked = context.settings().showContentRecommendations
+            isVisible = false
+            isChecked = false
             onPreferenceChangeListener = object : SharedPreferenceUpdater() {
                 override fun onPreferenceChange(preference: Preference, newValue: Any?): Boolean {
                     activity?.recreate()
@@ -122,14 +123,14 @@ class SecretSettingsFragment : PreferenceFragmentCompat() {
         }
 
         requirePreference<SwitchPreference>(R.string.pref_key_enable_trending_searches).apply {
-            isVisible = Config.channel.isNightlyOrDebug
-            isChecked = context.settings().isTrendingSearchesVisible
+            isVisible = false
+            isChecked = false
             onPreferenceChangeListener = SharedPreferenceUpdater()
         }
 
         requirePreference<SwitchPreference>(R.string.pref_key_enable_fxsuggest).apply {
-            isVisible = FeatureFlags.FX_SUGGEST
-            isChecked = context.settings().enableFxSuggest
+            isVisible = false
+            isChecked = false
             onPreferenceChangeListener = object : Preference.OnPreferenceChangeListener {
                 override fun onPreferenceChange(preference: Preference, newValue: Any?): Boolean {
                     val newBooleanValue = newValue as? Boolean ?: return false
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index b800c22431..4e7e477336 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -1094,10 +1094,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      * Indicates if the user have enabled trending search in search suggestions.
      */
     @VisibleForTesting
-    internal var trendingSearchSuggestionsEnabled by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_show_trending_search_suggestions),
-        default = true,
-    )
+    internal var trendingSearchSuggestionsEnabled = false
 
     /**
      * Returns true if trending searches should be shown to the user.
@@ -1694,11 +1691,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     /**
      * Indicates if Merino content recommendations should be shown.
      */
-    var showContentRecommendations by lazyFeatureFlagPreference(
-        key = appContext.getPreferenceKey(R.string.pref_key_pocket_content_recommendations),
-        default = { FxNimbus.features.merinoRecommendations.value().enabled },
-        featureFlag = true,
-    )
+    var showContentRecommendations = false
 
     /**
      *  Whether or not to display the Pocket sponsored stories parameter secret settings.
@@ -1948,11 +1941,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     /**
      * Indicates if Trending Searches is enabled.
      */
-    var isTrendingSearchesVisible by lazyFeatureFlagPreference(
-        key = appContext.getPreferenceKey(R.string.pref_key_enable_trending_searches),
-        default = { FxNimbus.features.trendingSearches.value().enabled },
-        featureFlag = true,
-    )
+    var isTrendingSearchesVisible = false
 
     /**
      * Adjust Activated User sent
@@ -2009,17 +1998,12 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     /**
      * Indicates if Firefox Suggest is enabled.
      */
-    var enableFxSuggest by lazyFeatureFlagPreference(
-        key = appContext.getPreferenceKey(R.string.pref_key_enable_fxsuggest),
-        default = { FxNimbus.features.fxSuggest.value().enabled },
-        featureFlag = FeatureFlags.FX_SUGGEST,
-    )
+    var enableFxSuggest = false
 
     /**
      * Indicates if boosting AMP/wiki suggestions is enabled.
      */
-    val boostAmpWikiSuggestions: Boolean
-        get() = FxNimbus.features.fxSuggest.value().boostAmpWiki
+    val boostAmpWikiSuggestions: Boolean = false
 
     /**
      * Indicates first time engaging with signup
@@ -2033,22 +2017,14 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      * Indicates if the user has chosen to show sponsored search suggestions in the awesomebar.
      * The default value is computed lazily, and based on whether Firefox Suggest is enabled.
      */
-    var showSponsoredSuggestions by lazyFeatureFlagPreference(
-        key = appContext.getPreferenceKey(R.string.pref_key_show_sponsored_suggestions),
-        default = { enableFxSuggest },
-        featureFlag = FeatureFlags.FX_SUGGEST,
-    )
+    var showSponsoredSuggestions = false
 
     /**
      * Indicates if the user has chosen to show search suggestions for web content in the
      * awesomebar. The default value is computed lazily, and based on whether Firefox Suggest
      * is enabled.
      */
-    var showNonSponsoredSuggestions by lazyFeatureFlagPreference(
-        key = appContext.getPreferenceKey(R.string.pref_key_show_nonsponsored_suggestions),
-        default = { enableFxSuggest },
-        featureFlag = FeatureFlags.FX_SUGGEST,
-    )
+    var showNonSponsoredSuggestions = false
 
     /**
      * Indicates that the user does not want warned of a translations
diff --git a/mobile/android/fenix/app/src/main/res/xml/search_settings_preferences.xml b/mobile/android/fenix/app/src/main/res/xml/search_settings_preferences.xml
index da8132091b..107ccbc235 100644
--- a/mobile/android/fenix/app/src/main/res/xml/search_settings_preferences.xml
+++ b/mobile/android/fenix/app/src/main/res/xml/search_settings_preferences.xml
@@ -38,12 +38,13 @@
             android:title="@string/preferences_show_search_suggestions_in_private"
             app:iconSpaceReserved="false" />
         <CheckBoxPreference
-            android:defaultValue="true"
+            android:defaultValue="false"
             android:dependency="@string/pref_key_show_search_suggestions"
             android:key="@string/pref_key_show_trending_search_suggestions"
             android:layout="@layout/checkbox_left_preference"
             android:title="@string/preferences_show_trending_search_suggestions"
-            app:iconSpaceReserved="false" />
+            app:iconSpaceReserved="false"
+            app:isPreferenceVisible="false" />
     </PreferenceCategory>
 
     <PreferenceCategory
@@ -70,16 +71,19 @@
             app:iconSpaceReserved="false"
             android:key="@string/pref_key_show_nonsponsored_suggestions"
             android:title="@string/preferences_show_nonsponsored_suggestions"
-            android:summary="@string/preferences_show_nonsponsored_suggestions_summary" />
+            android:summary="@string/preferences_show_nonsponsored_suggestions_summary"
+            app:isPreferenceVisible="false" />
         <SwitchPreference
             app:iconSpaceReserved="false"
             android:key="@string/pref_key_show_sponsored_suggestions"
             android:title="@string/preferences_show_sponsored_suggestions"
-            android:summary="@string/preferences_show_sponsored_suggestions_summary" />
+            android:summary="@string/preferences_show_sponsored_suggestions_summary"
+            app:isPreferenceVisible="false" />
         <Preference
             app:iconSpaceReserved="false"
             android:key="@string/pref_key_learn_about_fx_suggest"
-            android:title="@string/preference_search_learn_about_fx_suggest" />
+            android:title="@string/preference_search_learn_about_fx_suggest"
+            app:isPreferenceVisible="false" />
     </PreferenceCategory>
 
     <PreferenceCategory
diff --git a/mobile/android/fenix/app/src/main/res/xml/secret_settings_preferences.xml b/mobile/android/fenix/app/src/main/res/xml/secret_settings_preferences.xml
index 334ee5ac0b..884faa12a8 100644
--- a/mobile/android/fenix/app/src/main/res/xml/secret_settings_preferences.xml
+++ b/mobile/android/fenix/app/src/main/res/xml/secret_settings_preferences.xml
@@ -44,7 +44,8 @@
         android:defaultValue="false"
         android:key="@string/pref_key_pocket_content_recommendations"
         android:title="@string/preferences_debug_settings_merino_content_recommendations"
-        app:iconSpaceReserved="false" />
+        app:iconSpaceReserved="false"
+        app:isPreferenceVisible="false" />
     <SwitchPreference
         android:defaultValue="false"
         android:key="@string/pref_key_enable_menu_redesign"
@@ -59,7 +60,8 @@
         android:defaultValue="false"
         android:key="@string/pref_key_enable_fxsuggest"
         android:title="@string/preferences_debug_settings_fxsuggest"
-        app:iconSpaceReserved="false" />
+        app:iconSpaceReserved="false"
+        app:isPreferenceVisible="false" />
     <SwitchPreference
         android:key="@string/pref_key_should_enable_felt_privacy"
         android:title="@string/preferences_debug_felt_privacy"
@@ -71,7 +73,8 @@
     <SwitchPreference
         android:key="@string/pref_key_enable_trending_searches"
         android:title="@string/preferences_debug_settings_trending_searches"
-        app:iconSpaceReserved="false" />
+        app:iconSpaceReserved="false"
+        app:isPreferenceVisible="false" />
     <SwitchPreference
         android:defaultValue="false"
         android:key="@string/pref_key_enable_debug_drawer"
