diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/AccessibilityFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/AccessibilityFragment.kt
index 1122dc0b0f..456f387812 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/AccessibilityFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/AccessibilityFragment.kt
@@ -23,6 +23,20 @@ class AccessibilityFragment : PreferenceFragmentCompat() {
         super.onResume()
         showToolbar(getString(R.string.preferences_accessibility))
 
+        val accessibilityEnabledPreference = requirePreference<SwitchPreference>(
+            R.string.pref_key_accessibility_enabled,
+        )
+
+        accessibilityEnabledPreference.setOnPreferenceChangeListener<Boolean> { preference, accessibilityEnabled ->
+            val settings = preference.context.settings()
+            val components = preference.context.components
+
+            settings.accessibilityEnabled = accessibilityEnabled
+            components.core.engine.settings.accessibilityEnabled = accessibilityEnabled
+
+            true
+        }
+
         val forceZoomPreference = requirePreference<SwitchPreference>(
             R.string.pref_key_accessibility_force_enable_zoom,
         )
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 7f19ba4304..e321174ae1 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -1076,7 +1076,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
             accessibilityManager?.getEnabledAccessibilityServiceList(0)?.let { activeServices ->
                 for (service in activeServices) {
                     if (service.capabilities.and(CAPABILITY_CAN_PERFORM_GESTURES) == 1) {
-                        return true
+                        return accessibilityEnabled
                     }
                 }
             }
@@ -1088,7 +1088,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         get() {
             val accessibilityManager =
                 appContext.getSystemService(Context.ACCESSIBILITY_SERVICE) as? AccessibilityManager
-            return accessibilityManager?.isTouchExplorationEnabled ?: false
+            return accessibilityEnabled && (accessibilityManager?.isTouchExplorationEnabled ?: false)
         }
 
     val accessibilityServicesEnabled: Boolean
diff --git a/mobile/android/fenix/app/src/main/res/xml/accessibility_preferences.xml b/mobile/android/fenix/app/src/main/res/xml/accessibility_preferences.xml
index a810f76208..f9381cbc86 100644
--- a/mobile/android/fenix/app/src/main/res/xml/accessibility_preferences.xml
+++ b/mobile/android/fenix/app/src/main/res/xml/accessibility_preferences.xml
@@ -4,6 +4,11 @@
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
 <PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto">
+    <SwitchPreference
+        android:defaultValue="false"
+        android:key="@string/pref_key_accessibility_enabled"
+        android:summary="@string/preference_accessibility_enabled_summary"
+        android:title="@string/preference_accessibility_enabled" />
     <SwitchPreference
         android:defaultValue="true"
         android:key="@string/pref_key_accessibility_auto_size"
