diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 8913d9f230..50bf05c440 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -1803,10 +1803,10 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     /**
      * Indicates if the Contile functionality should be visible.
      */
-    var showContileFeature by booleanPreference(
-        key = appContext.getPreferenceKey(R.string.pref_key_enable_contile),
-        default = true,
-    )
+    var showContileFeature = false
+//        key = appContext.getPreferenceKey(R.string.pref_key_enable_contile),
+//        default = true,
+//    )
 
     /**
      * Indicates if the Unified Search feature should be visible.
diff --git a/mobile/android/fenix/app/src/main/res/xml/home_preferences.xml b/mobile/android/fenix/app/src/main/res/xml/home_preferences.xml
index 4fd1f8945a..d8586426ab 100644
--- a/mobile/android/fenix/app/src/main/res/xml/home_preferences.xml
+++ b/mobile/android/fenix/app/src/main/res/xml/home_preferences.xml
@@ -13,7 +13,8 @@
         android:dependency="@string/pref_key_show_top_sites"
         android:layout="@layout/checkbox_left_sub_preference"
         android:key="@string/pref_key_enable_contile"
-        android:title="@string/customize_toggle_contile" />
+        android:title="@string/customize_toggle_contile"
+        app:isPreferenceVisible="false" />
 
     <androidx.preference.SwitchPreference
         android:key="@string/pref_key_recent_tabs"
