From 2e80981c352c7b2fa6be97c10d95e03f667da0c4 Mon Sep 17 00:00:00 2001
From: Akash Yadav <itsaky01@gmail.com>
Date: Mon, 28 Apr 2025 10:43:51 +0530
Subject: [PATCH] fix(patches): update 'disable-csp-reporting.patch' for
 'FIREFOX_138_0_BUILD1'

Signed-off-by: Akash Yadav <itsaky01@gmail.com>
---
 dom/security/nsCSPContext.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/dom/security/nsCSPContext.cpp b/dom/security/nsCSPContext.cpp
index 6e867f1fe6..a997355e31 100644
--- a/dom/security/nsCSPContext.cpp
+++ b/dom/security/nsCSPContext.cpp
@@ -1209,7 +1209,6 @@ nsresult nsCSPContext::SendReports(
 
   //[Deprecated] CSP Level 2 Reporting
   if (!reportURIs.IsEmpty()) {
-    return SendReportsToURIs(reportURIs, aViolationEventInit);
   }
 
   return NS_OK;
-- 
2.49.0

