diff --git a/toolkit/components/doh/DoHHeuristics.sys.mjs b/toolkit/components/doh/DoHHeuristics.sys.mjs
index 812af19f00..2bb4337c40 100644
--- a/toolkit/components/doh/DoHHeuristics.sys.mjs
+++ b/toolkit/components/doh/DoHHeuristics.sys.mjs
@@ -29,7 +29,7 @@ ChromeUtils.defineESModuleGetters(lazy, {
   DoHConfigController: "moz-src:///toolkit/components/doh/DoHConfig.sys.mjs",
 });
 
-const GLOBAL_CANARY = "use-application-dns.net.";
+const GLOBAL_CANARY = "";
 
 const NXDOMAIN_ERR = "NS_ERROR_UNKNOWN_HOST";
 
@@ -390,7 +390,7 @@ async function safeSearch() {
 }
 
 async function zscalerCanary() {
-  const ZSCALER_CANARY = "sitereview.zscaler.com.";
+  const ZSCALER_CANARY = "";
 
   let { addresses } = await dnsLookup(ZSCALER_CANARY);
   for (let address of addresses) {
