diff --git a/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/ContentBlocking.java b/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/ContentBlocking.java
index b761e1126e..70b15bfa03 100644
--- a/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/ContentBlocking.java
+++ b/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/ContentBlocking.java
@@ -364,9 +364,9 @@ public class ContentBlocking {
             ContentBlocking.catToPref(AntiTracking.NONE, AntiTracking.STP, STP));
 
     /* package */ final Pref<Boolean> mSbMalware =
-        new Pref<Boolean>("browser.safebrowsing.malware.enabled", true);
+        new Pref<Boolean>("browser.safebrowsing.malware.enabled", false);
     /* package */ final Pref<Boolean> mSbPhishing =
-        new Pref<Boolean>("browser.safebrowsing.phishing.enabled", true);
+        new Pref<Boolean>("browser.safebrowsing.phishing.enabled", false);
     /* package */ final Pref<Integer> mCookieBehavior =
         new Pref<Integer>(
             "network.cookie.cookieBehavior", CookieBehavior.ACCEPT_FIRST_PARTY_AND_ISOLATE_OTHERS);
