diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/browser/desktopmode/DefaultDesktopModeFeatureFlag.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/browser/desktopmode/DefaultDesktopModeFeatureFlag.kt
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/browser/desktopmode/DefaultDesktopModeFeatureFlag.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/browser/desktopmode/DefaultDesktopModeFeatureFlag.kt
@@ -11,7 +11,7 @@
  */
 class DefaultDesktopModeFeatureFlagImpl : DefaultDesktopModeFeatureFlag {
 
-    override fun isDesktopModeEnabled(): Boolean = Config.channel.isDebug
+    override fun isDesktopModeEnabled(): Boolean = true
 }
 
 /**
