diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
index b5cd2868a6..9e116a0a70 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
@@ -439,9 +439,9 @@ open class DefaultToolbarMenu(
                 bookmarksItem,
                 historyItem,
                 downloadsItem,
-                passwordsItem,
+                if (context.settings().shouldPromptToSaveLogins) passwordsItem else null,
                 extensionsItem,
-                syncMenuItem(),
+                if (context.settings().signedInFxaAccount) syncMenuItem() else null,
                 BrowserMenuDivider(),
                 findInPageItem,
                 translationsItem.apply { visible = ::shouldShowTranslations },
@@ -453,9 +453,9 @@ open class DefaultToolbarMenu(
                 BrowserMenuDivider(),
                 addToHomeScreenItem.apply { visible = ::canAddToHomescreen },
                 addAppToHomeScreenItem.apply { visible = ::canAddAppToHomescreen },
-                if (shouldShowTopSites) addRemoveTopSitesItem else null,
-                saveToCollectionItem,
-                if (FxNimbus.features.print.value().browserPrintEnabled &&
+                if (shouldShowTopSites && !context.settings().alwaysUsePrivateBrowsing) addRemoveTopSitesItem else null,
+                if (context.settings().collections && !context.settings().alwaysUsePrivateBrowsing) saveToCollectionItem else null,
+                if (context.settings().printEnabled &&
                     !context.isAndroidAutomotiveAvailable()
                 ) {
                     printPageItem
@@ -464,7 +464,7 @@ open class DefaultToolbarMenu(
                 },
                 BrowserMenuDivider(),
                 settingsItem,
-                if (shouldDeleteDataOnQuit) deleteDataOnQuit else null,
+                deleteDataOnQuit,
                 if (shouldUseBottomToolbar) BrowserMenuDivider() else null,
                 if (shouldUseBottomToolbar) menuToolbar else null,
             )
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/home/HomeMenu.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/home/HomeMenu.kt
index 849660153d..60a57c6b77 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/home/HomeMenu.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/home/HomeMenu.kt
@@ -206,17 +206,17 @@ class HomeMenu(
             bookmarksItem,
             historyItem,
             downloadsItem,
-            passwordsItem,
+            if (settings.shouldPromptToSaveLogins) passwordsItem else null,
             extensionsItem,
-            syncSignInMenuItem,
-            accountAuthItem,
+            if (settings.signedInFxaAccount) syncSignInMenuItem else null,
+            if (settings.signedInFxaAccount) accountAuthItem else null,
             BrowserMenuDivider(),
             BrowserMenuDivider(),
             whatsNewItem,
             helpItem,
-            customizeHomeItem,
+            if (!settings.alwaysUsePrivateBrowsing) customizeHomeItem else null,
             settingsItem,
-            if (settings.shouldDeleteBrowsingDataOnQuit) quitItem else null,
+            quitItem,
         ).also { items ->
             items.getHighlight()?.let { onHighlightPresent(it) }
         }
