diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/account/AccountUiView.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/account/AccountUiView.kt
index da8fa9afaa7f..b9986c14447c 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/account/AccountUiView.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/account/AccountUiView.kt
@@ -86,7 +86,7 @@ class AccountUiView(
             preferenceSignIn.isVisible = true
             preferenceFirefoxAccount.isVisible = false
             preferenceFirefoxAccountAuthError.isVisible = false
-            accountPreferenceCategory.isVisible = false
+            accountPreferenceCategory.isVisible = true
         }
     }
 
