diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 5c3035c5b371f..066d464175888 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -2716,9 +2716,9 @@ class Settings(
      * - 3: Maximum protection
      * - 5: DoH is disabled
      */
-    private var trrMode by intPreference(
+    var trrMode by intPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_doh_settings_mode),
-        default = DOH_SETTINGS_DEFAULT,
+        default = DOH_SETTINGS_MAX,
     )
 
     /**
@@ -2736,7 +2736,7 @@ class Settings(
      */
     val dohDefaultProviderUrl by stringPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_doh_default_provider_uri),
-        default = CLOUDFLARE_URI,
+        default = "https://dns.quad9.net/dns-query",
     )
 
     /**
