diff --git a/mobile/android/fenix/app/build.gradle b/mobile/android/fenix/app/build.gradle
index 1b27020a20..db3d3fd09c 100644
--- a/mobile/android/fenix/app/build.gradle
+++ b/mobile/android/fenix/app/build.gradle
@@ -37,7 +37,7 @@ apply plugin: 'jacoco'
 apply plugin: 'androidx.navigation.safeargs.kotlin'
 apply plugin: 'kotlinx-serialization'
 
-if (gradle.mozconfig.substs.MOZILLA_OFFICIAL) {
+if (false) {
     apply plugin: 'com.google.android.gms.oss-licenses-plugin'
 }
 
