diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index c7db941b0f7e..254fb016d831 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -364,7 +364,7 @@ class Settings(
     )
 
     val canShowCfr: Boolean
-        get() = (System.currentTimeMillis() - lastCfrShownTimeInMillis) > THREE_DAYS_MS
+        get() = false
 
     var forceEnableZoom by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_accessibility_force_enable_zoom),
@@ -485,7 +485,7 @@ class Settings(
      */
     var showWallpaperOnboarding by booleanPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_wallpapers_onboarding),
-        default = true,
+        default = false,
     )
 
     var openLinksInAPrivateTab by booleanPreference(
@@ -530,7 +530,7 @@ class Settings(
     var shouldShowMenuBanner by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_show_menu_banner),
         default = { FxNimbus.features.menuRedesign.value().menuBanner },
-        featureFlag = true,
+        featureFlag = false,
     )
 
     var defaultSearchEngineName by stringPreference(
@@ -940,7 +940,7 @@ class Settings(
      */
     var showFirstTimeTranslation: Boolean by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_show_first_time_translation),
-        default = true,
+        default = false,
     )
 
     /**
@@ -1126,7 +1126,7 @@ class Settings(
 
     var reEngagementNotificationShown by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_re_engagement_notification_shown),
-        default = false,
+        default = true,
     )
 
     /**
@@ -1149,7 +1149,7 @@ class Settings(
     var reEngagementNotificationEnabled by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_re_engagement_notification_enabled),
         default = { FxNimbus.features.reEngagementNotification.value().enabled },
-        featureFlag = true,
+        featureFlag = false,
     )
 
     /**
@@ -1236,7 +1236,7 @@ class Settings(
      */
     var shouldShowCookieBannersCFR by lazyFeatureFlagPreference(
         appContext.getPreferenceKey(R.string.pref_key_should_show_cookie_banners_action_popup),
-        featureFlag = true,
+        featureFlag = false,
         default = { shouldShowCookieBannerUI },
     )
 
@@ -1247,7 +1247,7 @@ class Settings(
 
     var hasShownTabSwipeCFR by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_toolbar_has_shown_tab_swipe_cfr),
-        default = false,
+        default = true,
     )
 
     val blockCookiesSelectionInCustomTrackingProtection by stringPreference(
@@ -1573,12 +1573,12 @@ class Settings(
 
     var userKnowsAboutPwas by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_user_knows_about_pwa),
-        default = false,
+        default = true,
     )
 
     var shouldShowOpenInAppBanner by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_should_show_open_in_app_banner),
-        default = true,
+        default = false,
     )
 
     val shouldShowOpenInAppCfr: Boolean
@@ -1586,17 +1586,17 @@ class Settings(
 
     var shouldShowAutoCloseTabsBanner by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_should_show_auto_close_tabs_banner),
-        default = true,
+        default = false,
     )
 
     var shouldShowLockPbmBanner by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_should_show_lock_pbm_banner),
-        true,
+        false,
     )
 
     var shouldShowInactiveTabsOnboardingPopup by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_should_show_inactive_tabs_popup),
-        default = true,
+        default = false,
     )
 
     /**
@@ -1604,7 +1604,7 @@ class Settings(
      */
     var hasInactiveTabsAutoCloseDialogBeenDismissed by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_has_inactive_tabs_auto_close_dialog_dismissed),
-        default = false,
+        default = true,
     )
 
     /**
@@ -2242,7 +2242,7 @@ class Settings(
      */
     var hasSeenBrowserToolbarCFR by booleanPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_toolbar_cfr),
-        default = isBenchmarkBuild,
+        default = true,
     )
 
     /**
@@ -2454,7 +2454,7 @@ class Settings(
      */
     var isFirstTimeEngagingWithSignup: Boolean by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_first_time_engage_with_signup),
-        default = true,
+        default = false,
     )
 
     /**
@@ -2807,7 +2807,7 @@ class Settings(
             FxNimbus.features.setupChecklist.value().enabled &&
                     canShowAddSearchWidgetPrompt(AppWidgetManager.getInstance(appContext))
         },
-        featureFlag = true,
+        featureFlag = false,
     )
 
     /**
