diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/sitepermissions/SiteSettingsFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/sitepermissions/SiteSettingsFragment.kt
index d435dae1f2..1a3b2902e2 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/sitepermissions/SiteSettingsFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/sitepermissions/SiteSettingsFragment.kt
@@ -111,6 +111,12 @@ class SiteSettingsFragment : PreferenceFragmentCompat() {
             navigateToPhoneFeature(phoneFeature)
             true
         }
+
+        with(settings) {
+            findPreference<Preference>(
+                getPreferenceKey(R.string.pref_key_browser_feature_media_key_system_access),
+            )?.isVisible = emeEnabled
+        }
     }
 
     private fun navigateToPhoneFeature(phoneFeature: PhoneFeature) {
