diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 83e0f40483b7..37677c52bbc9 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -2458,7 +2458,7 @@ class Settings(
      */
     var shouldShowCustomTabExtensions by booleanPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_should_show_custom_tab_extensions),
-        default = false,
+        default = true,
     )
 
     /**
