diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/TrackingProtectionFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/TrackingProtectionFragment.kt
index cd4ad113098b..828bc8b21559 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/TrackingProtectionFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/TrackingProtectionFragment.kt
@@ -26,6 +26,7 @@ import org.mozilla.fenix.ext.settings
 import org.mozilla.fenix.ext.showToolbar
 import org.mozilla.fenix.trackingprotection.TrackingProtectionMode
 import org.mozilla.fenix.utils.view.addToRadioGroup
+import org.ironfoxoss.ironfox.utils.FenixSettingsVisibilityManager
 
 /**
  * Displays the toggle for tracking protection, options for tracking protection policy and a button
@@ -91,6 +92,7 @@ class TrackingProtectionFragment : PreferenceFragmentCompat() {
         addToRadioGroup(radioStrict, radioStandard, radioCustom)
         updateStrictOptionsVisibility()
         updateCustomOptionsVisibility()
+        FenixSettingsVisibilityManager.TrackingProtectionFragment(requireContext(), this)
     }
 
     override fun onResume() {
