diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 0bd15b53a6..a1b1be26aa 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -1321,7 +1321,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
             accessibilityManager?.getEnabledAccessibilityServiceList(0)?.let { activeServices ->
                 for (service in activeServices) {
                     if (service.capabilities.and(CAPABILITY_CAN_PERFORM_GESTURES) == 1) {
-                        return true
+                        return accessibilityEnabled
                     }
                 }
             }
@@ -1333,7 +1333,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         get() {
             val accessibilityManager =
                 appContext.getSystemService(Context.ACCESSIBILITY_SERVICE) as? AccessibilityManager
-            return accessibilityManager?.isTouchExplorationEnabled ?: false
+            return accessibilityEnabled && (accessibilityManager?.isTouchExplorationEnabled ?: false)
         }
 
     val accessibilityServicesEnabled: Boolean
