diff --git a/mobile/android/android-components/components/compose/base/src/main/java/mozilla/components/compose/base/theme/AcornColors.kt b/mobile/android/android-components/components/compose/base/src/main/java/mozilla/components/compose/base/theme/AcornColors.kt
index 4f08bd5ea3..70ae01e667 100644
--- a/mobile/android/android-components/components/compose/base/src/main/java/mozilla/components/compose/base/theme/AcornColors.kt
+++ b/mobile/android/android-components/components/compose/base/src/main/java/mozilla/components/compose/base/theme/AcornColors.kt
@@ -808,6 +808,17 @@ val darkColorPalette = AcornColors(
     surfaceDimVariant = PhotonColors.DarkGrey80,
 )
 
+val oledColorPalette = darkColorPalette.copy(
+    layer1 = PhotonColors.Black,
+    layer2 = PhotonColors.DarkGrey90,
+    layer3 = PhotonColors.DarkGrey80,
+    layer4Start = PhotonColors.Black,
+    layer4Center = PhotonColors.Black,
+    layer4End = PhotonColors.Black,
+    actionTertiary = PhotonColors.DarkGrey80,
+    borderPrimary = PhotonColors.DarkGrey70,
+)
+
 val lightColorPalette = AcornColors(
     layer1 = PhotonColors.LightGrey10,
     layer2 = PhotonColors.White,
@@ -1001,6 +1012,8 @@ fun acornDarkColorScheme(): ColorScheme = darkColorPalette.toM3ColorScheme(
     surfaceContainerLowest = PhotonColors.DarkGrey80,
 )
 
+fun acornOledColorScheme(): ColorScheme = acornDarkColorScheme().copy()
+
 /**
  * Returns a light Material color scheme mapped from Acorn.
  */
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt
index 55ba069afd..0aae104b10 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt
@@ -616,7 +616,7 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
                     AppCompatDelegate.MODE_NIGHT_NO,
                 )
             }
-            settings.shouldUseDarkTheme -> {
+            settings.shouldUseDarkTheme || settings.shouldUseOledTheme -> {
                 AppCompatDelegate.setDefaultNightMode(
                     AppCompatDelegate.MODE_NIGHT_YES,
                 )
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt
index cb28dc38cd..30586678e6 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt
@@ -153,6 +153,12 @@ class Core(
      * configuration (see build variants).
      */
     val engine: Engine by lazyMonitored {
+        val clearColor = ContextCompat.getColor(context, if (context.settings().shouldUseOledTheme) {
+            R.color.fx_mobile_layer_color_1_oled
+        } else {
+            R.color.fx_mobile_layer_color_1
+        })
+        
         val defaultSettings = DefaultSettings(
             requestInterceptor = requestInterceptor,
             remoteDebuggingEnabled = context.settings().isRemoteDebuggingEnabled,
@@ -166,10 +172,11 @@ class Core(
             forceUserScalableContent = context.settings().forceEnableZoom,
             loginAutofillEnabled = context.settings().shouldAutofillLogins,
             enterpriseRootsEnabled = context.settings().allowThirdPartyRootCerts,
-            clearColor = ContextCompat.getColor(
-                context,
-                R.color.fx_mobile_layer_color_1,
-            ),
+//            clearColor = ContextCompat.getColor(
+//                context,
+//                R.color.fx_mobile_layer_color_1,
+//            ),
+            clearColor = clearColor,
             httpsOnlyMode = context.settings().getHttpsOnlyMode(),
             dohSettingsMode = context.settings().getDohSettingsMode(),
             dohProviderUrl = context.settings().dohProviderUrl,
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/customtabs/FennecWebAppIntentProcessor.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/customtabs/FennecWebAppIntentProcessor.kt
index c678b1928b..9c9fef76b6 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/customtabs/FennecWebAppIntentProcessor.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/customtabs/FennecWebAppIntentProcessor.kt
@@ -29,6 +29,7 @@ import org.json.JSONException
 import org.json.JSONObject
 import org.mozilla.fenix.R
 import org.mozilla.fenix.perf.runBlockingIncrement
+import org.mozilla.fenix.theme.ThemeManager
 import java.io.File
 import java.io.IOException
 
@@ -143,7 +144,7 @@ class FennecWebAppIntentProcessor(
         return CustomTabConfig(
             colorSchemes = ColorSchemes(
                 defaultColorSchemeParams = ColorSchemeParams(
-                    toolbarColor = ContextCompat.getColor(context, R.color.fx_mobile_layer_color_1),
+                    toolbarColor = ThemeManager.resolveAttributeColor(R.attr.layer1, context),
                 ),
             ),
         )
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/CustomizationFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/CustomizationFragment.kt
index 23d7f4c26a..318b685a05 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/CustomizationFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/CustomizationFragment.kt
@@ -37,6 +37,7 @@ import org.mozilla.fenix.utils.view.addToRadioGroup
 class CustomizationFragment : PreferenceFragmentCompat() {
     private lateinit var radioLightTheme: RadioButtonPreference
     private lateinit var radioDarkTheme: RadioButtonPreference
+    private lateinit var radioOledTheme: RadioButtonPreference
     private lateinit var radioAutoBatteryTheme: RadioButtonPreference
     private lateinit var radioFollowDeviceTheme: RadioButtonPreference
     private val args by navArgs<CustomizationFragmentArgs>()
@@ -59,6 +60,7 @@ class CustomizationFragment : PreferenceFragmentCompat() {
         bindFollowDeviceTheme()
         bindDarkTheme()
         bindLightTheme()
+        bindOledTheme()
         bindAutoBatteryTheme()
         setupRadioGroups()
         val tabletAndTabStripEnabled = Settings(requireContext()).isTabStripEnabled
@@ -111,6 +113,7 @@ class CustomizationFragment : PreferenceFragmentCompat() {
         addToRadioGroup(
             radioLightTheme,
             radioDarkTheme,
+            radioOledTheme,
             if (SDK_INT >= Build.VERSION_CODES.P) {
                 radioFollowDeviceTheme
             } else {
@@ -133,6 +136,13 @@ class CustomizationFragment : PreferenceFragmentCompat() {
         }
     }
 
+    private fun bindOledTheme() {
+        radioOledTheme = requirePreference(R.string.pref_key_oled_theme)
+        radioOledTheme.onClickListener {
+            setNewTheme(AppCompatDelegate.MODE_NIGHT_YES)
+        }
+    }
+
     private fun bindDarkTheme() {
         radioDarkTheme = requirePreference(R.string.pref_key_dark_theme)
         radioDarkTheme.onClickListener {
@@ -155,8 +165,10 @@ class CustomizationFragment : PreferenceFragmentCompat() {
     }
 
     private fun setNewTheme(mode: Int) {
-        if (AppCompatDelegate.getDefaultNightMode() == mode) return
-        AppCompatDelegate.setDefaultNightMode(mode)
+        if (AppCompatDelegate.getDefaultNightMode() != mode) {
+            AppCompatDelegate.setDefaultNightMode(mode)
+        }
+        
         activity?.recreate()
         with(requireComponents.core) {
             engine.settings.preferredColorScheme = getPreferredColorScheme()
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/theme/FirefoxTheme.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/theme/FirefoxTheme.kt
index 2c737e026e..29df2b06b3 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/theme/FirefoxTheme.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/theme/FirefoxTheme.kt
@@ -14,11 +14,13 @@ import mozilla.components.compose.base.theme.AcornTheme
 import mozilla.components.compose.base.theme.AcornTypography
 import mozilla.components.compose.base.theme.acornDarkColorScheme
 import mozilla.components.compose.base.theme.acornLightColorScheme
+import mozilla.components.compose.base.theme.acornOledColorScheme
 import mozilla.components.compose.base.theme.acornPrivateColorScheme
 import mozilla.components.compose.base.theme.darkColorPalette
 import mozilla.components.compose.base.theme.layout.AcornLayout
 import mozilla.components.compose.base.theme.layout.AcornWindowSize
 import mozilla.components.compose.base.theme.lightColorPalette
+import mozilla.components.compose.base.theme.oledColorPalette
 import mozilla.components.compose.base.theme.privateColorPalette
 import mozilla.components.compose.base.utils.inComposePreview
 import org.mozilla.fenix.ext.settings
@@ -37,12 +39,14 @@ fun FirefoxTheme(
     val colors: AcornColors = when (theme) {
         Theme.Light -> lightColorPalette
         Theme.Dark -> darkColorPalette
+        Theme.Oled -> oledColorPalette
         Theme.Private -> privateColorPalette
     }
 
     val colorScheme: ColorScheme = when (theme) {
         Theme.Light -> acornLightColorScheme()
         Theme.Dark -> acornDarkColorScheme()
+        Theme.Oled -> acornOledColorScheme()
         Theme.Private -> acornPrivateColorScheme()
     }
 
@@ -59,6 +63,7 @@ fun FirefoxTheme(
 enum class Theme {
     Light,
     Dark,
+    Oled,
     Private,
     ;
 
@@ -79,7 +84,11 @@ enum class Theme {
             ) {
                 Private
             } else if (isSystemInDarkTheme()) {
-                Dark
+                if (LocalContext.current.settings().shouldUseOledTheme) {
+                    Oled
+                } else {
+                    Dark
+                }
             } else {
                 Light
             }
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/theme/ThemeManager.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/theme/ThemeManager.kt
index d6a7782e1a..14a5e3f001 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/theme/ThemeManager.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/theme/ThemeManager.kt
@@ -11,10 +11,13 @@ import android.content.res.Configuration
 import android.util.TypedValue
 import android.view.Window
 import androidx.annotation.AnyRes
+import androidx.annotation.AttrRes
+import androidx.annotation.ColorInt
 import androidx.annotation.StyleRes
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.res.colorResource
+import androidx.core.content.ContextCompat
 import mozilla.components.support.ktx.android.content.getColorFromAttr
 import mozilla.components.support.ktx.android.content.getStatusBarColor
 import mozilla.components.support.ktx.android.view.createWindowInsetsController
@@ -24,9 +27,11 @@ import org.mozilla.fenix.HomeActivity
 import org.mozilla.fenix.R
 import org.mozilla.fenix.browser.browsingmode.BrowsingMode
 import org.mozilla.fenix.customtabs.ExternalAppBrowserActivity
+import org.mozilla.fenix.ext.settings
 
 abstract class ThemeManager {
 
+    protected abstract val activity: Activity
     abstract var currentTheme: BrowsingMode
 
     /**
@@ -34,7 +39,11 @@ abstract class ThemeManager {
      */
     @get:StyleRes
     val currentThemeResource get() = when (currentTheme) {
-        BrowsingMode.Normal -> R.style.NormalTheme
+        BrowsingMode.Normal -> if (activity.settings().shouldUseOledTheme) {
+            R.style.IronFox_OLEDTheme
+        } else {
+            R.style.NormalTheme
+        }
         BrowsingMode.Private -> R.style.PrivateTheme
     }
 
@@ -96,6 +105,12 @@ abstract class ThemeManager {
             return typedValue.resourceId
         }
 
+        @ColorInt
+        fun resolveAttributeColor(@AttrRes attribute: Int, context: Context): Int {
+            val resourceId = resolveAttribute(attribute, context)
+            return ContextCompat.getColor(context, resourceId)
+        }
+
         @Composable
         fun resolveAttributeColor(attribute: Int): androidx.compose.ui.graphics.Color {
             val resourceId = resolveAttribute(attribute, LocalContext.current)
@@ -139,7 +154,7 @@ abstract class ThemeManager {
 
 class DefaultThemeManager(
     currentTheme: BrowsingMode,
-    private val activity: Activity,
+    override val activity: Activity,
 ) : ThemeManager() {
     override var currentTheme: BrowsingMode = currentTheme
         set(value) {
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index aec4180603..c35f487733 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -717,6 +717,11 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         default = false,
     )
 
+    var shouldUseOledTheme by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_oled_theme),
+        default = false,
+    )
+
     var shouldUseAutoSize by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_accessibility_auto_size),
         default = true,
diff --git a/mobile/android/fenix/app/src/main/res/drawable/home_bottom_bar_background.xml b/mobile/android/fenix/app/src/main/res/drawable/home_bottom_bar_background.xml
index ea99799d42..80ed1401b8 100644
--- a/mobile/android/fenix/app/src/main/res/drawable/home_bottom_bar_background.xml
+++ b/mobile/android/fenix/app/src/main/res/drawable/home_bottom_bar_background.xml
@@ -6,7 +6,7 @@
 <layer-list xmlns:android="http://schemas.android.com/apk/res/android">
 <item>
     <shape>
-        <solid android:color="@color/fx_mobile_layer_color_1" />
+        <solid android:color="?attr/layer1" />
     </shape>
 </item>
 <item android:bottom="-2dp" android:left="-2dp" android:right="-2dp">
diff --git a/mobile/android/fenix/app/src/main/res/drawable/home_bottom_bar_background_no_divider.xml b/mobile/android/fenix/app/src/main/res/drawable/home_bottom_bar_background_no_divider.xml
index 5a2c61c89d..4bc14e53bd 100644
--- a/mobile/android/fenix/app/src/main/res/drawable/home_bottom_bar_background_no_divider.xml
+++ b/mobile/android/fenix/app/src/main/res/drawable/home_bottom_bar_background_no_divider.xml
@@ -5,7 +5,7 @@
 <layer-list xmlns:android="http://schemas.android.com/apk/res/android">
 <item>
     <shape>
-        <solid android:color="@color/fx_mobile_layer_color_1" />
+        <solid android:color="?attr/layer1" />
     </shape>
 </item>
 </layer-list>
diff --git a/mobile/android/fenix/app/src/main/res/drawable/home_bottom_bar_background_top.xml b/mobile/android/fenix/app/src/main/res/drawable/home_bottom_bar_background_top.xml
index 11e8ca17bb..f8e805ca2a 100644
--- a/mobile/android/fenix/app/src/main/res/drawable/home_bottom_bar_background_top.xml
+++ b/mobile/android/fenix/app/src/main/res/drawable/home_bottom_bar_background_top.xml
@@ -6,7 +6,7 @@
 <layer-list xmlns:android="http://schemas.android.com/apk/res/android">
 <item>
     <shape>
-        <solid android:color="@color/fx_mobile_layer_color_1" />
+        <solid android:color="?attr/layer1" />
     </shape>
 </item>
 <item android:top="-2dp" android:left="-2dp" android:right="-2dp">
diff --git a/mobile/android/fenix/app/src/main/res/drawable/recent_apps_background.xml b/mobile/android/fenix/app/src/main/res/drawable/recent_apps_background.xml
index 8149bfee3c..9f8677e08c 100644
--- a/mobile/android/fenix/app/src/main/res/drawable/recent_apps_background.xml
+++ b/mobile/android/fenix/app/src/main/res/drawable/recent_apps_background.xml
@@ -3,6 +3,6 @@
    - License, v. 2.0. If a copy of the MPL was not distributed with this
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
 <shape xmlns:android="http://schemas.android.com/apk/res/android">
-    <solid android:color="@color/fx_mobile_layer_color_1"/>
+    <solid android:color="?attr/layer1"/>
     <corners android:radius="@dimen/share_recent_apps_background_radius"/>
 </shape>
diff --git a/mobile/android/fenix/app/src/main/res/drawable/rounded_search_widget_background.xml b/mobile/android/fenix/app/src/main/res/drawable/rounded_search_widget_background.xml
index 4cd3c13437..bc8547a8ad 100644
--- a/mobile/android/fenix/app/src/main/res/drawable/rounded_search_widget_background.xml
+++ b/mobile/android/fenix/app/src/main/res/drawable/rounded_search_widget_background.xml
@@ -4,6 +4,6 @@
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
 
 <shape xmlns:android="http://schemas.android.com/apk/res/android">
-    <solid android:color="@color/fx_mobile_layer_color_1" />
+    <solid android:color="?attr/layer1" />
     <corners android:radius="@dimen/tab_corner_radius"/>
 </shape>
diff --git a/mobile/android/fenix/app/src/main/res/drawable/toolbar_background_no_divider.xml b/mobile/android/fenix/app/src/main/res/drawable/toolbar_background_no_divider.xml
index eb23393ffb..92983b9058 100644
--- a/mobile/android/fenix/app/src/main/res/drawable/toolbar_background_no_divider.xml
+++ b/mobile/android/fenix/app/src/main/res/drawable/toolbar_background_no_divider.xml
@@ -4,5 +4,5 @@
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
 <shape xmlns:android="http://schemas.android.com/apk/res/android"
     android:shape="rectangle">
-    <solid android:color="@color/fx_mobile_layer_color_1" />
+    <solid android:color="?attr/layer1" />
 </shape>
diff --git a/mobile/android/fenix/app/src/main/res/values-v27/styles.xml b/mobile/android/fenix/app/src/main/res/values-v27/styles.xml
index 70ebd5efb2..99734eb5ad 100644
--- a/mobile/android/fenix/app/src/main/res/values-v27/styles.xml
+++ b/mobile/android/fenix/app/src/main/res/values-v27/styles.xml
@@ -51,6 +51,6 @@
     <style name="TabTrayDialogStyle" parent="TabTrayDialogStyleBase">
         <item name="android:navigationBarDividerColor">@android:color/transparent</item>
         <item name="android:windowLightNavigationBar">@bool/theme_is_light</item>
-        <item name="android:navigationBarColor">@color/fx_mobile_layer_color_1</item>
+        <item name="android:navigationBarColor">?attr/layer1</item>
     </style>
 </resources>
diff --git a/mobile/android/fenix/app/src/main/res/values/styles.xml b/mobile/android/fenix/app/src/main/res/values/styles.xml
index 43f5deb1c9..416c54fb31 100644
--- a/mobile/android/fenix/app/src/main/res/values/styles.xml
+++ b/mobile/android/fenix/app/src/main/res/values/styles.xml
@@ -112,6 +112,7 @@
         <!-- Action -->
         <!-- Primary button, Snackbar, Floating action button, Chip selected -->
         <item name="actionPrimary">@color/fx_mobile_action_color_primary</item>
+        <item name="actionSecondary">@color/fx_mobile_action_color_secondary</item>
         <!-- Primary button in a disabled state -->
         <item name="actionPrimaryDisabled">@color/fx_mobile_action_color_primary_disabled</item>
         <!-- Selected chip -->
@@ -531,7 +532,7 @@
         <item name="android:layout_height">48dp</item>
         <item name="android:textStyle">bold</item>
         <item name="android:textAllCaps">false</item>
-        <item name="backgroundTint">@color/fx_mobile_action_color_secondary</item>
+        <item name="backgroundTint">?attr/actionSecondary</item>
         <item name="android:textColor">@color/fx_mobile_text_color_action_secondary</item>
         <item name="rippleColor">?attr/textSecondary</item>
         <item name="android:letterSpacing">0</item>
@@ -809,7 +810,7 @@
     <!-- Tab Tray does not present a private theme, so it needs to be separate from other bottom sheet styles -->
     <style name="TabTrayDialogStyleBase" parent="BottomSheetBase">
         <item name="bottomSheetStyle">@style/BottomSheetModal</item>
-        <item name="android:colorBackground">@color/fx_mobile_layer_color_1</item>
+        <item name="android:colorBackground">?attr/layer1</item>
     </style>
 
     <style name="TabTrayDialogStyle" parent="TabTrayDialogStyleBase" />
diff --git a/mobile/android/fenix/app/src/main/res/xml/customization_preferences.xml b/mobile/android/fenix/app/src/main/res/xml/customization_preferences.xml
index 6f05cd5ec6..260311a647 100644
--- a/mobile/android/fenix/app/src/main/res/xml/customization_preferences.xml
+++ b/mobile/android/fenix/app/src/main/res/xml/customization_preferences.xml
@@ -28,6 +28,11 @@
             android:key="@string/pref_key_dark_theme"
             android:title="@string/preference_dark_theme" />
 
+        <org.mozilla.fenix.settings.RadioButtonPreference
+            android:defaultValue="false"
+            android:key="@string/pref_key_oled_theme"
+            android:title="@string/preference_oled_theme" />
+
         <org.mozilla.fenix.settings.RadioButtonPreference
             android:defaultValue="false"
             android:key="@string/pref_key_auto_battery_theme"
