diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/home/pocket/ContentRecommendationsFeatureHelper.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/home/pocket/ContentRecommendationsFeatureHelper.kt
index d211d3625d..cbdc241b9b 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/home/pocket/ContentRecommendationsFeatureHelper.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/home/pocket/ContentRecommendationsFeatureHelper.kt
@@ -47,6 +47,6 @@ object ContentRecommendationsFeatureHelper {
     fun isContentRecommendationsFeatureEnabled(context: Context): Boolean {
         val langTag = LocaleManager.getCurrentLocale(context)
             ?.toLanguageTag() ?: getSystemDefault().toLanguageTag()
-        return CONTENT_RECOMMENDATIONS_SUPPORTED_LOCALE.contains(langTag)
+        return false && CONTENT_RECOMMENDATIONS_SUPPORTED_LOCALE.contains(langTag)
     }
 }
