diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index a46a154665..5954401610 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -195,7 +195,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         } else {
             preferences.getBoolean(
                 appContext.getPreferenceKey(R.string.pref_key_customization_bookmarks),
-                homescreenSections[HomeScreenSection.BOOKMARKS] == true,
+                false,
             )
         }
 
@@ -208,7 +208,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         } else {
             preferences.getBoolean(
                 appContext.getPreferenceKey(R.string.pref_key_recent_tabs),
-                homescreenSections[HomeScreenSection.JUMP_BACK_IN] == true,
+                false,
             )
         }
         set(value) {
@@ -244,7 +244,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         } else {
             preferences.getBoolean(
                 appContext.getPreferenceKey(R.string.pref_key_history_metadata_feature),
-                homescreenSections[HomeScreenSection.RECENT_EXPLORATIONS] == true,
+                false,
             )
         }
         set(value) {
@@ -262,8 +262,10 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     /**
      * Indicates whether or not the "Collections" section should be shown on the home screen.
      */
-    val collections: Boolean
-        get() = FxNimbus.features.homescreen.value().sectionsEnabled[HomeScreenSection.COLLECTIONS] == true
+    var collections: Boolean by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_collections_enabled),
+        default = true,
+    )
 
     /**
      * Indicates whether or not the homepage header should be shown.
@@ -319,7 +321,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     val overrideUserSpecifiedHomepageSections by lazyFeatureFlagPreference(
         appContext.getPreferenceKey(R.string.pref_key_override_user_specified_homepage_sections),
-        featureFlag = true,
+        featureFlag = false,
         default = { FxNimbus.features.overrideUserSpecifiedHomepageSections.value().enabled },
     )
 
@@ -555,7 +557,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
 
     var showCollectionsPlaceholderOnHome by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_show_collections_placeholder_home),
-        default = true,
+        default = false,
     )
 
     val isCrashReportingEnabled: Boolean
@@ -1422,7 +1424,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
 
     var defaultTopSitesAdded by booleanPreference(
         appContext.getPreferenceKey(R.string.default_top_sites_added),
-        default = false,
+        default = true,
     )
 
     var shouldShowSearchSuggestionsInPrivate by booleanPreference(
