diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt
index 2be5597f36..1b3866a281 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt
@@ -136,6 +136,7 @@ import org.mozilla.geckoview.GeckoRuntime
 import java.util.UUID
 import java.util.concurrent.TimeUnit
 import mozilla.components.service.pocket.mars.api.Placement as MarsSpocsPlacement
+import org.ironfoxoss.ironfox.utils.GeckoSettingsBridge
 
 /**
  * Component group for all core browser functionality.
@@ -228,6 +229,7 @@ class Core(
             defaultSettings,
             geckoRuntime,
         ).also {
+            GeckoSettingsBridge.setIronFoxPrefs(context, it)
             WebCompatFeature.install(it)
         }
     }
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
index e1a67a6da4..a98b4656aa 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
@@ -591,6 +591,7 @@ class SettingsFragment : PreferenceFragmentCompat() {
         setupHomepagePreference(settings)
         setupTrackingProtectionPreference(settings)
         setupDnsOverHttpsPreference(settings)
+        setupIronFoxPreferences()
     }
 
     /**
@@ -789,6 +790,19 @@ class SettingsFragment : PreferenceFragmentCompat() {
             }
     }
 
+    @VisibleForTesting
+    internal fun setupIronFoxPreferences() {
+        requirePreference<Preference>(R.string.pref_key_if_preferences).apply {
+            setOnPreferenceClickListener {
+                val directions =
+                    SettingsFragmentDirections
+                        .actionSettingsFragmentToIronFoxSettingsFragment()
+                requireView().findNavController().navigate(directions)
+                true
+            }
+        }
+    }
+
     private fun updateProfilerUI(profilerStatus: Boolean) {
         if (profilerStatus) {
             findPreference<Preference>(getPreferenceKey(R.string.pref_key_start_profiler))?.title =
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 1c7b0fce0d..8cfbbf9cfa 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -2736,4 +2736,174 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         val cleanupPreferenceKey = appContext.getString(R.string.pref_key_downloads_clean_up_files_automatically)
         return sharedPreferences.getBoolean(cleanupPreferenceKey, false)
     }
+
+    var accessibilityEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_accessibility_enabled),
+        default = false,
+    )
+
+    var autoplayBlockingClickToPlay by booleanPreference(
+        key = appContext.getPreferenceKey(R.string.pref_key_autoplay_policy_click_to_play),
+        default = false,
+    )
+
+    var autoplayBlockingSticky by booleanPreference(
+        key = appContext.getPreferenceKey(R.string.pref_key_autoplay_policy_sticky),
+        default = false,
+    )
+
+    var autoplayBlockingTransient by booleanPreference(
+        key = appContext.getPreferenceKey(R.string.pref_key_autoplay_policy_transient),
+        default = true,
+    )
+
+    var cacheEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_cache_enabled),
+        default = false,
+    )
+
+    var emeEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_eme_enabled),
+        default = false,
+    )
+
+    var enableUnifiedPush by booleanPreference(
+        key = appContext.getPreferenceKey(R.string.pref_key_enable_unifiedpush),
+        default = true,
+    )
+
+    var fppOverridesIronFoxEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_fpp_overrides_ironfox_enabled),
+        default = true,
+    )
+
+    var fppOverridesIronFoxTimezoneEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_fpp_overrides_ironfox_timezone_enabled),
+        default = true,
+    )
+
+    var fppOverridesIronFoxWebGLEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_fpp_overrides_ironfox_webgl_enabled),
+        default = true,
+    )
+
+    var fppOverridesMozillaEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_fpp_overrides_mozilla_enabled),
+        default = true,
+    )
+
+    var ironFoxOnboardingCompleted by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_if_onboarding_completed),
+        default = false,
+    )
+
+    var ipv6Enabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_ipv6_enabled),
+        default = true,
+    )
+
+    var javascriptEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_javascript_enabled),
+        default = true,
+    )
+
+    var javascriptJitEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_jit_enabled),
+        default = false,
+    )
+
+    var javascriptJitTrustedPrincipalsEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_jit_trusted_principals_enabled),
+        default = false,
+    )
+
+    var pdfjsDisabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_pdfjs_disabled),
+        default = false,
+    )
+
+    var prefersBrowserColorScheme by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_prefers_browser_color_scheme),
+        default = false,
+    )
+
+    var prefersDarkColorScheme by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_prefers_dark_color_scheme),
+        default = false,
+    )
+
+    var prefersLightColorScheme by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_prefers_light_color_scheme),
+        default = true,
+    )
+
+    var printEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_print_enabled),
+        default = true,
+    )
+
+    var refererXOriginAlways by booleanPreference(
+        key = appContext.getPreferenceKey(R.string.pref_key_referer_policy_always),
+        default = true,
+    )
+
+    var refererXOriginBaseDomainsMatch by booleanPreference(
+        key = appContext.getPreferenceKey(R.string.pref_key_referer_policy_base_domains_match),
+        default = false,
+    )
+
+    var refererXOriginHostsMatch by booleanPreference(
+        key = appContext.getPreferenceKey(R.string.pref_key_referer_policy_hosts_match),
+        default = false,
+    )
+
+    var safeBrowsingEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_safe_browsing_enabled),
+        default = true,
+    )
+
+    var spoofEnglish by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_spoof_english),
+        default = true,
+    )
+
+    var spoofTimezone by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_spoof_timezone),
+        default = false,
+    )
+
+    var svgEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_svg_enabled),
+        default = true,
+    )
+
+    var translationsEnabled by booleanPreference(
+        key = appContext.getPreferenceKey(R.string.pref_key_translations_enabled),
+        default = true,
+    )
+
+    var useUnifiedPush by booleanPreference(
+        key = appContext.getPreferenceKey(R.string.pref_key_use_unifiedpush),
+        default = false,
+    )
+
+    var wasmEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_wasm_enabled),
+        default = true,
+    )
+
+    var webglDisabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_webgl_disabled),
+        default = true,
+    )
+
+    var webrtcEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_webrtc_enabled),
+        default = true,
+    )
+
+    var xpinstallEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_xpinstall_enabled),
+        default = false,
+    )
 }
diff --git a/mobile/android/fenix/app/src/main/res/navigation/nav_graph.xml b/mobile/android/fenix/app/src/main/res/navigation/nav_graph.xml
index a4fd0897fa..634f9c1b8c 100644
--- a/mobile/android/fenix/app/src/main/res/navigation/nav_graph.xml
+++ b/mobile/android/fenix/app/src/main/res/navigation/nav_graph.xml
@@ -676,6 +676,13 @@
             app:exitAnim="@anim/slide_out_left"
             app:popEnterAnim="@anim/slide_in_left"
             app:popExitAnim="@anim/slide_out_right" />
+        <action
+            android:id="@+id/action_settingsFragment_to_ironFoxSettingsFragment"
+            app:destination="@id/ironFoxSettingsFragment"
+            app:enterAnim="@anim/slide_in_right"
+            app:exitAnim="@anim/slide_out_left"
+            app:popEnterAnim="@anim/slide_in_left"
+            app:popExitAnim="@anim/slide_out_right" />
         <action
             android:id="@+id/action_settingsFragment_to_trackingProtectionFragment"
             app:destination="@id/trackingProtectionFragment"
@@ -947,6 +954,10 @@
         android:id="@+id/sponsoredStoriesSettings"
         android:name="org.mozilla.fenix.settings.SponsoredStoriesSettingsFragment"
         android:label="@string/preferences_debug_settings_custom_sponsored_stories_parameters" />
+    <fragment
+        android:id="@+id/ironFoxSettingsFragment"
+        android:name="org.mozilla.fenix.settings.IronFoxSettingsFragment"
+        android:label="@string/if_preferences" />
     <fragment
         android:id="@+id/trackingProtectionFragment"
         android:name="org.mozilla.fenix.settings.TrackingProtectionFragment">
diff --git a/mobile/android/fenix/app/src/main/res/xml/preferences.xml b/mobile/android/fenix/app/src/main/res/xml/preferences.xml
index 5058f40c0b..a040bb1643 100644
--- a/mobile/android/fenix/app/src/main/res/xml/preferences.xml
+++ b/mobile/android/fenix/app/src/main/res/xml/preferences.xml
@@ -198,10 +198,22 @@
         <androidx.preference.SwitchPreference
             android:defaultValue="false"
             android:key="@string/pref_key_enable_gecko_logs"
+            android:summary="@string/restart_required_summary"
             app:iconSpaceReserved="false"
             android:title="@string/preferences_enable_gecko_logs" />
     </PreferenceCategory>
 
+    <androidx.preference.PreferenceCategory
+        android:layout="@layout/preference_category_no_icon_style"
+        android:selectable="false"
+        android:title="@string/if_title"
+        app:iconSpaceReserved="false">
+        <androidx.preference.Preference
+            android:key="@string/pref_key_if_preferences"
+            android:title="@string/if_preferences"
+            app:iconSpaceReserved="false" />
+    </androidx.preference.PreferenceCategory>
+
     <androidx.preference.PreferenceCategory
         android:title="@string/preferences_category_about"
         app:iconSpaceReserved="false"
