diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
index 25b489cf24e9..be062b986ba8 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
@@ -27,7 +27,7 @@ import org.mozilla.fenix.ext.components
 import org.mozilla.fenix.ext.nav
 import org.mozilla.fenix.ext.settings
 import org.mozilla.fenix.ext.showToolbar
-import org.mozilla.fenix.GleanMetrics.DebugDrawer as DebugDrawerMetrics
+import org.ironfoxoss.ironfox.utils.FenixSettingsVisibilityManager
 
 class SecretSettingsFragment : PreferenceFragmentCompat() {
 
@@ -434,6 +434,7 @@ class SecretSettingsFragment : PreferenceFragmentCompat() {
             isChecked = context.settings().isRelayFeatureEnabled
             onPreferenceChangeListener = SharedPreferenceUpdater()
         }
+        FenixSettingsVisibilityManager.SecretSettingsFragment(requireContext(), this)
     }
 
     override fun onPreferenceTreeClick(preference: Preference): Boolean {
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 276ad8cf1df1..b2730addde36 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -1383,7 +1383,7 @@ class Settings(
 
     var shouldShowToolbarCustomization by booleanPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_enable_toolbar_customization),
-        default = { FxNimbus.features.toolbarRedesignOption.value().showCustomization },
+        default = { true },
     )
 
     val toolbarPosition: ToolbarPosition
@@ -1965,7 +1965,7 @@ class Settings(
 
     var isTabStripEnabled by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_tab_strip_show),
-        default = FxNimbus.features.tabStrip.value().enabled &&
+        default = true &&
                 (isTabStripEligible(appContext) || FxNimbus.features.tabStrip.value().allowOnAllDevices),
     )
 
@@ -2015,7 +2015,7 @@ class Settings(
      */
     private fun isAddressFeatureEnabled(context: Context): Boolean {
         val locale = LocaleManager.getCurrentLocale(context) ?: LocaleManager.getSystemDefault()
-        val debugRepository = if (Config.channel.isNightlyOrDebug) {
+        val debugRepository = if (true) {
             SharedPrefsAddressesDebugRegionRepository(context)
         } else {
             EmptyAddressesDebugRegionRepository()
@@ -2201,7 +2201,7 @@ class Settings(
      */
     var toolbarRedesignEnabled by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_enable_toolbar_redesign),
-        default = { FxNimbus.features.toolbarRedesignOption.value().showOptions },
+        default = { true },
     )
 
     /**
