diff --git a/mobile/android/fenix/app/build.gradle b/mobile/android/fenix/app/build.gradle
index 1b27020a20ee7..e793f03573c9d 100644
--- a/mobile/android/fenix/app/build.gradle
+++ b/mobile/android/fenix/app/build.gradle
@@ -118,7 +118,7 @@ android {
 
 	// If building locally, just use debug signing. In official automation builds, the
 	// files are signed using a separate service and not within gradle.
-        if (!gradle.mozconfig.substs.MOZ_AUTOMATION) {
+        if (false) {
             signingConfig = signingConfigs.debug
         }
 
