diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt
index ec3c88ea9961..c97d048e42d9 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt
@@ -203,7 +203,7 @@ class BrowserFragment : BaseBrowserFragment(), UserInteractionHandler {
 
     private fun initTranslationsAction(context: Context, view: View) {
         if (
-            !FxNimbus.features.translations.value().mainFlowToolbarEnabled
+            !requireContext().settings().ironfox.translationsEnabled
         ) {
             return
         }
@@ -301,7 +301,7 @@ class BrowserFragment : BaseBrowserFragment(), UserInteractionHandler {
             view = rootView,
         )
 
-        if (FxNimbus.features.translations.value().mainFlowToolbarEnabled) {
+        if (requireContext().settings().ironfox.translationsEnabled) {
             translationsBinding.set(
                 feature = TranslationsBinding(
                     browserStore = rootView.context.components.core.store,
