diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
index e7e19273f9..8572b6bfdb 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
@@ -179,7 +179,7 @@ open class DefaultToolbarMenu(
     @VisibleForTesting(otherwise = PRIVATE)
     fun shouldShowOpenInRegularTab(): Boolean = selectedSession?.let { session ->
         // This feature is gated behind Nightly for the time being.
-        Config.channel.isNightlyOrDebug &&
+        true &&
             // This feature is explicitly for users opening links in private tabs.
             context.settings().openLinksInAPrivateTab &&
             // and is only visible in private tabs.
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
index 9113dc53a9..93b9698db5 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
@@ -64,13 +64,13 @@ class SecretSettingsFragment : PreferenceFragmentCompat() {
         }
 
         requirePreference<SwitchPreference>(R.string.pref_key_enable_composable_toolbar).apply {
-            isVisible = Config.channel.isDebug
+            isVisible = true
             isChecked = context.settings().shouldUseComposableToolbar
             onPreferenceChangeListener = SharedPreferenceUpdater()
         }
 
         requirePreference<SwitchPreference>(R.string.pref_key_enable_toolbar_redesign).apply {
-            isVisible = Config.channel.isDebug
+            isVisible = true
             isChecked = context.settings().toolbarRedesignEnabled
             onPreferenceChangeListener = SharedPreferenceUpdater()
         }
@@ -81,7 +81,7 @@ class SecretSettingsFragment : PreferenceFragmentCompat() {
             onPreferenceChangeListener = SharedPreferenceUpdater()
         }
         requirePreference<SwitchPreference>(R.string.pref_key_enable_address_sync).apply {
-            isVisible = Config.channel.isNightlyOrDebug
+            isVisible = true
             isChecked = context.settings().isAddressSyncEnabled
             onPreferenceChangeListener = SharedPreferenceUpdater()
         }
@@ -92,13 +92,13 @@ class SecretSettingsFragment : PreferenceFragmentCompat() {
         }
 
         requirePreference<SwitchPreference>(R.string.pref_key_enable_menu_redesign).apply {
-            isVisible = Config.channel.isNightlyOrDebug
+            isVisible = true
             isChecked = context.settings().enableMenuRedesign
             onPreferenceChangeListener = SharedPreferenceUpdater()
         }
 
         requirePreference<SwitchPreference>(R.string.pref_key_enable_homepage_searchbar).apply {
-            isVisible = Config.channel.isNightlyOrDebug
+            isVisible = true
             isChecked = context.settings().enableHomepageSearchBar
             onPreferenceChangeListener = SharedPreferenceUpdater()
         }
@@ -110,13 +110,13 @@ class SecretSettingsFragment : PreferenceFragmentCompat() {
         }
 
         requirePreference<SwitchPreference>(R.string.pref_key_enable_homepage_as_new_tab).apply {
-            isVisible = Config.channel.isNightlyOrDebug
+            isVisible = true
             isChecked = context.settings().enableHomepageAsNewTab
             onPreferenceChangeListener = SharedPreferenceUpdater()
         }
 
         requirePreference<SwitchPreference>(R.string.pref_key_enable_unified_trust_panel).apply {
-            isVisible = Config.channel.isNightlyOrDebug
+            isVisible = true
             isChecked = context.settings().enableUnifiedTrustPanel
             onPreferenceChangeListener = SharedPreferenceUpdater()
         }
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 3fa564af49..9ae54264fa 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -449,7 +449,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     var privateBrowsingLockedFeatureEnabled by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_private_browsing_locked_enabled),
         featureFlag = true,
-        default = { FxNimbus.features.privateBrowsingLock.value().enabled },
+        default = { true },
     )
 
     var privateBrowsingModeLocked by booleanPreference(
@@ -622,7 +622,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var isShortcutSuggestionsVisible by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_enable_shortcuts_suggestions),
-        default = { FxNimbus.features.topSitesSuggestions.value().enabled },
+        default = { true },
         featureFlag = true,
     )
 
@@ -1778,7 +1778,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
 
     var isTabStripEnabled by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_tab_strip_show),
-        default = FxNimbus.features.tabStrip.value().enabled &&
+        default = true &&
                 (isTabStripEligible(appContext) || FxNimbus.features.tabStrip.value().allowOnAllDevices),
     )
 
@@ -1828,7 +1828,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
             "en-CA",
             "fr-CA",
         )
-        val currentlyEnabledLanguages = if (Config.channel.isNightlyOrDebug) {
+        val currentlyEnabledLanguages = if (true) {
             releaseEnabledLanguages + SharedPrefsAddressesDebugLocalesRepository(context)
                 .getAllEnabledLocales().map { it.langTag }
         } else {
@@ -1986,7 +1986,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
 
     var shouldUseComposableToolbar by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_enable_composable_toolbar),
-        default = { FxNimbus.features.composableToolbar.value().enabled },
+        default = { false },
         featureFlag = true,
     )
 
@@ -1995,7 +1995,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var toolbarRedesignEnabled by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_enable_toolbar_redesign),
-        default = { FxNimbus.features.toolbarRedesignOption.value().showOptions },
+        default = { true },
     )
 
     /**
@@ -2105,7 +2105,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var enableMenuRedesign by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_enable_menu_redesign),
-        default = { FxNimbus.features.menuRedesign.value().enabled },
+        default = { false },
         featureFlag = true,
     )
 
@@ -2114,7 +2114,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var enableHomepageAsNewTab by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_enable_homepage_as_new_tab),
-        default = { FxNimbus.features.homepageAsNewTab.value().enabled },
+        default = { false },
         featureFlag = true,
     )
 
@@ -2123,7 +2123,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var enableHomepageSearchBar by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_enable_homepage_searchbar),
-        default = { FxNimbus.features.homepageSearchBar.value().enabled },
+        default = { false },
         featureFlag = true,
     )
 
