diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/bookmarks/BookmarksScreen.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/bookmarks/BookmarksScreen.kt
index 0ae6d1f72bb7..9d62be2add5f 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/bookmarks/BookmarksScreen.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/bookmarks/BookmarksScreen.kt
@@ -1121,7 +1121,7 @@ private fun EmptyListState.drawableId(): Int = when (this) {
 
 @StringRes
 private fun EmptyListState.descriptionId(): Int = when (this) {
-    is EmptyListState.NotAuthenticated -> R.string.bookmark_empty_list_guest_description
+    is EmptyListState.NotAuthenticated -> R.string.bookmark_empty_list_folder_description
     EmptyListState.Authenticated -> R.string.bookmark_empty_list_authenticated_description
     EmptyListState.Folder -> R.string.bookmark_empty_list_folder_description
 }
@@ -1158,7 +1158,7 @@ private fun EmptyList(
                 color = FirefoxTheme.colors.textPrimary,
                 textAlign = TextAlign.Center,
             )
-            if (state is EmptyListState.NotAuthenticated) {
+            if (false) {
                 FilledButton(
                     text = stringResource(R.string.bookmark_empty_list_guest_cta),
                     onClick = { dispatcher(SignIntoSyncClicked) },
