From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sat, 27 Jun 2026 20:11:41 +0000
Subject: [PATCH] Fenix - IronFox Settings: Collections

Provides support for a UI setting to enable/disable collections.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index e1eedc43d242..0ba3068b54e7 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -251,8 +251,10 @@ class Settings(
     /**
      * Indicates whether or not the "Collections" section should be shown on the home screen.
      */
-    val collections: Boolean
-        get() = FxNimbus.features.homescreen.value().sectionsEnabled[HomeScreenSection.COLLECTIONS] == true
+    var collections: Boolean by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_collections_enabled),
+        default = true,
+    )
 
     /**
      * Indicates whether or not the Firefox Japan Guide default site should be shown.
-- 
