diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
index 4d499da8de..df086baa43 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
@@ -532,7 +532,7 @@ class SettingsFragment : PreferenceFragmentCompat() {
             }
         }
 
-        preferenceRemoteDebugging?.isVisible = true
+        preferenceRemoteDebugging?.isVisible = false
         preferenceRemoteDebugging?.setOnPreferenceChangeListener<Boolean> { preference, newValue ->
             settings.preferences.edit { putBoolean(preference.key, newValue) }
             requireComponents.core.engine.settings.remoteDebuggingEnabled = newValue
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 312b5b3de3..b9ca88c804 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -568,10 +568,10 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         default = CrashReportOption.Ask.toString(),
     )
 
-    val isRemoteDebuggingEnabled by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_remote_debugging),
-        default = false,
-    )
+    val isRemoteDebuggingEnabled = false
+//        appContext.getPreferenceKey(R.string.pref_key_remote_debugging),
+//        default = false,
+//    )
 
     var isTelemetryEnabled by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_telemetry),
