diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index c7db941b0f7e..faa6a52a4fbf 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -872,7 +872,7 @@ class Settings(
     var isLnaBlockingEnabled by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_enable_lna_blocking_enabled),
         featureFlag = true,
-        default = { FxNimbus.features.lnaBlocking.value().blocking },
+        default = { true },
     )
 
     /**
@@ -881,7 +881,7 @@ class Settings(
     var isLnaTrackerBlockingEnabled by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_enable_lna_tracker_blocking_enabled),
         featureFlag = true,
-        default = { FxNimbus.features.lnaBlocking.value().blockTrackers },
+        default = { true },
     )
 
     /**
@@ -894,7 +894,7 @@ class Settings(
     var isLnaFeatureEnabled by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_enable_lna_feature_enabled),
         featureFlag = true,
-        default = { FxNimbus.features.lnaBlocking.value().enabled },
+        default = { true },
     )
 
     /**
