diff --git a/mobile/android/android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/worker/Extensions.kt b/mobile/android/android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/worker/Extensions.kt
index d897b9af6f..086f87d87b 100644
--- a/mobile/android/android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/worker/Extensions.kt
+++ b/mobile/android/android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/worker/Extensions.kt
@@ -12,9 +12,9 @@ import java.io.IOException
  * Indicates if an exception should be reported to the crash reporter.
  */
 internal fun Exception.shouldReport(): Boolean {
-    val isRecoverable = (this as? WebExtensionException)?.isRecoverable ?: true
-    return cause !is IOException &&
-        cause !is CancellationException &&
-        this !is CancellationException &&
-        isRecoverable
+//    val isRecoverable = (this as? WebExtensionException)?.isRecoverable ?: true
+    return false
+//        cause !is CancellationException &&
+//        this !is CancellationException &&
+//        isRecoverable
 }
diff --git a/mobile/android/android-components/components/lib/crash/src/main/java/mozilla/components/lib/crash/service/GleanCrashReporterService.kt b/mobile/android/android-components/components/lib/crash/src/main/java/mozilla/components/lib/crash/service/GleanCrashReporterService.kt
index 68dd4d13fa..7266ac2b6e 100644
--- a/mobile/android/android-components/components/lib/crash/src/main/java/mozilla/components/lib/crash/service/GleanCrashReporterService.kt
+++ b/mobile/android/android-components/components/lib/crash/src/main/java/mozilla/components/lib/crash/service/GleanCrashReporterService.kt
@@ -103,7 +103,7 @@ class GleanCrashReporterService(
         @SerialName("count")
         data class Count(val label: String) : GleanCrashAction() {
             override fun submit() {
-                CrashMetrics.crashCount[label].add()
+//                CrashMetrics.crashCount[label].add()
             }
         }
 
@@ -119,11 +119,11 @@ class GleanCrashReporterService(
                 val minidumpHash: String?,
             ) : PingCause() {
                 override fun setMetrics() {
-                    GleanCrash.cause.set("os_fault")
+/*                    GleanCrash.cause.set("os_fault")
                     remoteType?.let { GleanCrash.remoteType.set(it) }
                     minidumpHash?.let { GleanCrash.minidumpSha256Hash.set(it) }
 
-                    extras?.let(::setExtraMetrics)
+                    extras?.let(::setExtraMetrics)*/
                 }
 
                 private fun setExtraMetrics(extras: JsonObject) {
@@ -306,13 +306,13 @@ class GleanCrashReporterService(
                 val appBuildId: String? = null,
             ) : PingCause() {
                 override fun setMetrics() {
-                    GleanCrash.cause.set("java_exception")
+/*                    GleanCrash.cause.set("java_exception")
                     appChannel?.let { GleanCrash.appChannel.set(it) }
                     appBuildId?.let { GleanCrash.appBuild.set(it) }
                     appVersion?.let { GleanCrash.appDisplayVersion.set(it) }
                     GleanCrash.javaException.set(
                         Json.decodeFromJsonElement<GleanCrash.JavaExceptionObject>(throwableJson),
-                    )
+                    )*/
                 }
             }
         }
@@ -334,7 +334,7 @@ class GleanCrashReporterService(
                 // case a previous crash submission attempt was aborted due to
                 // an unexpected exception and the metrics were left
                 // partially-populated. See bug 1961202.
-                Pings.crash.setEnabled(false)
+/*                Pings.crash.setEnabled(false)
                 Pings.crash.setEnabled(true)
 
                 GleanEnvironment.uptime.setRawNanos(uptimeNanos)
@@ -373,7 +373,7 @@ class GleanCrashReporterService(
                     )
                 }
 
-                Pings.crash.submit(reason)
+                Pings.crash.submit(reason)*/
             }
         }
     }
@@ -430,7 +430,7 @@ class GleanCrashReporterService(
             logger.error("Expected file, but found directory")
             false
         } else {
-            true
+            false
         }
     }
 
@@ -519,7 +519,7 @@ class GleanCrashReporterService(
     }
 
     override fun record(crash: Crash.UncaughtExceptionCrash) {
-        recordCrashAction(GleanCrashAction.Count(UNCAUGHT_EXCEPTION_KEY))
+/*        recordCrashAction(GleanCrashAction.Count(UNCAUGHT_EXCEPTION_KEY))
 
         recordCrashAction(
             GleanCrashAction.Ping(
@@ -535,7 +535,7 @@ class GleanCrashReporterService(
                 ),
                 breadcrumbs = crash.breadcrumbs.map { it.toBreadcrumb() },
             ),
-        )
+        )*/
     }
 
     private fun getExtrasJson(path: String): JsonObject? {
@@ -614,7 +614,7 @@ class GleanCrashReporterService(
     }
 
     override fun record(crash: Crash.NativeCodeCrash) {
-        when (crash.processVisibility) {
+/*        when (crash.processVisibility) {
             Crash.NativeCodeCrash.PROCESS_VISIBILITY_MAIN ->
                 recordCrashAction(GleanCrashAction.Count(MAIN_PROCESS_NATIVE_CODE_CRASH_KEY))
             Crash.NativeCodeCrash.PROCESS_VISIBILITY_FOREGROUND_CHILD ->
@@ -654,10 +654,10 @@ class GleanCrashReporterService(
                 ),
                 breadcrumbs = crash.breadcrumbs.map { it.toBreadcrumb() },
             ),
-        )
+        )*/
     }
 
     override fun record(throwable: Throwable) {
-        recordCrashAction(GleanCrashAction.Count(CAUGHT_EXCEPTION_KEY))
+//        recordCrashAction(GleanCrashAction.Count(CAUGHT_EXCEPTION_KEY))
     }
 }
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt
index a88ff8d3ac..a1ed1502ff 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt
@@ -514,10 +514,10 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
     }
 
     private fun setupCrashReporting() {
-        components
+/*        components
             .analytics
             .crashReporter
-            .install(this)
+            .install(this)*/
     }
 
     protected open fun initializeNimbus() {
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Analytics.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Analytics.kt
index f04adf8312..fb8c49c827 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Analytics.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Analytics.kt
@@ -16,7 +16,7 @@ import mozilla.components.lib.crash.service.GleanCrashReporterService
 import mozilla.components.lib.crash.service.MozillaSocorroService
 import mozilla.components.support.ktx.android.content.isMainProcess
 import mozilla.components.support.utils.BrowsersCache
-import mozilla.components.support.utils.RunWhenReadyQueue
+//import mozilla.components.support.utils.RunWhenReadyQueue
 import org.mozilla.fenix.BuildConfig
 import org.mozilla.fenix.Config
 import org.mozilla.fenix.HomeActivity
@@ -45,7 +45,7 @@ import org.mozilla.geckoview.BuildConfig.MOZ_UPDATE_CHANNEL
  */
 class Analytics(
     private val context: Context,
-    private val runWhenReadyQueue: RunWhenReadyQueue,
+//    private val runWhenReadyQueue: RunWhenReadyQueue,
 ) {
     val crashReporter: CrashReporter by lazyMonitored {
         val services = mutableListOf<CrashReporterService>()
@@ -74,12 +74,12 @@ class Analytics(
 
             // We only want to initialize Sentry on startup on the main process.
             if (context.isMainProcess()) {
-                runWhenReadyQueue.runIfReadyOrQueue {
-                    sentryService.initIfNeeded()
-                }
+//                runWhenReadyQueue.runIfReadyOrQueue {
+//                    sentryService.initIfNeeded()
+//                }
             }
 
-            services.add(sentryService)
+//            services.add(sentryService)
         }
 
         // The name "Fenix" here matches the product name on Socorro and is unrelated to the actual app name:
@@ -93,7 +93,7 @@ class Analytics(
             releaseChannel = MOZ_UPDATE_CHANNEL,
             distributionId = distributionId,
         )
-        services.add(socorroService)
+//        services.add(socorroService)
 
         val intent = Intent(context, HomeActivity::class.java).apply {
             flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TOP
@@ -126,7 +126,7 @@ class Analytics(
                 appName = context.getString(R.string.app_name),
                 organizationName = "Mozilla",
             ),
-            enabled = true,
+            enabled = false,
             nonFatalCrashIntent = pendingIntent,
             useLegacyReporting = !context.settings().crashReportAlwaysSend &&
                 !context.settings().useNewCrashReporterDialog,
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/experiments/NimbusSetup.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/experiments/NimbusSetup.kt
index 36d4b4c349..f9ff9db3bf 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/experiments/NimbusSetup.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/experiments/NimbusSetup.kt
@@ -104,7 +104,7 @@ private fun Context.reportError(message: String, e: Throwable) {
 fun NimbusException.isReportableError(): Boolean {
     return when (this) {
         is NimbusException.ClientException -> false
-        else -> true
+        else -> false
     }
 }
 
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/SearchDialogController.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/SearchDialogController.kt
index 819b24954e..0f9bb0a905 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/SearchDialogController.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/SearchDialogController.kt
@@ -83,13 +83,13 @@ class SearchDialogController(
         }
 
         when (url) {
-            "about:crashes" -> {
+/*            "about:crashes" -> {
                 // The list of past crashes can be accessed via "settings > about", but desktop and
                 // fennec users may be used to navigating to "about:crashes". So we intercept this here
                 // and open the crash list activity instead.
                 val directions = SearchDialogFragmentDirections.actionCrashListFragment()
                 navController.navigate(directions)
-            }
+            }*/
             "about:addons" -> {
                 val directions =
                     SearchDialogFragmentDirections.actionGlobalAddonsManagementFragment()
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 8913d9f230..317cc8ad23 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -2300,10 +2300,10 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      * A user preference indicating that crash reports should always be automatically sent. This can be updated
      * through the unsubmitted crash dialog or through data choice preferences.
      */
-    var crashReportAlwaysSend by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_crash_reporting_always_report),
-        default = false,
-    )
+    var crashReportAlwaysSend = false
+//        appContext.getPreferenceKey(R.string.pref_key_crash_reporting_always_report),
+//        default = false,
+//    )
 
     /**
      * Indicates whether or not we should use the new crash reporter dialog.
