diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 83e0f40483b7..61833cefeabf 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -1146,10 +1146,10 @@ class Settings(
         default = false,
     )
 
-    var shouldUseTrackingProtection by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_tracking_protection),
+    var shouldUseTrackingProtection = true
+/*        appContext.getPreferenceKey(R.string.pref_key_tracking_protection),
         default = true,
-    )
+    ) */
 
     var shouldUseTrackingProtectionDatabase by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_tracking_protection_database_status),
@@ -1212,20 +1212,20 @@ class Settings(
         default = false,
     )
 
-    val useStandardTrackingProtection by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_tracking_protection_standard_option),
+    val useStandardTrackingProtection = false
+/*        appContext.getPreferenceKey(R.string.pref_key_tracking_protection_standard_option),
         true,
-    )
+    ) */
 
-    val useStrictTrackingProtection by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_tracking_protection_strict_default),
+    val useStrictTrackingProtection = true
+/*        appContext.getPreferenceKey(R.string.pref_key_tracking_protection_strict_default),
         false,
-    )
+    ) */
 
-    val useCustomTrackingProtection by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_tracking_protection_custom_option),
+    val useCustomTrackingProtection = false
+/*        appContext.getPreferenceKey(R.string.pref_key_tracking_protection_custom_option),
         false,
-    )
+    ) */
 
     var strictAllowListBaselineTrackingProtection by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_tracking_protection_strict_allow_list_baseline),
