diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingMapper.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingMapper.kt
index ba8f67c8749c..2e1bcc5719bf 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingMapper.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingMapper.kt
@@ -32,7 +32,7 @@ internal fun Collection<OnboardingCardData>.toPageUiData(
         .sortedBy { it.ordering }
         .mapIndexed { index, onboardingCardData ->
             // only first onboarding card shows privacy caption
-            onboardingCardData.toPageUiData(if (index == 0) privacyCaption else null)
+            onboardingCardData.toPageUiData(null)
         }
 }
 
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index c7db941b0f7e..c7038359e56c 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -587,7 +587,7 @@ class Settings(
 
     var hasAcceptedTermsOfService by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_terms_accepted),
-        default = false,
+        default = true,
         persistDefaultIfNotExists = true,
     )
 
@@ -623,7 +623,7 @@ class Settings(
     var isTermsOfUsePromptEnabled by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_terms_prompt_enabled),
         default = { FxNimbus.features.termsOfUsePrompt.value().enabled },
-        featureFlag = true,
+        featureFlag = false,
     )
 
     /**
@@ -632,7 +632,7 @@ class Settings(
     var shouldShowTermsOfUsePromptDragHandle by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_terms_prompt_drag_handle_enabled),
         default = { FxNimbus.features.termsOfUsePrompt.value().enableDragToDismiss },
-        featureFlag = true,
+        featureFlag = false,
     )
 
     /**
