diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/bookmarks/BookmarksScreen.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/bookmarks/BookmarksScreen.kt
index 58b970f69707..b04bd854df1f 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/bookmarks/BookmarksScreen.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/bookmarks/BookmarksScreen.kt
@@ -1017,7 +1017,7 @@ private fun EmptyList(
             when (state) {
                 is EmptyListState.NotAuthenticated -> RootEmptyContent(
                     dispatcher,
-                    showSignIn = true,
+                    showSignIn = false,
                     showBookmarksImport = showBookmarksImport,
                 )
                 EmptyListState.Authenticated -> RootEmptyContent(
