diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index c7db941b0f7e..e15d0bc7e8cd 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -1066,10 +1066,10 @@ class Settings(
         default = false,
     )
 
-    var shouldUseTrackingProtection by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_tracking_protection),
+    var shouldUseTrackingProtection = true
+/*        appContext.getPreferenceKey(R.string.pref_key_tracking_protection),
         default = true,
-    )
+    ) */
 
     var shouldEnableGlobalPrivacyControl by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_privacy_enable_global_privacy_control),
@@ -1164,20 +1164,20 @@ class Settings(
         default = false,
     )
 
-    val useStandardTrackingProtection by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_tracking_protection_standard_option),
+    val useStandardTrackingProtection = false
+/*        appContext.getPreferenceKey(R.string.pref_key_tracking_protection_standard_option),
         true,
-    )
+    ) */
 
-    val useStrictTrackingProtection by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_tracking_protection_strict_default),
+    val useStrictTrackingProtection = true
+/*        appContext.getPreferenceKey(R.string.pref_key_tracking_protection_strict_default),
         false,
-    )
+    ) */
 
-    val useCustomTrackingProtection by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_tracking_protection_custom_option),
+    val useCustomTrackingProtection = false
+/*        appContext.getPreferenceKey(R.string.pref_key_tracking_protection_custom_option),
         false,
-    )
+    ) */
 
     var strictAllowListBaselineTrackingProtection by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_tracking_protection_strict_allow_list_baseline),
