diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/redesign/view/TermsOfServiceOnboardingPageRedesign.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/redesign/view/TermsOfServiceOnboardingPageRedesign.kt
index de41d50564..b786835445 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/redesign/view/TermsOfServiceOnboardingPageRedesign.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/redesign/view/TermsOfServiceOnboardingPageRedesign.kt
@@ -131,7 +131,7 @@ private fun Subheaders(termsOfService: OnboardingTermsOfService) {
 @Composable
 private fun SubHeaderText(text: String) {
     Text(
-        text = text,
+        text = "",
         style = FirefoxTheme.typography.body2.copy(
             color = FirefoxTheme.colors.textSecondary,
             textAlign = TextAlign.Center,
@@ -147,9 +147,9 @@ private fun BodyText(
     with(termsOfService) {
         Column(modifier = Modifier.padding(horizontal = 8.dp)) {
             val bodyOneLinkState = LinkTextState(
-                text = lineOneLinkText,
+                text = "",
                 url = lineOneLinkUrl,
-                onClick = eventHandler::onTermsOfServiceLinkClicked,
+                onClick = {},
             )
             BodyLinkText(
                 lineOneText.updateFirstPlaceholder(lineOneLinkText),
@@ -157,9 +157,9 @@ private fun BodyText(
             )
 
             val bodyTwoLinkState = LinkTextState(
-                text = lineTwoLinkText,
+                text = "",
                 url = lineTwoLinkUrl,
-                onClick = eventHandler::onPrivacyNoticeLinkClicked,
+                onClick = {},
             )
             BodyLinkText(
                 lineTwoText.updateFirstPlaceholder(lineTwoLinkText),
@@ -167,9 +167,9 @@ private fun BodyText(
             )
 
             val bodyThreeLinkState = LinkTextState(
-                text = lineThreeLinkText,
+                text = "",
                 url = "", // No URL
-                onClick = { _ -> eventHandler.onManagePrivacyPreferencesLinkClicked() },
+                onClick = {},
             )
             BodyLinkText(
                 lineThreeText.updateFirstPlaceholder(lineThreeLinkText),
@@ -190,7 +190,7 @@ private fun BodyLinkText(
     )
 
     LinkText(
-        text = text,
+        text = "",
         linkTextStates = listOf(linkState),
         style = style,
         shouldApplyAccessibleSize = true,
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingMapper.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingMapper.kt
index 795fcd7b7e..1b4d2eef78 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingMapper.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingMapper.kt
@@ -32,7 +32,7 @@ internal fun Collection<OnboardingCardData>.toPageUiData(
         .sortedBy { it.ordering }
         .mapIndexed { index, onboardingCardData ->
             // only first onboarding card shows privacy caption
-            onboardingCardData.toPageUiData(if (index == 0) privacyCaption else null)
+            onboardingCardData.toPageUiData(null)
         }
 }
 
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/TermsOfServiceOnboardingPage.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/TermsOfServiceOnboardingPage.kt
index 92a8a9a36d..c67e70ba65 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/TermsOfServiceOnboardingPage.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/TermsOfServiceOnboardingPage.kt
@@ -135,22 +135,22 @@ private fun BodyText(
             modifier = Modifier.padding(horizontal = 10.dp),
         ) {
             val lineOneState = LinkTextState(
-                text = it.lineOneLinkText,
+                text = "",
                 url = it.lineOneLinkUrl,
-                onClick = eventHandler::onTermsOfServiceLinkClicked,
+                onClick = {},
             )
             val lineTwoState = LinkTextState(
-                text = it.lineTwoLinkText,
+                text = "",
                 url = it.lineTwoLinkUrl,
-                onClick = eventHandler::onPrivacyNoticeLinkClicked,
+                onClick = {},
             )
             val lineThreeState = LinkTextState(
-                text = it.lineThreeLinkText,
+                text = "",
                 url = "",
-                onClick = { _ -> eventHandler.onManagePrivacyPreferencesLinkClicked() },
+                onClick = {},
             )
             LinkText(
-                text = it.lineOneText.updateFirstPlaceholder(it.lineOneLinkText),
+                text = "",
                 linkTextStates = listOf(
                     lineOneState,
                 ),
@@ -162,7 +162,7 @@ private fun BodyText(
             )
 
             LinkText(
-                text = it.lineTwoText.updateFirstPlaceholder(it.lineTwoLinkText),
+                text = "",
                 linkTextStates = listOf(
                     lineTwoState,
                 ),
@@ -174,7 +174,7 @@ private fun BodyText(
             )
 
             LinkText(
-                text = it.lineThreeText.updateFirstPlaceholder(it.lineThreeLinkText),
+                text = "",
                 linkTextStates = listOf(
                     lineThreeState,
                 ),
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 312b5b3de3..3f93f28f4d 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -590,7 +590,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
 
     var hasAcceptedTermsOfService by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_terms_accepted),
-        default = false,
+        default = true,
         persistDefaultIfNotExists = true,
     )
 
@@ -600,7 +600,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     var isTermsOfUsePromptEnabled by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_terms_prompt_enabled),
         default = { FxNimbus.features.termsOfUsePrompt.value().enabled },
-        featureFlag = true,
+        featureFlag = false,
     )
 
     /**
