diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/TrackingProtectionFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/TrackingProtectionFragment.kt
index cd4ad113098b..828bc8b21559 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/TrackingProtectionFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/TrackingProtectionFragment.kt
@@ -26,6 +26,7 @@ import org.mozilla.fenix.ext.settings
 import org.mozilla.fenix.ext.showToolbar
 import org.mozilla.fenix.trackingprotection.TrackingProtectionMode
 import org.mozilla.fenix.utils.view.addToRadioGroup
+import org.ironfoxoss.ironfox.utils.FenixSettingsVisibilityManager
 
 /**
  * Displays the toggle for tracking protection, options for tracking protection policy and a button
@@ -91,6 +92,7 @@ class TrackingProtectionFragment : PreferenceFragmentCompat() {
         addToRadioGroup(radioStrict, radioStandard, radioCustom)
         updateStrictOptionsVisibility()
         updateCustomOptionsVisibility()
+        FenixSettingsVisibilityManager.TrackingProtectionFragment(requireContext(), this)
     }
 
     override fun onResume() {
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 35d4c5a9c0..e249d418ce 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -1016,10 +1016,10 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         default = false,
     )
 
-    var shouldUseTrackingProtection by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_tracking_protection),
+    var shouldUseTrackingProtection = true
+/*        appContext.getPreferenceKey(R.string.pref_key_tracking_protection),
         default = true,
-    )
+    ) */
 
     var shouldEnableGlobalPrivacyControl by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_privacy_enable_global_privacy_control),
@@ -1126,20 +1126,20 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         default = false,
     )
 
-    val useStandardTrackingProtection by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_tracking_protection_standard_option),
+    val useStandardTrackingProtection = false
+/*        appContext.getPreferenceKey(R.string.pref_key_tracking_protection_standard_option),
         true,
-    )
+    ) */
 
-    val useStrictTrackingProtection by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_tracking_protection_strict_default),
+    val useStrictTrackingProtection = true
+/*        appContext.getPreferenceKey(R.string.pref_key_tracking_protection_strict_default),
         false,
-    )
+    ) */
 
-    val useCustomTrackingProtection by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_tracking_protection_custom_option),
+    val useCustomTrackingProtection = false
+/*        appContext.getPreferenceKey(R.string.pref_key_tracking_protection_custom_option),
         false,
-    )
+    ) */
 
     var strictAllowListBaselineTrackingProtection by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_tracking_protection_strict_allow_list_baseline),
