diff --git a/mobile/android/fenix/.buildconfig.yml b/mobile/android/fenix/.buildconfig.yml
index 0620fba71e..36fad137e1 100644
--- a/mobile/android/fenix/.buildconfig.yml
+++ b/mobile/android/fenix/.buildconfig.yml
@@ -69,7 +69,7 @@ projects:
     - components:lib-state
     - components:service-digitalassetlinks
     - components:service-firefox-accounts
-    - components:service-glean
+#    - components:service-glean
     - components:service-location
     - components:service-mars
     - components:service-nimbus
diff --git a/mobile/android/fenix/app/build.gradle b/mobile/android/fenix/app/build.gradle
index 029784efad..8a978e887d 100644
--- a/mobile/android/fenix/app/build.gradle
+++ b/mobile/android/fenix/app/build.gradle
@@ -628,7 +628,7 @@ dependencies {
     implementation project(':components:service-sync-autofill')
     implementation project(':components:service-sync-logins')
     implementation project(':components:service-firefox-accounts')
-    implementation project(':components:service-glean')
+//    implementation project(':components:service-glean')
     implementation libs.mozilla.glean
     implementation project(':components:service-location')
     implementation project(':components:service-nimbus')
