diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
index b5cd2868a6..f2e135d4ec 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
@@ -180,7 +180,7 @@ open class DefaultToolbarMenu(
     @VisibleForTesting(otherwise = PRIVATE)
     fun shouldShowOpenInRegularTab(): Boolean = selectedSession?.let { session ->
         // This feature is gated behind Nightly for the time being.
-        Config.channel.isNightlyOrDebug &&
+        !context.settings().alwaysUsePrivateBrowsing &&
             // This feature is explicitly for users opening links in private tabs.
             context.settings().openLinksInAPrivateTab &&
             // and is only visible in private tabs.
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/PrivateBrowsingFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/PrivateBrowsingFragment.kt
index f0868c1821..50532aac0b 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/PrivateBrowsingFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/PrivateBrowsingFragment.kt
@@ -71,6 +71,7 @@ class PrivateBrowsingFragment : PreferenceFragmentCompat() {
         requirePreference<SwitchPreference>(R.string.pref_key_open_links_in_a_private_tab).apply {
             onPreferenceChangeListener = SharedPreferenceUpdater()
             isChecked = context.settings().openLinksInAPrivateTab
+            isVisible = !context.settings().alwaysUsePrivateBrowsing
         }
 
         requirePreference<SwitchPreference>(R.string.pref_key_allow_screenshots_in_private_mode).apply {
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 35d4c5a9c0..4ba8277f1c 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -1292,7 +1292,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
                 false,
             )
 
-            return if (lastKnownModeWasPrivate) {
+            return if (alwaysUsePrivateBrowsing || lastKnownModeWasPrivate) {
                 BrowsingMode.Private
             } else {
                 BrowsingMode.Normal
