diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt
index 935a71c0bb..df5862f2f0 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt
@@ -188,7 +188,7 @@ class BrowserFragment : BaseBrowserFragment(), UserInteractionHandler {
 
     private fun initTranslationsAction(context: Context, view: View) {
         if (
-            !FxNimbus.features.translations.value().mainFlowToolbarEnabled
+            !requireContext().settings().translationsEnabled
         ) {
             return
         }
@@ -271,7 +271,7 @@ class BrowserFragment : BaseBrowserFragment(), UserInteractionHandler {
     }
 
     private fun initTranslationsUpdates(context: Context, view: View) {
-        if (!FxNimbus.features.translations.value().mainFlowToolbarEnabled) return
+        if (!requireContext().settings().translationsEnabled) return
 
         translationsBinding.set(
             feature = TranslationsBinding(
