diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index c7db941b0f7e..eecd0c207c3b 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -903,7 +903,7 @@ class Settings(
     var isIsolatedProcessEnabled by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_enable_isolated_process),
         featureFlag = true,
-        default = { FxNimbus.features.isolatedContentProcesses.value().enabled },
+        default = { true },
     )
 
     /**
