diff --git a/mobile/android/fenix/app/src/main/AndroidManifest.xml b/mobile/android/fenix/app/src/main/AndroidManifest.xml
index 63398b2c71..1674b913c3 100644
--- a/mobile/android/fenix/app/src/main/AndroidManifest.xml
+++ b/mobile/android/fenix/app/src/main/AndroidManifest.xml
@@ -488,12 +488,12 @@
             </intent-filter>
         </activity-alias>
 
-        <activity
+       <!-- <activity
             android:name=".startupCrash.StartupCrashActivity"
             android:exported="false"
             android:process=":StartupCrashActivityProcess"
             >
-        </activity>
+        </activity> -->
 
         <activity
             android:name=".HomeActivity"
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/debugsettings/navigation/DebugDrawerRoute.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/debugsettings/navigation/DebugDrawerRoute.kt
index e53b2e6ad2..4665880a98 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/debugsettings/navigation/DebugDrawerRoute.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/debugsettings/navigation/DebugDrawerRoute.kt
@@ -70,10 +70,10 @@ enum class DebugDrawerRoute(val route: String, @param:StringRes val title: Int)
         route = "addons_debug_tools",
         title = R.string.debug_drawer_addons_tools_title,
     ),
-    CrashDebugTools(
+/*    CrashDebugTools(
         route = "crash_debug_tools",
         title = R.string.crash_debug_tools_title,
-    ),
+    ), */
     ;
 
     companion object {
@@ -189,14 +189,14 @@ enum class DebugDrawerRoute(val route: String, @param:StringRes val title: Int)
                         }
                     }
 
-                    CrashDebugTools -> {
+/*                    CrashDebugTools -> {
                         onClick = {
                             debugDrawerStore.dispatch(DebugDrawerAction.NavigateTo.CrashDebugTools)
                         }
                         content = {
                             CrashTools()
                         }
-                    }
+                    } */
                 }
 
                 DebugDrawerDestination(
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/debugsettings/store/DebugDrawerNavigationMiddleware.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/debugsettings/store/DebugDrawerNavigationMiddleware.kt
index 1e17f81bda..a5b7e6c84e 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/debugsettings/store/DebugDrawerNavigationMiddleware.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/debugsettings/store/DebugDrawerNavigationMiddleware.kt
@@ -50,7 +50,7 @@ class DebugDrawerNavigationMiddleware(
                 is DebugDrawerAction.NavigateTo.AddonsDebugTools ->
                     navController.navigate(route = DebugDrawerRoute.AddonsDebugTools.route)
                 is DebugDrawerAction.NavigateTo.CrashDebugTools ->
-                    navController.navigate(route = DebugDrawerRoute.CrashDebugTools.route)
+                    Unit // no-op
                 is DebugDrawerAction.OnBackPressed -> navController.popBackStack()
                 is DebugDrawerAction.DrawerOpened, DebugDrawerAction.DrawerClosed -> Unit // no-op
             }
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt
index 53190b359c..9f0defb030 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt
@@ -510,14 +510,14 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
     }
 
     private fun setupCrashReporting() {
-        components
+/*        components
             .analytics
             .crashReporter
-            .install(this, ::handleCaughtException)
+            .install(this, ::handleCaughtException) */
     }
 
     private fun handleCaughtException() {
-        if (
+/*        if (
             isMainProcess() &&
             Config.channel.isNightlyOrDebug &&
             !components.performance.visualCompletenessQueue.isReady()
@@ -526,7 +526,7 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
 
             intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
             applicationContext.startActivity(intent)
-        }
+        } */
     }
 
     protected open fun initializeNimbus() {
@@ -555,7 +555,7 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
     private fun beginSetupMegazord() {
         // Rust components must be initialized at the very beginning, before any other Rust call, ...
         AppServicesInitializer.init(
-            AppServicesConfig(components.analytics.crashReporter),
+            AppServicesConfig(),
         )
     }
 
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/BrowserToolbarSearchMiddleware.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/BrowserToolbarSearchMiddleware.kt
index 14f85cc900..a44dfc3b3a 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/BrowserToolbarSearchMiddleware.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/BrowserToolbarSearchMiddleware.kt
@@ -238,14 +238,14 @@ class BrowserToolbarSearchMiddleware(
                 val navController = environment?.navController ?: return
 
                 when (action.text) {
-                    "about:crashes" -> {
+/*                    "about:crashes" -> {
                         // The list of past crashes can be accessed via "settings > about", but desktop and
                         // fennec users may be used to navigating to "about:crashes". So we intercept this here
                         // and open the crash list activity instead.
                         navController.navigate(
                             NavGraphDirections.actionGlobalCrashListFragment(),
                         )
-                    }
+                    } */
                     "about:addons" -> {
                         navController.navigate(
                             NavGraphDirections.actionGlobalAddonsManagementFragment(),
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/SearchDialogController.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/SearchDialogController.kt
index b57cdc1bf5..163f38cbd7 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/SearchDialogController.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/SearchDialogController.kt
@@ -84,13 +84,13 @@ class SearchDialogController(
         }
 
         when (url) {
-            "about:crashes" -> {
+/*            "about:crashes" -> {
                 // The list of past crashes can be accessed via "settings > about", but desktop and
                 // fennec users may be used to navigating to "about:crashes". So we intercept this here
                 // and open the crash list activity instead.
                 val directions = SearchDialogFragmentDirections.actionCrashListFragment()
                 navController.navigate(directions)
-            }
+            } */
             "about:addons" -> {
                 val directions =
                     SearchDialogFragmentDirections.actionGlobalAddonsManagementFragment()
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
index 3c071edf51..172f1e3af9 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
@@ -246,7 +246,7 @@ class SecretSettingsFragment : PreferenceFragmentCompat() {
         }
 
         requirePreference<SwitchPreference>(R.string.pref_key_use_new_crash_reporter).apply {
-            isVisible = true
+            isVisible = false
             isChecked = context.settings().useNewCrashReporterDialog
             onPreferenceChangeListener =
                 Preference.OnPreferenceChangeListener { _, newValue ->
@@ -314,7 +314,7 @@ class SecretSettingsFragment : PreferenceFragmentCompat() {
         }
 
         requirePreference<SwitchPreference>(R.string.pref_key_crash_pull_never_show_again).apply {
-            isVisible = true
+            isVisible = false
             isChecked = context.settings().crashPullNeverShowAgain
             onPreferenceChangeListener = SharedPreferenceUpdater()
         }
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 35d4c5a9c0..10b92d81fa 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -538,7 +538,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
 
     var crashReportChoice by stringPreference(
         appContext.getPreferenceKey(R.string.pref_key_crash_reporting_choice),
-        default = CrashReportOption.Ask.toString(),
+        default = CrashReportOption.Never.toString(),
     )
 
     val isRemoteDebuggingEnabled by booleanPreference(
@@ -907,7 +907,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var crashPullNeverShowAgain: Boolean by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_crash_pull_never_show_again),
-        default = false,
+        default = true,
     )
 
     @VisibleForTesting
