diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt
index 53190b359c..78081d3dbe 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt
@@ -377,7 +377,7 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
                         // new search suggestions. The worker requires us to have called
                         // `GlobalFxSuggestDependencyProvider.initialize`, which we did before
                         // scheduling these tasks. When disabled we stop the periodic work.
-                        if (settings().enableFxSuggest) {
+                        if (settings().enableFxSuggest && settings().showNonSponsoredSuggestions) {
                             components.fxSuggest.ingestionScheduler.startPeriodicIngestion()
                         } else {
                             components.fxSuggest.ingestionScheduler.stopPeriodicIngestion()
@@ -466,7 +466,7 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
         queueStorageMaintenance()
         queueNimbusFetchInForeground()
         queueDownloadWallpapers()
-        if (settings().enableFxSuggest) {
+        if (settings().enableFxSuggest && settings().showNonSponsoredSuggestions) {
             queueSuggestIngest()
         }
         queueCollectProcessExitInfo()
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/awesomebar/SearchSuggestionsProvidersBuilder.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/awesomebar/SearchSuggestionsProvidersBuilder.kt
index 4115feeb4c..bfe0e3bcab 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/awesomebar/SearchSuggestionsProvidersBuilder.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/awesomebar/SearchSuggestionsProvidersBuilder.kt
@@ -78,7 +78,7 @@ class SearchSuggestionsProvidersBuilder(
                 components.core.icons,
                 engineForSpeculativeConnects,
                 showEditSuggestion = false,
-                suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
+//                suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
             )
 
         defaultCombinedHistoryProvider =
@@ -90,7 +90,7 @@ class SearchSuggestionsProvidersBuilder(
                 engine = engineForSpeculativeConnects,
                 maxNumberOfSuggestions = METADATA_SUGGESTION_LIMIT,
                 showEditSuggestion = false,
-                suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
+//                suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
             )
 
         val searchBitmap = suggestionIconProvider.getSearchIconBitmap()
@@ -301,7 +301,7 @@ class SearchSuggestionsProvidersBuilder(
                     engine = engineForSpeculativeConnects,
                     maxNumberOfSuggestions = METADATA_SUGGESTION_LIMIT,
                     showEditSuggestion = false,
-                    suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
+//                    suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
                     resultsUriFilter = filter::shouldIncludeUri,
                 )
             } else {
@@ -316,7 +316,7 @@ class SearchSuggestionsProvidersBuilder(
                     engine = engineForSpeculativeConnects,
                     maxNumberOfSuggestions = METADATA_SUGGESTION_LIMIT,
                     showEditSuggestion = false,
-                    suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
+//                    suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
                     resultsUriFilter = filter::shouldIncludeUri,
                 )
             } else {
@@ -428,7 +428,7 @@ class SearchSuggestionsProvidersBuilder(
                 suggestionIconProvider.getMobileIconDrawable(),
                 suggestionIconProvider.getTabletIconDrawable(),
             ),
-            suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
+//            suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
             resultsUrlFilter = filter?.let { it::shouldIncludeUrl },
         )
     }
@@ -450,7 +450,7 @@ class SearchSuggestionsProvidersBuilder(
             components.core.icons,
             suggestionIconProvider.getLocalTabIconDrawable(),
             excludeSelectedSession = !includeSelectedTab,
-            suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
+//            suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
             switchToTabDescription = suggestionsStringsProvider.getSwitchToTabDescriptionString(),
             resultsUriFilter = filter?.let { it::shouldIncludeUri },
         )
@@ -474,7 +474,7 @@ class SearchSuggestionsProvidersBuilder(
             indicatorIcon = suggestionIconProvider.getBookmarkIconDrawable(),
             engine = engineForSpeculativeConnects,
             showEditSuggestion = false,
-            suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
+//            suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
             resultsUriFilter = filter?.let { it::shouldIncludeUri },
         )
     }
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt
index c0fb12426e..dd229dcd18 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt
@@ -39,7 +39,7 @@ class SearchEngineFragment : PreferenceFragmentCompat() {
         )
 
         requirePreference<SwitchPreference>(R.string.pref_key_show_sponsored_suggestions).apply {
-            isVisible = context.settings().enableFxSuggest
+            isVisible = false
         }
         requirePreference<SwitchPreference>(R.string.pref_key_show_nonsponsored_suggestions).apply {
             isVisible = context.settings().enableFxSuggest
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 35d4c5a9c0..5baa4620a9 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -2348,7 +2348,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var enableFxSuggest by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_enable_fxsuggest),
-        default = { FxNimbus.features.fxSuggest.value().enabled },
+        default = { false },
         featureFlag = FeatureFlags.FX_SUGGEST,
     )
 
@@ -2373,7 +2373,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     var showSponsoredSuggestions by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_show_sponsored_suggestions),
         default = { enableFxSuggest },
-        featureFlag = FeatureFlags.FX_SUGGEST,
+        featureFlag = false,
     )
 
     /**
@@ -2383,7 +2383,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var showNonSponsoredSuggestions by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_show_nonsponsored_suggestions),
-        default = { enableFxSuggest },
+        default = { false },
         featureFlag = FeatureFlags.FX_SUGGEST,
     )
 
