From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sun, 05 Jul 2026 05:15:52 +0000
Subject: [PATCH] Fenix - IronFox settings search

Allows users to search for IronFox-specific UI settings.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/settingssearch/DefaultFenixSettingsIndexer.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/settingssearch/DefaultFenixSettingsIndexer.kt
index 1edff31a4346..35e2b8f20f34 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/settingssearch/DefaultFenixSettingsIndexer.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/settingssearch/DefaultFenixSettingsIndexer.kt
@@ -366,6 +366,7 @@ class DefaultFenixSettingsIndexer(
                 PreferenceFileInformation.TabsPreferences,
                 PreferenceFileInformation.TrackingProtectionPreferences,
                 PreferenceFileInformation.SaveLoginsPreferences,
+                PreferenceFileInformation.IronFoxPreferences,
             )
 
         /**
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/settingssearch/PreferenceFileInformation.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/settingssearch/PreferenceFileInformation.kt
index 08049f328a85..b4a48df5c8c0 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/settingssearch/PreferenceFileInformation.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/settingssearch/PreferenceFileInformation.kt
@@ -5,6 +5,7 @@
 package org.mozilla.fenix.settings.settingssearch
 
 import org.mozilla.fenix.R
+import org.ironfoxoss.core.R as ironfoxR
 
 /**
  * Data class for a settings search item navigation information based on the xml file it comes from.
@@ -202,4 +203,12 @@ sealed class PreferenceFileInformation(
             categoryHeaderResourceId = R.string.preferences_category_advanced,
             fragmentId = R.id.firefoxLabsFragment,
         )
+
+    object IronFoxPreferences :
+        PreferenceFileInformation(
+            xmlResourceId = R.xml.ironfox_preferences,
+            topBreadcrumbResourceId = ironfoxR.string.if_preferences,
+            categoryHeaderResourceId = ironfoxR.string.if_title,
+            fragmentId = R.id.ironFoxSettingsFragment,
+        )
 }
-- 
