diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt
index 2be5597f36..22ffd0bdfb 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt
@@ -195,6 +195,38 @@ class Core(
             bannedPorts = FxNimbus.features.networkingBannedPorts.value().bannedPortList,
             lnaBlockingEnabled = context.settings().isLnaBlockingEnabled,
             crliteChannel = FxNimbus.features.pki.value().crliteChannel,
+            accessibilityEnabled = context.settings().accessibilityEnabled,
+            addressAutofillEnabled = context.settings().shouldAutofillAddressDetails,
+            autoplayBlockingPolicy = context.settings().getAutoplayBlockingPolicy(),
+            cacheEnabled = context.settings().cacheEnabled,
+            cardAutofillEnabled = context.settings().shouldAutofillCreditCardDetails,
+            emeEnabled = context.settings().emeEnabled,
+            fppOverridesIronFoxEnabled = context.settings().fppOverridesIronFoxEnabled,
+            fppOverridesIronFoxTimezoneEnabled = context.settings().fppOverridesIronFoxTimezoneEnabled,
+            fppOverridesIronFoxWebGLEnabled = context.settings().fppOverridesIronFoxWebGLEnabled,
+            fppOverridesMozillaEnabled = context.settings().fppOverridesMozillaEnabled,
+            ipv6Enabled = context.settings().ipv6Enabled,
+            javascriptEnabled = context.settings().javascriptEnabled,
+            javascriptJitBaselineEnabled = context.settings().javascriptJitEnabled,
+            javascriptJitHintsEnabled = context.settings().javascriptJitEnabled,
+            javascriptJitIonEnabled = context.settings().javascriptJitEnabled,
+            javascriptJitIonWasmEnabled = context.settings().javascriptJitEnabled,
+            javascriptJitNativeRegexpEnabled = context.settings().javascriptJitEnabled,
+            javascriptJitTrustedPrincipalsEnabled = context.settings().javascriptJitTrustedPrincipalsEnabled,
+            passwordManagerEnabled = context.settings().shouldPromptToSaveLogins,
+            pdfjsDisabled = context.settings().pdfjsDisabled,
+            printEnabled = context.settings().printEnabled,
+            refererXOriginPolicy = context.settings().getRefererXOriginPolicy(),
+            spoofEnglish = context.settings().spoofEnglish,
+            spoofTimezone = context.settings().spoofTimezone,
+            svgEnabled = context.settings().svgEnabled,
+            translationsEnabled = context.settings().translationsEnabled,
+            translationsSupported = context.settings().translationsEnabled,
+            wasmEnabled = context.settings().wasmEnabled,
+            webglDisabled = context.settings().webglDisabled,
+            webrtcEnabled = context.settings().webrtcEnabled,
+            widevineEnabled = context.settings().widevineEnabled,
+            xpinstallEnabled = context.settings().xpinstallEnabled,
         )
 
         // Apply fingerprinting protection overrides if the feature is enabled in Nimbus
@@ -228,6 +260,7 @@ class Core(
             defaultSettings,
             geckoRuntime,
         ).also {
+            GeckoProvider.setupSafeBrowsing(it, context.settings().safeBrowsingEnabled)
             WebCompatFeature.install(it)
         }
     }
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
index e1a67a6da4..a98b4656aa 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
@@ -591,6 +591,7 @@ class SettingsFragment : PreferenceFragmentCompat() {
         setupHomepagePreference(settings)
         setupTrackingProtectionPreference(settings)
         setupDnsOverHttpsPreference(settings)
+        setupIronFoxPreferences()
     }
 
     /**
@@ -789,6 +790,19 @@ class SettingsFragment : PreferenceFragmentCompat() {
             }
     }
 
+    @VisibleForTesting
+    internal fun setupIronFoxPreferences() {
+        requirePreference<Preference>(R.string.pref_key_if_preferences).apply {
+            setOnPreferenceClickListener {
+                val directions =
+                    SettingsFragmentDirections
+                        .actionSettingsFragmentToIronFoxSettingsFragment()
+                requireView().findNavController().navigate(directions)
+                true
+            }
+        }
+    }
+
     private fun updateProfilerUI(profilerStatus: Boolean) {
         if (profilerStatus) {
             findPreference<Preference>(getPreferenceKey(R.string.pref_key_start_profiler))?.title =
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 1c7b0fce0d..f27ee6b414 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -75,6 +75,8 @@ import org.mozilla.fenix.wallpapers.Wallpaper
 import java.security.InvalidParameterException
 import java.util.UUID
 import java.util.concurrent.TimeUnit.MILLISECONDS
+import mozilla.components.concept.engine.Engine.AutoplayBlockingPolicy
+import mozilla.components.concept.engine.Engine.RefererXOriginPolicy
 
 private const val AUTOPLAY_USER_SETTING = "AUTOPLAY_USER_SETTING"
 
@@ -2736,4 +2738,198 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         val cleanupPreferenceKey = appContext.getString(R.string.pref_key_downloads_clean_up_files_automatically)
         return sharedPreferences.getBoolean(cleanupPreferenceKey, false)
     }
+
+    var accessibilityEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_accessibility_enabled),
+        default = false,
+    )
+
+    var autoplayBlockingClickToPlay by booleanPreference(
+        key = appContext.getPreferenceKey(R.string.pref_key_autoplay_policy_click_to_play),
+        default = false,
+    )
+
+    var autoplayBlockingSticky by booleanPreference(
+        key = appContext.getPreferenceKey(R.string.pref_key_autoplay_policy_sticky),
+        default = false,
+    )
+
+    var autoplayBlockingTransient by booleanPreference(
+        key = appContext.getPreferenceKey(R.string.pref_key_autoplay_policy_transient),
+        default = true,
+    )
+
+    var cacheEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_cache_enabled),
+        default = false,
+    )
+
+    var emeEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_eme_enabled),
+        default = false,
+    )
+
+    var enableUnifiedPush by booleanPreference(
+        key = appContext.getPreferenceKey(R.string.pref_key_enable_unifiedpush),
+        default = true,
+    )
+
+    var fppOverridesIronFoxEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_fpp_overrides_ironfox_enabled),
+        default = true,
+    )
+
+    var fppOverridesIronFoxTimezoneEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_fpp_overrides_ironfox_timezone_enabled),
+        default = true,
+    )
+
+    var fppOverridesIronFoxWebGLEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_fpp_overrides_ironfox_webgl_enabled),
+        default = true,
+    )
+
+    var fppOverridesMozillaEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_fpp_overrides_mozilla_enabled),
+        default = true,
+    )
+
+    var ipv6Enabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_ipv6_enabled),
+        default = true,
+    )
+
+    var javascriptEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_javascript_enabled),
+        default = true,
+    )
+
+    var javascriptJitEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_jit_enabled),
+        default = false,
+    )
+
+    var javascriptJitTrustedPrincipalsEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_jit_trusted_principals_enabled),
+        default = false,
+    )
+
+    var pdfjsDisabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_pdfjs_disabled),
+        default = false,
+    )
+
+    var prefersBrowserColorScheme by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_prefers_browser_color_scheme),
+        default = false,
+    )
+
+    var prefersDarkColorScheme by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_prefers_dark_color_scheme),
+        default = false,
+    )
+
+    var prefersLightColorScheme by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_prefers_light_color_scheme),
+        default = true,
+    )
+
+    var printEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_print_enabled),
+        default = true,
+    )
+
+    var refererXOriginAlways by booleanPreference(
+        key = appContext.getPreferenceKey(R.string.pref_key_referer_policy_always),
+        default = true,
+    )
+
+    var refererXOriginBaseDomainsMatch by booleanPreference(
+        key = appContext.getPreferenceKey(R.string.pref_key_referer_policy_base_domains_match),
+        default = false,
+    )
+
+    var refererXOriginHostsMatch by booleanPreference(
+        key = appContext.getPreferenceKey(R.string.pref_key_referer_policy_hosts_match),
+        default = false,
+    )
+
+    var safeBrowsingEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_safe_browsing_enabled),
+        default = true,
+    )
+
+    var spoofEnglish by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_spoof_english),
+        default = true,
+    )
+
+    var spoofTimezone by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_spoof_timezone),
+        default = false,
+    )
+
+    var svgEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_svg_enabled),
+        default = true,
+    )
+
+    var translationsEnabled by booleanPreference(
+        key = appContext.getPreferenceKey(R.string.pref_key_translations_enabled),
+        default = true,
+    )
+
+    var useUnifiedPush by booleanPreference(
+        key = appContext.getPreferenceKey(R.string.pref_key_use_unifiedpush),
+        default = false,
+    )
+
+    var wasmEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_wasm_enabled),
+        default = true,
+    )
+
+    var webglDisabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_webgl_disabled),
+        default = true,
+    )
+
+    var webrtcEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_webrtc_enabled),
+        default = true,
+    )
+
+    var widevineEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_widevine_enabled),
+        default = false,
+    )
+
+    var xpinstallEnabled by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_xpinstall_enabled),
+        default = false,
+    )
+
+    fun getAutoplayBlockingPolicy(): AutoplayBlockingPolicy {
+        return if (autoplayBlockingSticky) {
+            AutoplayBlockingPolicy.AUTOPLAY_BLOCKING_STICKY
+        } else if (autoplayBlockingTransient) {
+            AutoplayBlockingPolicy.AUTOPLAY_BLOCKING_TRANSIENT
+        } else if (autoplayBlockingClickToPlay) {
+            AutoplayBlockingPolicy.AUTOPLAY_BLOCKING_CLICKTOPLAY
+        } else {
+            AutoplayBlockingPolicy.AUTOPLAY_BLOCKING_TRANSIENT
+        }
+    }
+
+    fun getRefererXOriginPolicy(): RefererXOriginPolicy {
+        return if (refererXOriginAlways) {
+            RefererXOriginPolicy.REFERER_XORIGIN_ALWAYS
+        } else if (refererXOriginBaseDomainsMatch) {
+            RefererXOriginPolicy.REFERER_XORIGIN_IF_BASE_DOMAINS_MATCH
+        } else if (refererXOriginHostsMatch) {
+            RefererXOriginPolicy.REFERER_XORIGIN_IF_HOSTS_MATCH
+        } else {
+            RefererXOriginPolicy.REFERER_XORIGIN_ALWAYS
+        }
+    }
 }
diff --git a/mobile/android/fenix/app/src/main/res/navigation/nav_graph.xml b/mobile/android/fenix/app/src/main/res/navigation/nav_graph.xml
index a4fd0897fa..634f9c1b8c 100644
--- a/mobile/android/fenix/app/src/main/res/navigation/nav_graph.xml
+++ b/mobile/android/fenix/app/src/main/res/navigation/nav_graph.xml
@@ -676,6 +676,13 @@
             app:exitAnim="@anim/slide_out_left"
             app:popEnterAnim="@anim/slide_in_left"
             app:popExitAnim="@anim/slide_out_right" />
+        <action
+            android:id="@+id/action_settingsFragment_to_ironFoxSettingsFragment"
+            app:destination="@id/ironFoxSettingsFragment"
+            app:enterAnim="@anim/slide_in_right"
+            app:exitAnim="@anim/slide_out_left"
+            app:popEnterAnim="@anim/slide_in_left"
+            app:popExitAnim="@anim/slide_out_right" />
         <action
             android:id="@+id/action_settingsFragment_to_trackingProtectionFragment"
             app:destination="@id/trackingProtectionFragment"
@@ -947,6 +954,10 @@
         android:id="@+id/sponsoredStoriesSettings"
         android:name="org.mozilla.fenix.settings.SponsoredStoriesSettingsFragment"
         android:label="@string/preferences_debug_settings_custom_sponsored_stories_parameters" />
+    <fragment
+        android:id="@+id/ironFoxSettingsFragment"
+        android:name="org.mozilla.fenix.settings.IronFoxSettingsFragment"
+        android:label="@string/if_preferences" />
     <fragment
         android:id="@+id/trackingProtectionFragment"
         android:name="org.mozilla.fenix.settings.TrackingProtectionFragment">
diff --git a/mobile/android/fenix/app/src/main/res/xml/preferences.xml b/mobile/android/fenix/app/src/main/res/xml/preferences.xml
index 5058f40c0b..a040bb1643 100644
--- a/mobile/android/fenix/app/src/main/res/xml/preferences.xml
+++ b/mobile/android/fenix/app/src/main/res/xml/preferences.xml
@@ -198,10 +198,22 @@
         <androidx.preference.SwitchPreference
             android:defaultValue="false"
             android:key="@string/pref_key_enable_gecko_logs"
+            android:summary="@string/restart_required_summary"
             app:iconSpaceReserved="false"
             android:title="@string/preferences_enable_gecko_logs" />
     </PreferenceCategory>
 
+    <androidx.preference.PreferenceCategory
+        android:layout="@layout/preference_category_no_icon_style"
+        android:selectable="false"
+        android:title="@string/if_title"
+        app:iconSpaceReserved="false">
+        <androidx.preference.Preference
+            android:key="@string/pref_key_if_preferences"
+            android:title="@string/if_preferences"
+            app:iconSpaceReserved="false" />
+    </androidx.preference.PreferenceCategory>
+
     <androidx.preference.PreferenceCategory
         android:title="@string/preferences_category_about"
         app:iconSpaceReserved="false"
