diff --git a/dom/canvas/ClientWebGLContext.cpp b/dom/canvas/ClientWebGLContext.cpp
index 6f6f38acbc..d0f4b16c22 100644
--- a/dom/canvas/ClientWebGLContext.cpp
+++ b/dom/canvas/ClientWebGLContext.cpp
@@ -880,6 +880,10 @@ bool ClientWebGLContext::CreateHostContext(const uvec2& requestedSize) {
         .options = options,
     };
 
+    if (ShouldResistFingerprinting(RFPTarget::DisableWebGL)) {
+      return Err("WebGL is currently disabled.");
+    }
+
     // -
 
     auto useOop = IsWebglOutOfProcessEnabled();
diff --git a/dom/canvas/WebGLContext.cpp b/dom/canvas/WebGLContext.cpp
index d672c003c1..0e7205f1a0 100644
--- a/dom/canvas/WebGLContext.cpp
+++ b/dom/canvas/WebGLContext.cpp
@@ -533,7 +533,7 @@ RefPtr<WebGLContext> WebGLContext::Create(HostWebGLContext* host,
   ScopedGfxFeatureReporter reporter("WebGL", forceEnabled);
 
   auto res = [&]() -> Result<RefPtr<WebGLContext>, std::string> {
-    bool disabled = StaticPrefs::webgl_disabled();
+    bool disabled = false;
 
     // TODO: When we have software webgl support we should use that instead.
     disabled |= gfxPlatform::InSafeMode();
diff --git a/toolkit/components/resistfingerprinting/RFPTargets.inc b/toolkit/components/resistfingerprinting/RFPTargets.inc
index 62062e9ae4..478db44f89 100644
--- a/toolkit/components/resistfingerprinting/RFPTargets.inc
+++ b/toolkit/components/resistfingerprinting/RFPTargets.inc
@@ -108,7 +108,7 @@ ITEM_VALUE(MaxTouchPointsCollapse,      73)
 ITEM_VALUE(NavigatorHWConcurrencyTiered,74)
 ITEM_VALUE(WebGLRandomization,          75)
 ITEM_VALUE(EfficientCanvasRandomization, 76)
-
+ITEM_VALUE(DisableWebGL,77)
 
 /*
  * In certain cases, we precompute the value of ShouldRFP for e.g. a Document.
