diff --git a/toolkit/components/resistfingerprinting/nsRFPService.cpp b/toolkit/components/resistfingerprinting/nsRFPService.cpp
index 9554fbb21b..ed93e4e1de 100644
--- a/toolkit/components/resistfingerprinting/nsRFPService.cpp
+++ b/toolkit/components/resistfingerprinting/nsRFPService.cpp
@@ -351,6 +351,12 @@ bool nsRFPService::IsRFPEnabledFor(
     return aOverriddenFingerprintingSettings.ref().contains(aTarget);
   }
 
+  if (aTarget == RFPTarget::DisableWebGL) {
+    return Preferences::GetBool("browser.ironfox.webgl.disabled") && IsTargetActiveForMode(aTarget, mode);
+  }
+  if (aTarget == RFPTarget::JSDateTimeUTC) {
+    return Preferences::GetBool("browser.ironfox.fingerprintingProtection.timezoneSpoofing.enabled") && IsTargetActiveForMode(aTarget, mode);
+  }
   return IsTargetActiveForMode(aTarget, mode);
 }
 
