diff --git a/mobile/android/geckoview/build.gradle b/mobile/android/geckoview/build.gradle
index 9f2993090f..7b93871e74 100644
--- a/mobile/android/geckoview/build.gradle
+++ b/mobile/android/geckoview/build.gradle
@@ -218,7 +218,7 @@ dependencies {
     implementation libs.androidx.collection
     implementation libs.androidx.core
 
-    implementation libs.play.services.fido
+    implementation "org.microg.gms:play-services-fido:v0.0.0.250932"
     implementation "org.yaml:snakeyaml:2.2"
 
     implementation libs.androidx.lifecycle.common
