diff --git a/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/RuntimeSettings.java b/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/RuntimeSettings.java
index dcd5d39ad9a3b..b1791288f1759 100644
--- a/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/RuntimeSettings.java
+++ b/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/RuntimeSettings.java
@@ -106,7 +106,7 @@ public abstract class RuntimeSettings implements Parcelable {
      */
     public void commit(final T newValue) {
       if (newValue.equals(mValue)) {
-        return;
+//        return;
       }
       set(newValue);
       commit();
diff --git a/mobile/shared/components/geckoview/GeckoViewPreferences.sys.mjs b/mobile/shared/components/geckoview/GeckoViewPreferences.sys.mjs
index a795b36e6a8f8e..191d50f53ec960 100644
--- a/mobile/shared/components/geckoview/GeckoViewPreferences.sys.mjs
+++ b/mobile/shared/components/geckoview/GeckoViewPreferences.sys.mjs
@@ -7,6 +7,7 @@ import { GeckoViewUtils } from "resource://gre/modules/GeckoViewUtils.sys.mjs";
 const lazy = {};
 ChromeUtils.defineESModuleGetters(lazy, {
   EventDispatcher: "resource://gre/modules/Messaging.sys.mjs",
+  GeckoSettingsBridge: "resource://gre/modules/GeckoSettingsBridge.sys.mjs",
 });
 
 const { PREF_STRING, PREF_BOOL, PREF_INT } = Ci.nsIPrefBranch;
@@ -38,7 +39,8 @@ export const GeckoViewPreferences = {
               pref.pref,
               pref.type,
               pref.value,
-              pref.branch
+              pref.branch,
+              "GSB"
             ),
           })),
         });
@@ -123,7 +125,7 @@ export const GeckoViewPreferences = {
    * @param {"user" | "default"} setBranch - The branch to operate on.
    * @return {boolean} Will return true if the preference set as requested or else false.
    */
-  setPreference(pref, type, value, setBranch) {
+  setPreference(pref, type, value, setBranch, method) {
     const branch =
       setBranch == "user"
         ? Services.prefs
@@ -131,15 +133,42 @@ export const GeckoViewPreferences = {
     try {
       switch (type) {
         case PREF_STRING:
-          branch.setStringPref(pref, value);
+          if (method === "GSB" && setBranch !== "user") {
+            lazy.GeckoSettingsBridge.setPref(pref, value, "string");
+            return true;
+          };
+          if (branch.getStringPref(pref) !== value) {
+            branch.setStringPref(pref, value);
+          };
+          if (Services.prefs.prefIsLocked(pref) !== true) {
+            Services.prefs.lockPref(pref);
+          };
           return true;
 
         case PREF_BOOL:
-          branch.setBoolPref(pref, value);
+          if (method === "GSB" && setBranch !== "user") {
+            lazy.GeckoSettingsBridge.setPref(pref, value, "boolean");
+            return true;
+          };
+          if (branch.getBoolPref(pref) !== value) {
+            branch.setBoolPref(pref, value);
+          };
+          if (Services.prefs.prefIsLocked(pref) !== true) {
+            Services.prefs.lockPref(pref);
+          };
           return true;
 
         case PREF_INT:
-          branch.setIntPref(pref, value);
+          if (method === "GSB" && setBranch !== "user") {
+            lazy.GeckoSettingsBridge.setPref(pref, value, "int");
+            return true;
+          };
+          if (branch.getIntPref(pref) !== value) {
+            branch.setIntPref(pref, value);
+          };
+          if (Services.prefs.prefIsLocked(pref) !== true) {
+            Services.prefs.lockPref(pref);
+          };
           return true;
 
         default:
diff --git a/mobile/shared/components/geckoview/GeckoViewStartup.sys.mjs b/mobile/shared/components/geckoview/GeckoViewStartup.sys.mjs
index 83f98bb7196ab1..f06ae5912c64dd 100644
--- a/mobile/shared/components/geckoview/GeckoViewStartup.sys.mjs
+++ b/mobile/shared/components/geckoview/GeckoViewStartup.sys.mjs
@@ -347,23 +347,115 @@ export class GeckoViewStartup {
       }
       case "GeckoView:ResetUserPrefs": {
         for (const name of aData.names) {
-          Services.prefs.clearUserPref(name);
+//          Services.prefs.clearUserPref(name);
         }
         break;
       }
       case "GeckoView:SetDefaultPrefs": {
         const prefs = Services.prefs.getDefaultBranch("");
+        const ifPrefix = "browser.ironfox.geckoRuntimeSettings.";
+        const ifModified = ".modified";
         for (const [name, value] of Object.entries(aData)) {
+          if (prefs.getBoolPref("browser.ironfox.geckoSettingsBridge.initialized") !== true) {
+            if (Services.prefs.prefIsLocked(name) === true) {
+                // If a preference is locked at this point, it means it was locked from ironfox.cfg (or ironfox.js) - so we don't want to try
+                /// to override it
+                break;
+            };
+          } else if (Services.prefs.prefIsLocked(name) === true) {
+            Services.prefs.unlockPref(name);
+          };
           try {
             switch (typeof value) {
               case "string":
-                prefs.setStringPref(name, value);
+                if (prefs.getStringPref(name) !== value) {
+                    if (Services.prefs.prefIsLocked(`${ifPrefix}${name}`) === true) {
+                        Services.prefs.unlockPref(`${ifPrefix}${name}`);
+                    };
+                    Services.prefs.setStringPref(`${ifPrefix}${name}`, value);
+                    prefs.setStringPref(`${ifPrefix}${name}`, value);
+                    if (Services.prefs.prefIsLocked(name) === true) {
+                        Services.prefs.unlockPref(name);
+                    };
+                    Services.prefs.setStringPref(name, value);
+                    prefs.setStringPref(name, value);
+                    if (Services.prefs.prefIsLocked(`${ifPrefix}${name}${ifModified}`) === true) {
+                        Services.prefs.unlockPref(`${ifPrefix}${name}${ifModified}`);
+                    };
+                    if (Services.prefs.getBoolPref(`${ifPrefix}${name}${ifModified}`) !== true) {
+                        Services.prefs.setBoolPref(`${ifPrefix}${name}${ifModified}`, true);
+                        prefs.setBoolPref(`${ifPrefix}${name}${ifModified}`, true);
+                    };
+                };
+                if (Services.prefs.prefIsLocked(`${ifPrefix}${name}`) !== true) {
+                    Services.prefs.lockPref(`${ifPrefix}${name}`);
+                };
+                if (Services.prefs.prefIsLocked(name) !== true) {
+                    Services.prefs.lockPref(name);
+                };
+                if (Services.prefs.prefIsLocked(`${ifPrefix}${name}${ifModified}`) !== true) {
+                    Services.prefs.lockPref(`${ifPrefix}${name}${ifModified}`);
+                };
                 break;
               case "number":
-                prefs.setIntPref(name, value);
+                if (prefs.getIntPref(name) !== value) {
+                    if (Services.prefs.prefIsLocked(`${ifPrefix}${name}`) === true) {
+                        Services.prefs.unlockPref(`${ifPrefix}${name}`);
+                    };
+                    Services.prefs.setIntPref(`${ifPrefix}${name}`, value);
+                    prefs.setIntPref(`${ifPrefix}${name}`, value);
+                    if (Services.prefs.prefIsLocked(name) === true) {
+                        Services.prefs.unlockPref(name);
+                    };
+                    Services.prefs.setIntPref(name, value);
+                    prefs.setIntPref(name, value);
+                    if (Services.prefs.prefIsLocked(`${ifPrefix}${name}${ifModified}`) === true) {
+                        Services.prefs.unlockPref(`${ifPrefix}${name}${ifModified}`);
+                    };
+                    if (Services.prefs.getBoolPref(`${ifPrefix}${name}${ifModified}`) !== true) {
+                        Services.prefs.setBoolPref(`${ifPrefix}${name}${ifModified}`, true);
+                        prefs.setBoolPref(`${ifPrefix}${name}${ifModified}`, true);
+                    };
+                };
+                if (Services.prefs.prefIsLocked(`${ifPrefix}${name}`) !== true) {
+                    Services.prefs.lockPref(`${ifPrefix}${name}`);
+                };
+                if (Services.prefs.prefIsLocked(name) !== true) {
+                    Services.prefs.lockPref(name);
+                };
+                if (Services.prefs.prefIsLocked(`${ifPrefix}${name}${ifModified}`) !== true) {
+                    Services.prefs.lockPref(`${ifPrefix}${name}${ifModified}`);
+                };
                 break;
               case "boolean":
-                prefs.setBoolPref(name, value);
+                if (prefs.getBoolPref(name) !== value) {
+                    if (Services.prefs.prefIsLocked(`${ifPrefix}${name}`) === true) {
+                        Services.prefs.unlockPref(`${ifPrefix}${name}`);
+                    };
+                    Services.prefs.setBoolPref(`${ifPrefix}${name}`, value);
+                    prefs.setBoolPref(`${ifPrefix}${name}`, value);
+                    if (Services.prefs.prefIsLocked(name) === true) {
+                        Services.prefs.unlockPref(name);
+                    };
+                    Services.prefs.setBoolPref(name, value);
+                    prefs.setBoolPref(name, value);
+                    if (Services.prefs.prefIsLocked(`${ifPrefix}${name}${ifModified}`) === true) {
+                        Services.prefs.unlockPref(`${ifPrefix}${name}${ifModified}`);
+                    };
+                    if (Services.prefs.getBoolPref(`${ifPrefix}${name}${ifModified}`) !== true) {
+                        Services.prefs.setBoolPref(`${ifPrefix}${name}${ifModified}`, true);
+                        prefs.setBoolPref(`${ifPrefix}${name}${ifModified}`, true);
+                    };
+                };
+                if (Services.prefs.prefIsLocked(`${ifPrefix}${name}`) !== true) {
+                    Services.prefs.lockPref(`${ifPrefix}${name}`);
+                };
+                if (Services.prefs.prefIsLocked(name) !== true) {
+                    Services.prefs.lockPref(name);
+                };
+                if (Services.prefs.prefIsLocked(`${ifPrefix}${name}${ifModified}`) !== true) {
+                    Services.prefs.lockPref(`${ifPrefix}${name}${ifModified}`);
+                };
                 break;
               default:
                 throw new Error(
