diff --git a/mfbt/SmallPointerArray.h b/mfbt/SmallPointerArray.h
index c63e3980f9..061bb1fc8b 100644
--- a/mfbt/SmallPointerArray.h
+++ b/mfbt/SmallPointerArray.h
@@ -201,9 +201,9 @@ class SmallPointerArray {
  private:
   T** beginInternal() const {
     if (first()) {
-      static_assert(sizeof(T*) == sizeof(Element),
+      static_assert(sizeof(T*) == sizeof(SmallArrayElement),
                     "pointer ops on &first() must produce adjacent "
-                    "Element::mValue arms");
+                    "SmallArrayElement::mValue arms");
       return &first();
     }
 
@@ -259,7 +259,7 @@ class SmallPointerArray {
   // When !mArray[0].mValue && mArray[1].mVector, mArray[1].mVector contains
   // the contents of an array of arbitrary size (even less than two if it ever
   // contained three elements and elements were removed).
-  union Element {
+  union SmallArrayElement {
     T* mValue;
     std::vector<T*>* mVector;
   } mArray[2];
