diff --git a/mobile/android/gradle/plugins/nimbus-gradle-plugin/src/main/kotlin/org/mozilla/appservices/tooling/nimbus/NimbusGradlePlugin.kt b/mobile/android/gradle/plugins/nimbus-gradle-plugin/src/main/kotlin/org/mozilla/appservices/tooling/nimbus/NimbusGradlePlugin.kt
index fcb3b4b682..cb1b14ab73 100644
--- a/mobile/android/gradle/plugins/nimbus-gradle-plugin/src/main/kotlin/org/mozilla/appservices/tooling/nimbus/NimbusGradlePlugin.kt
+++ b/mobile/android/gradle/plugins/nimbus-gradle-plugin/src/main/kotlin/org/mozilla/appservices/tooling/nimbus/NimbusGradlePlugin.kt
@@ -118,7 +118,7 @@ class NimbusPlugin : Plugin<Project> {
         @Suppress("UNCHECKED_CAST")
         val mozconfigSubsts = mozconfig?.get("substs") as? Map<String, Any>
 
-        if (mozconfigSubsts?.get("MOZ_APPSERVICES_IN_TREE").isTruthy()) {
+        if (true) {
             // we assume the binary has been built and `gradle.ext.mozconfig.substs.NIMBUS_FML` tells us where to find it.
             val nimbusFmlPath = requireNotNull(mozconfigSubsts?.get("NIMBUS_FML") as? String) {
                 "NIMBUS_FML not found in mozconfig substs"
