diff --git a/dom/base/Navigator.cpp b/dom/base/Navigator.cpp
index 11a57adbb5..71a8762dd5 100644
--- a/dom/base/Navigator.cpp
+++ b/dom/base/Navigator.cpp
@@ -1269,7 +1269,7 @@ bool Navigator::SendBeaconInternal(const nsAString& aUrl,
   } else {
     securityFlags |= nsILoadInfo::SEC_ALLOW_CROSS_ORIGIN_INHERITS_SEC_CONTEXT;
   }
-
+  return true;
   nsCOMPtr<nsIChannel> channel;
   rv = NS_NewChannel(getter_AddRefs(channel), uri, doc, securityFlags,
                      nsIContentPolicy::TYPE_BEACON);
