diff --git a/toolkit/components/resistfingerprinting/nsRFPService.cpp b/toolkit/components/resistfingerprinting/nsRFPService.cpp
index 5a422395f5..bd0ffc72e2 100644
--- a/toolkit/components/resistfingerprinting/nsRFPService.cpp
+++ b/toolkit/components/resistfingerprinting/nsRFPService.cpp
@@ -359,6 +359,10 @@ bool nsRFPService::IsRFPEnabledFor(
     return aOverriddenFingerprintingSettings.ref().contains(aTarget);
   }
 
+  if (aTarget == RFPTarget::JSDateTimeUTC) {
+    return Preferences::GetBool("browser.ironfox.fingerprintingProtection.timezoneSpoofing.enabled", false) && IsTargetActiveForMode(aTarget, mode);
+  }
+
   return IsTargetActiveForMode(aTarget, mode);
 }
 
