diff --git a/mobile/android/gradle/with_gecko_binaries.gradle b/mobile/android/gradle/with_gecko_binaries.gradle
index 74aa160449a2d..e86c3223ef3b9 100644
--- a/mobile/android/gradle/with_gecko_binaries.gradle
+++ b/mobile/android/gradle/with_gecko_binaries.gradle
@@ -24,7 +24,7 @@ ext.configureVariantWithGeckoBinaries = { variant ->
         // the moz.build system and should never be re-entrant in this way.
         def assetGenTask = tasks.findByName("generate${variant.name.capitalize()}Assets")
         def jniLibFoldersTask = tasks.findByName("merge${variant.name.capitalize()}JniLibFolders")
-        if (!mozconfig.substs.MOZILLA_OFFICIAL && !mozconfig.substs.ENABLE_MOZSEARCH_PLUGIN) {
+        if (!mozconfig.substs.ENABLE_MOZSEARCH_PLUGIN) {
             assetGenTask.dependsOn rootProject.machStagePackage
             jniLibFoldersTask.dependsOn rootProject.machStagePackage
         }
