From 40d09fef62184a8ccdcb8e8cf41c7b35d001fadf Mon Sep 17 00:00:00 2001
From: Akash Yadav <itsaky01@gmail.com>
Date: Mon, 28 Apr 2025 11:19:14 +0530
Subject: [PATCH] fix(patches): update 'gecko-fix-125-compile.patch' for
 'FIREFOX_138_0_BUILD1'

Signed-off-by: Akash Yadav <itsaky01@gmail.com>
---
 mfbt/SmallPointerArray.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/mfbt/SmallPointerArray.h b/mfbt/SmallPointerArray.h
index c63e3980f9..061bb1fc8b 100644
--- a/mfbt/SmallPointerArray.h
+++ b/mfbt/SmallPointerArray.h
@@ -201,9 +201,9 @@ class SmallPointerArray {
  private:
   T** beginInternal() const {
     if (first()) {
-      static_assert(sizeof(T*) == sizeof(Element),
+      static_assert(sizeof(T*) == sizeof(SmallArrayElement),
                     "pointer ops on &first() must produce adjacent "
-                    "Element::mValue arms");
+                    "SmallArrayElement::mValue arms");
       return &first();
     }
 
@@ -259,7 +259,7 @@ class SmallPointerArray {
   // When !mArray[0].mValue && mArray[1].mVector, mArray[1].mVector contains
   // the contents of an array of arbitrary size (even less than two if it ever
   // contained three elements and elements were removed).
-  union Element {
+  union SmallArrayElement {
     T* mValue;
     std::vector<T*>* mVector;
   } mArray[2];
-- 
2.49.0

