diff --git a/toolkit/components/translations/content/translations-document.sys.mjs b/toolkit/components/translations/content/translations-document.sys.mjs
index 8a82efc734..bf58aab266 100644
--- a/toolkit/components/translations/content/translations-document.sys.mjs
+++ b/toolkit/components/translations/content/translations-document.sys.mjs
@@ -5457,7 +5457,7 @@ class TranslationScheduler {
     let stackSizesAtStart = null;
     const activeRequestsAtStart = this.#activeRequests.size;
     const unscheduledRequestsAtStart = this.#unscheduledRequestPriorities.size;
-    if (Services.profiler.IsActive() || lazy.console.shouldLog("Debug")) {
+    if (lazy.console.shouldLog("Debug")) {
       // We need to preserve the sizes prior to scheduling only if we are adding profiler markers,
       // or if we are logging to console debug. Otherwise we shouldn't bother with these computations.
       stackSizesAtStart = this.#priorityStacks.map(stack => stack.size);
