diff --git a/toolkit/components/crashes/RemoteSettingsCrashPull.sys.mjs b/toolkit/components/crashes/RemoteSettingsCrashPull.sys.mjs
index d634f9def5..05e8007f24 100644
--- a/toolkit/components/crashes/RemoteSettingsCrashPull.sys.mjs
+++ b/toolkit/components/crashes/RemoteSettingsCrashPull.sys.mjs
@@ -12,7 +12,7 @@ ChromeUtils.defineESModuleGetters(lazy, {
     "resource://services-settings/RemoteSettingsClient.sys.mjs",
 });
 
-const REMOTE_SETTINGS_CRASH_COLLECTION = "crash-reports-ondemand";
+const REMOTE_SETTINGS_CRASH_COLLECTION = "";
 
 // Remote Settings collections might want a different limit
 const PENDING_REMOTE_CRASH_REPORT_DAYS = 90;
