diff --git a/toolkit/components/enterprisepolicies/EnterprisePoliciesParent.sys.mjs b/toolkit/components/enterprisepolicies/EnterprisePoliciesParent.sys.mjs
index bb451a279599da..5eb8e609952ebd 100644
--- a/toolkit/components/enterprisepolicies/EnterprisePoliciesParent.sys.mjs
+++ b/toolkit/components/enterprisepolicies/EnterprisePoliciesParent.sys.mjs
@@ -147,6 +147,10 @@ EnterprisePoliciesManager.prototype = {
   },
 
   _chooseProvider() {
+    const { IronFoxPoliciesProvider } = ChromeUtils.importESModule(
+      "resource://gre/modules/IronFoxPoliciesProvider.sys.mjs"
+    );
+    return new IronFoxPoliciesProvider();
     let platformProvider = null;
     if (AppConstants.platform == "win" && AppConstants.MOZ_SYSTEM_POLICIES) {
       platformProvider = new WindowsGPOPoliciesProvider();
diff --git a/toolkit/components/enterprisepolicies/moz.build b/toolkit/components/enterprisepolicies/moz.build
index 4fcb377c80..539080df94 100644
--- a/toolkit/components/enterprisepolicies/moz.build
+++ b/toolkit/components/enterprisepolicies/moz.build
@@ -15,7 +15,7 @@ XPIDL_MODULE = "enterprisepolicies"
 
 TEST_DIRS += ["tests"]
 
-if CONFIG["MOZ_WIDGET_TOOLKIT"] != "android":
+if CONFIG["IRONFOX"]:
     EXTRA_JS_MODULES += [
         "EnterprisePolicies.sys.mjs",
         "EnterprisePoliciesContent.sys.mjs",
