diff --git a/toolkit/content/jar.mn b/toolkit/content/jar.mn
index 83fcfac2d4..8207c2bf18 100644
--- a/toolkit/content/jar.mn
+++ b/toolkit/content/jar.mn
@@ -170,9 +170,9 @@ toolkit.jar:
 #ifdef XP_MACOSX
    content/global/macWindowMenu.js
 #endif
-   content/global/gmp-sources/openh264.json    (gmp-sources/openh264.json)
-   content/global/gmp-sources/widevinecdm.json (gmp-sources/widevinecdm.json)
-   content/global/gmp-sources/widevinecdm_l1.json (gmp-sources/widevinecdm_l1.json)
+#   content/global/gmp-sources/openh264.json    (gmp-sources/openh264.json)
+#   content/global/gmp-sources/widevinecdm.json (gmp-sources/widevinecdm.json)
+#   content/global/gmp-sources/widevinecdm_l1.json (gmp-sources/widevinecdm_l1.json)
 
 # Third party files
    content/global/third_party/d3/d3.js                (/third_party/js/d3/d3.js)
