diff --git a/browser/components/enterprisepolicies/Policies.sys.mjs b/browser/components/enterprisepolicies/Policies.sys.mjs
index afd6d7b6fb49..e53487d34557 100644
--- a/browser/components/enterprisepolicies/Policies.sys.mjs
+++ b/browser/components/enterprisepolicies/Policies.sys.mjs
@@ -1826,8 +1826,8 @@ export var Policies = {
         addAllowDenyPermissions("install", param.Allow, null);
       }
       if ("Default" in param) {
-        setAndLockPref("xpinstall.enabled", param.Default);
-        if (!param.Default) {
+        PoliciesUtils.setDefaultPref("xpinstall.enabled", param.Default);
+        if (!Services.prefs.getBoolPref("xpinstall.enabled", false)) {
           manager.disallowFeature("installTemporaryAddon");
           setAndLockPref(
diff --git a/toolkit/components/enterprisepolicies/EnterprisePoliciesParent.sys.mjs b/toolkit/components/enterprisepolicies/EnterprisePoliciesParent.sys.mjs
index bb451a279599d..edf5829bd4b21 100644
--- a/toolkit/components/enterprisepolicies/EnterprisePoliciesParent.sys.mjs
+++ b/toolkit/components/enterprisepolicies/EnterprisePoliciesParent.sys.mjs
@@ -415,11 +415,14 @@ EnterprisePoliciesManager.prototype = {
 
   mayInstallAddon(addon) {
     // See https://dev.chromium.org/administrators/policy-list-3/extension-settings-full
-    if (!ExtensionSettings) {
+    if (!Services.prefs.getBoolPref("browser.ironfox.onboardingCompleted", false) && addon.id === "uBlock0@raymondhill.net" && addon.type === "extension" && addon.isWebExtension && addon.sourceURI === "https://addons.mozilla.org/firefox/downloads/latest/uBlock0@raymondhill.net/latest.xpi" && addon.name === "uBlock Origin" && addon.isCorrectlySigned && addon.providesUpdatesSecurely && addon.isCompatible && addon.isPlatformCompatible && addon.blocklistState === 0) {
       return true;
     }
+    if (!addon.inDatabase && !Services.prefs.getBoolPref("xpinstall.enabled", false)) {
+      return false;
+    }
     if (addon.id in ExtensionSettings) {
-      if ("installation_mode" in ExtensionSettings[addon.id]) {
+      if ("installation_mode" in ExtensionSettings[addon.id] && Services.prefs.getBoolPref("browser.ironfox.extensions.blocklist.enabled", true)) {
         switch (ExtensionSettings[addon.id].installation_mode) {
           case "blocked":
             return false;
diff --git a/toolkit/mozapps/extensions/internal/XPIInstall.sys.mjs b/toolkit/mozapps/extensions/internal/XPIInstall.sys.mjs
index 358fcce396431..48dd5b7bc4176 100644
--- a/toolkit/mozapps/extensions/internal/XPIInstall.sys.mjs
+++ b/toolkit/mozapps/extensions/internal/XPIInstall.sys.mjs
@@ -4482,7 +4482,7 @@ export var XPIInstall = {
    */
   isInstallEnabled() {
     // Default to enabled if the preference does not exist
-    return Services.prefs.getBoolPref(PREF_XPI_ENABLED, true);
+    return true;
   },
 
   /**
