diff --git a/mobile/android/fenix/app/src/main/res/xml/tracking_protection_preferences.xml b/mobile/android/fenix/app/src/main/res/xml/tracking_protection_preferences.xml
--- a/mobile/android/fenix/app/src/main/res/xml/tracking_protection_preferences.xml
+++ b/mobile/android/fenix/app/src/main/res/xml/tracking_protection_preferences.xml
@@ -93,7 +93,8 @@
     <androidx.preference.SwitchPreference
         android:defaultValue="false"
         android:key="@string/pref_key_privacy_enable_global_privacy_control"
-        android:title="@string/preference_enhanced_tracking_protection_custom_global_privacy_control" />
+        android:title="@string/preference_enhanced_tracking_protection_custom_global_privacy_control"
+        app:isPreferenceVisible="false" />
     <Preference android:layout="@layout/preference_divider" />
     <Preference
         android:icon="@drawable/ic_internet"
diff --git a/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/GeckoRuntimeSettings.java b/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/GeckoRuntimeSettings.java
--- a/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/GeckoRuntimeSettings.java
+++ b/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/GeckoRuntimeSettings.java
@@ -673,11 +673,11 @@
   /* package */ final PrefWithoutDefault<Integer> mExtensionsProcessCrashThreshold =
       new PrefWithoutDefault<Integer>("extensions.webextensions.crash.threshold");
   /* package */ final Pref<Boolean> mGlobalPrivacyControlEnabled =
-      new Pref<Boolean>("privacy.globalprivacycontrol.enabled", false);
+      new Pref<Boolean>("z99.this.is.not", false);
   /* package */ final Pref<Boolean> mGlobalPrivacyControlEnabledPrivateMode =
-      new Pref<Boolean>("privacy.globalprivacycontrol.pbmode.enabled", true);
+      new Pref<Boolean>("z999.the.droid", true);
   /* package */ final Pref<Boolean> mGlobalPrivacyControlFunctionalityEnabled =
-      new Pref<Boolean>("privacy.globalprivacycontrol.functionality.enabled", true);
+      new Pref<Boolean>("z9999.youre.looking.for", true);
   /* package */ final PrefWithoutDefault<Boolean> mFingerprintingProtection =
       new PrefWithoutDefault<Boolean>("privacy.fingerprintingProtection");
   /* package */ final PrefWithoutDefault<Boolean> mFingerprintingProtectionPrivateMode =
