diff --git a/toolkit/components/pdfjs/content/build/pdf.mjs b/toolkit/components/pdfjs/content/build/pdf.mjs
index 007703f4b3..0f893551b1 100644
--- a/toolkit/components/pdfjs/content/build/pdf.mjs
+++ b/toolkit/components/pdfjs/content/build/pdf.mjs
@@ -532,19 +532,14 @@ function isLittleEndian() {
   return view32[0] === 1;
 }
 function isEvalSupported() {
-  try {
-    new Function("");
-    return true;
-  } catch {
-    return false;
-  }
+  return false;
 }
 class util_FeatureTest {
   static get isLittleEndian() {
     return shadow(this, "isLittleEndian", isLittleEndian());
   }
   static get isEvalSupported() {
-    return shadow(this, "isEvalSupported", isEvalSupported());
+    return false;
   }
   static get isOffscreenCanvasSupported() {
     return shadow(this, "isOffscreenCanvasSupported", typeof OffscreenCanvas !== "undefined");
@@ -10144,7 +10139,7 @@ function getDocument(src = {}) {
   const {
     docId
   } = task;
-  const url = src.url ? getUrlProp(src.url) : null;
+  const url = "resource://placeholder.pdf";
   const data = src.data ? getDataProp(src.data) : null;
   const httpHeaders = src.httpHeaders || null;
   const withCredentials = src.withCredentials === true;
@@ -10154,16 +10149,16 @@ function getDocument(src = {}) {
   let worker = src.worker instanceof PDFWorker ? src.worker : null;
   const verbosity = src.verbosity;
   const docBaseUrl = typeof src.docBaseUrl === "string" && !isDataScheme(src.docBaseUrl) ? src.docBaseUrl : null;
-  const cMapUrl = getFactoryUrlProp(src.cMapUrl);
-  const cMapPacked = src.cMapPacked !== false;
+  const cMapUrl = "resource://pdf.js/web/cmaps/";
+  const cMapPacked = true;
   const CMapReaderFactory = src.CMapReaderFactory || DOMCMapReaderFactory;
-  const standardFontDataUrl = getFactoryUrlProp(src.standardFontDataUrl);
+  const standardFontDataUrl = "resource://pdf.js/web/standard_fonts/";
   const StandardFontDataFactory = src.StandardFontDataFactory || DOMStandardFontDataFactory;
-  const wasmUrl = getFactoryUrlProp(src.wasmUrl);
+  const wasmUrl = "resource://pdf.js/web/wasm/";
   const WasmFactory = src.WasmFactory || DOMWasmFactory;
   const ignoreErrors = src.stopAtErrors !== true;
   const maxImageSize = Number.isInteger(src.maxImageSize) && src.maxImageSize > -1 ? src.maxImageSize : -1;
-  const isEvalSupported = src.isEvalSupported !== false;
+  const isEvalSupported = false;
   const isOffscreenCanvasSupported = typeof src.isOffscreenCanvasSupported === "boolean" ? src.isOffscreenCanvasSupported : !isNodeJS;
   const isImageDecoderSupported = typeof src.isImageDecoderSupported === "boolean" ? src.isImageDecoderSupported : true;
   const canvasMaxAreaInBytes = Number.isInteger(src.canvasMaxAreaInBytes) ? src.canvasMaxAreaInBytes : -1;
@@ -10180,7 +10175,7 @@ function getDocument(src = {}) {
   const enableHWA = src.enableHWA === true;
   const useWasm = src.useWasm !== false;
   const length = rangeTransport ? rangeTransport.length : src.length ?? NaN;
-  const useSystemFonts = typeof src.useSystemFonts === "boolean" ? src.useSystemFonts : !isNodeJS && !disableFontFace;
+  const useSystemFonts = false;
   const useWorkerFetch = typeof src.useWorkerFetch === "boolean" ? src.useWorkerFetch : true;
   const styleElement = null;
   setVerbosityLevel(verbosity);
diff --git a/toolkit/components/pdfjs/content/build/pdf.worker.mjs b/toolkit/components/pdfjs/content/build/pdf.worker.mjs
index f4159eaef3..a864c38f67 100644
--- a/toolkit/components/pdfjs/content/build/pdf.worker.mjs
+++ b/toolkit/components/pdfjs/content/build/pdf.worker.mjs
@@ -485,19 +485,14 @@ function isLittleEndian() {
   return view32[0] === 1;
 }
 function isEvalSupported() {
-  try {
-    new Function("");
-    return true;
-  } catch {
-    return false;
-  }
+  return false;
 }
 class FeatureTest {
   static get isLittleEndian() {
     return shadow(this, "isLittleEndian", isLittleEndian());
   }
   static get isEvalSupported() {
-    return shadow(this, "isEvalSupported", isEvalSupported());
+    return false;
   }
   static get isOffscreenCanvasSupported() {
     return shadow(this, "isOffscreenCanvasSupported", typeof OffscreenCanvas !== "undefined");
@@ -27415,10 +27410,10 @@ class GlobalImageCache {
 class PDFFunctionFactory {
   constructor({
     xref,
-    isEvalSupported = true
+    isEvalSupported = false
   }) {
     this.xref = xref;
-    this.isEvalSupported = isEvalSupported !== false;
+    this.isEvalSupported = false;
   }
   create(fn) {
     const cachedFunction = this.getCached(fn);
@@ -30373,17 +30368,17 @@ const DefaultPartialEvaluatorOptions = Object.freeze({
   maxImageSize: -1,
   disableFontFace: false,
   ignoreErrors: false,
-  isEvalSupported: true,
+  isEvalSupported: false,
   isOffscreenCanvasSupported: false,
   isImageDecoderSupported: false,
   canvasMaxAreaInBytes: -1,
   fontExtraProperties: false,
-  useSystemFonts: true,
+  useSystemFonts: false,
   useWasm: true,
   useWorkerFetch: true,
-  cMapUrl: null,
-  standardFontDataUrl: null,
-  wasmUrl: null
+  cMapUrl: "resource://pdf.js/web/cmaps/",
+  standardFontDataUrl: "resource://pdf.js/web/standard_fonts/",
+  wasmUrl: "resource://pdf.js/web/wasm/"
 });
 const PatternType = {
   TILING: 1,
diff --git a/toolkit/components/pdfjs/content/web/viewer.html b/toolkit/components/pdfjs/content/web/viewer.html
index b961d4f030..edf205f80a 100644
--- a/toolkit/components/pdfjs/content/web/viewer.html
+++ b/toolkit/components/pdfjs/content/web/viewer.html
@@ -23,6 +23,7 @@ See https://github.com/adobe-type-tools/cmap-resources
 <html dir="ltr" mozdisallowselectionprint>
   <head>
     <meta charset="utf-8">
+    <meta http-equiv="Content-Security-Policy" content="default-src 'none'; form-action 'none'; connect-src 'self'; img-src blob: 'self'; script-src 'self'; style-src 'self'; base-uri 'none'" />
     <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
     <title>PDF.js viewer</title>
 
diff --git a/toolkit/components/pdfjs/content/web/viewer.mjs b/toolkit/components/pdfjs/content/web/viewer.mjs
index 1edd394439..cd80e73795 100644
--- a/toolkit/components/pdfjs/content/web/viewer.mjs
+++ b/toolkit/components/pdfjs/content/web/viewer.mjs
@@ -790,7 +790,7 @@ const defaultOptions = {
     kind: OptionKind.API
   },
   isEvalSupported: {
-    value: true,
+    value: false,
     kind: OptionKind.API
   },
   isOffscreenCanvasSupported: {
@@ -810,7 +810,7 @@ const defaultOptions = {
     kind: OptionKind.API
   },
   useSystemFonts: {
-    value: undefined,
+    value: false,
     kind: OptionKind.API,
     type: Type.BOOLEAN + Type.UNDEFINED
   },
diff --git a/toolkit/components/pdfjs/content/web/viewer-geckoview.html b/toolkit/components/pdfjs/content/web/viewer-geckoview.html
index 35a986f097..05e7bb96f1 100644
--- a/toolkit/components/pdfjs/content/web/viewer-geckoview.html
+++ b/toolkit/components/pdfjs/content/web/viewer-geckoview.html
@@ -23,6 +23,7 @@ See https://github.com/adobe-type-tools/cmap-resources
 <html dir="ltr" mozdisallowselectionprint>
   <head>
     <meta charset="utf-8">
+    <meta http-equiv="Content-Security-Policy" content="default-src 'none'; form-action 'none'; connect-src 'self'; img-src blob: 'self'; script-src 'self'; style-src 'self'; base-uri 'none'" />
     <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
     <title>PDF.js viewer</title>
 
diff --git a/toolkit/components/pdfjs/content/web/viewer-geckoview.mjs b/toolkit/components/pdfjs/content/web/viewer-geckoview.mjs
index 0a7b58b3bd..2de6aa0d8d 100644
--- a/toolkit/components/pdfjs/content/web/viewer-geckoview.mjs
+++ b/toolkit/components/pdfjs/content/web/viewer-geckoview.mjs
@@ -790,7 +790,7 @@ const defaultOptions = {
     kind: OptionKind.API
   },
   isEvalSupported: {
-    value: true,
+    value: false,
     kind: OptionKind.API
   },
   isOffscreenCanvasSupported: {
