diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
index 43bea16286..f31b83b648 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
@@ -256,6 +256,33 @@ class SecretSettingsFragment : PreferenceFragmentCompat() {
             )
             onPreferenceChangeListener = SharedPreferenceUpdater()
         }
+
+        requirePreference<SwitchPreference>(R.string.pref_key_eme_enabled).apply {
+            isChecked = context.settings().emeEnabled
+            onPreferenceChangeListener = object : SharedPreferenceUpdater() {
+                override fun onPreferenceChange(preference: Preference, newValue: Any?): Boolean {
+                    context.components.core.engine.settings.emeEnabled =
+                        newValue as Boolean
+                    return super.onPreferenceChange(preference, newValue)
+                }
+            }
+        }
+
+        requirePreference<SwitchPreference>(R.string.pref_key_enable_unifiedpush).apply {
+            isChecked = context.settings().enableUnifiedPush
+            onPreferenceChangeListener = SharedPreferenceUpdater()
+        }
+
+        requirePreference<SwitchPreference>(R.string.pref_key_widevine_enabled).apply {
+            isChecked = context.settings().widevineEnabled
+            onPreferenceChangeListener = object : SharedPreferenceUpdater() {
+                override fun onPreferenceChange(preference: Preference, newValue: Any?): Boolean {
+                    context.components.core.engine.settings.widevineEnabled =
+                        newValue as Boolean
+                    return super.onPreferenceChange(preference, newValue)
+                }
+            }
+        }
     }
 
     override fun onPreferenceTreeClick(preference: Preference): Boolean {
diff --git a/mobile/android/fenix/app/src/main/res/xml/secret_settings_preferences.xml b/mobile/android/fenix/app/src/main/res/xml/secret_settings_preferences.xml
index 094406c52e..3fcbd7535e 100644
--- a/mobile/android/fenix/app/src/main/res/xml/secret_settings_preferences.xml
+++ b/mobile/android/fenix/app/src/main/res/xml/secret_settings_preferences.xml
@@ -10,6 +10,24 @@
         android:title="@string/preferences_debug_settings_allow_third_party_root_certs"
         android:summary="@string/preferences_debug_settings_allow_third_party_root_certs_summary"
         app:iconSpaceReserved="false" />
+    <SwitchPreference
+        android:defaultValue="true"
+        android:key="@string/pref_key_enable_unifiedpush"
+        android:title="@string/preferences_debug_settings_unifiedpush"
+        app:iconSpaceReserved="false" />
+    <SwitchPreference
+        android:defaultValue="false"
+        android:key="@string/pref_key_eme_enabled"
+        android:title="@string/preferences_debug_settings_eme_enabled"
+        android:summary="@string/unsupported_summary"
+        app:iconSpaceReserved="false" />
+    <SwitchPreference
+        android:defaultValue="false"
+        android:dependency="@string/pref_key_eme_enabled"
+        android:key="@string/pref_key_widevine_enabled"
+        android:title="@string/preferences_debug_settings_widevine_enabled"
+        android:summary="@string/unsupported_summary"
+        app:iconSpaceReserved="false" />
     <SwitchPreference
         android:defaultValue="false"
         android:key="@string/pref_key_nimbus_use_preview"
