diff --git a/python/mach/mach/sentry.py b/python/mach/mach/sentry.py
index b1a2513596..5c32785c99 100644
--- a/python/mach/mach/sentry.py
+++ b/python/mach/mach/sentry.py
@@ -21,7 +21,7 @@ from mach.util import get_state_dir
 
 # https://sentry.io/organizations/mozilla/projects/mach/
 _SENTRY_DSN = (
-    "https://5cfe351fb3a24e8d82c751252b48722b@o1069899.ingest.sentry.io/6250014"
+    ""
 )
 
 
@@ -223,5 +223,5 @@ def _is_unmodified_mach_core(topsrcdir: Path):
         _is_unmodified_mach_core_result = False
 
 
-_is_unmodified_mach_core_result = None
-_is_unmodified_mach_core_thread = None
+_is_unmodified_mach_core_result = False
+_is_unmodified_mach_core_thread = False
diff --git a/python/mach/mach/telemetry.py b/python/mach/mach/telemetry.py
index a3fda9a59b..48a593ac36 100644
--- a/python/mach/mach/telemetry.py
+++ b/python/mach/mach/telemetry.py
@@ -88,14 +88,14 @@ def is_applicable_telemetry_environment():
         # we don't want to collect telemetry again down here.
         return False
 
-    return True
+    return False
 
 
 def is_telemetry_enabled(settings):
     if os.environ.get("DISABLE_TELEMETRY") == "1":
         return False
 
-    return settings.mach_telemetry.is_enabled
+    return False
 
 
 def arcrc_path():
