diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 72493a2c21..deaf775539 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -1161,10 +1161,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         default = true,
     )
 
-    var defaultTopSitesAdded by booleanPreference(
-        appContext.getPreferenceKey(R.string.default_top_sites_added),
-        default = false,
-    )
+    var defaultTopSitesAdded = true
 
     var shouldShowSearchSuggestionsInPrivate by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_show_search_suggestions_in_private),
