diff --git a/mobile/android/android-components/.buildconfig.yml b/mobile/android/android-components/.buildconfig.yml
index 26e76a18d2..89d595d626 100644
--- a/mobile/android/android-components/.buildconfig.yml
+++ b/mobile/android/android-components/.buildconfig.yml
@@ -1521,28 +1521,28 @@ projects:
     - support-webextensions
     - tooling-lint
     - ui-icons
-  feature-webcompat-reporter:
-    description: Feature that enables users to report site issues to Mozilla's Web
-      Compatibility team.
-    path: components/feature/webcompat-reporter
-    publish: true
-    upstream_dependencies:
-    - browser-errorpages
-    - browser-state
-    - concept-awesomebar
-    - concept-base
-    - concept-engine
-    - concept-fetch
-    - concept-storage
-    - lib-publicsuffixlist
-    - lib-state
-    - support-base
-    - support-ktx
-    - support-test
-    - support-utils
-    - support-webextensions
-    - tooling-lint
-    - ui-icons
+#  feature-webcompat-reporter:
+#    description: Feature that enables users to report site issues to Mozilla's Web
+#      Compatibility team.
+#    path: components/feature/webcompat-reporter
+#    publish: true
+#    upstream_dependencies:
+#    - browser-errorpages
+#    - browser-state
+#    - concept-awesomebar
+#    - concept-base
+#    - concept-engine
+#    - concept-fetch
+#    - concept-storage
+#    - lib-publicsuffixlist
+#    - lib-state
+#    - support-base
+#    - support-ktx
+#    - support-test
+#    - support-utils
+#    - support-webextensions
+#    - tooling-lint
+#    - ui-icons
   feature-webnotifications:
     description: Feature component for Web Notifications.
     path: components/feature/webnotifications
@@ -1744,7 +1744,7 @@ projects:
     - feature-toolbar
     - feature-top-sites
     - feature-webcompat
-    - feature-webcompat-reporter
+#    - feature-webcompat-reporter
     - feature-webnotifications
     - lib-crash
     - lib-dataprotect
diff --git a/mobile/android/android-components/samples/browser/build.gradle b/mobile/android/android-components/samples/browser/build.gradle
index bd70dd476b..e95f76ffe4 100644
--- a/mobile/android/android-components/samples/browser/build.gradle
+++ b/mobile/android/android-components/samples/browser/build.gradle
@@ -114,7 +114,7 @@ dependencies {
     implementation project(':feature-findinpage')
     implementation project(':feature-sitepermissions')
     implementation project(':feature-webcompat')
-    implementation project(':feature-webcompat-reporter')
+//    implementation project(':feature-webcompat-reporter')
     implementation project(':feature-webnotifications')
     implementation project(':feature-addons')
 
diff --git a/mobile/android/fenix/.buildconfig.yml b/mobile/android/fenix/.buildconfig.yml
index b5aa2ce3a4..8837f09217 100644
--- a/mobile/android/fenix/.buildconfig.yml
+++ b/mobile/android/fenix/.buildconfig.yml
@@ -59,7 +59,7 @@ projects:
     - feature-top-sites
     - feature-webauthn
     - feature-webcompat
-    - feature-webcompat-reporter
+#    - feature-webcompat-reporter
     - feature-webnotifications
     - lib-crash
     - lib-crash-sentry
diff --git a/mobile/android/fenix/app/build.gradle b/mobile/android/fenix/app/build.gradle
index d9a6045423..16ac38e60b 100644
--- a/mobile/android/fenix/app/build.gradle
+++ b/mobile/android/fenix/app/build.gradle
@@ -618,7 +618,7 @@ dependencies {
     implementation project(':feature-webauthn')
     implementation project(':feature-webcompat')
     implementation project(':feature-webnotifications')
-    implementation project(':feature-webcompat-reporter')
+//    implementation project(':feature-webcompat-reporter')
 
     implementation project(':service-pocket')
     implementation project(':service-mars')
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/MenuDialogFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/MenuDialogFragment.kt
index 1443661bb8..218f4f1518 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/MenuDialogFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/MenuDialogFragment.kt
@@ -318,7 +318,7 @@ class MenuDialogFragment : BottomSheetDialogFragment() {
                         ?.requestedTranslationPair?.toLanguage
                     val isExtensionsProcessDisabled = browserStore.state.extensionsProcessDisabled
                     val isWebCompatReporterSupported =
-                        FxNimbus.features.menuRedesign.value().reportSiteIssue
+                        false
 
                     val isDesktopMode by store.observeAsState(initialValue = false) { state ->
                         state.isDesktopMode
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/compose/MainMenu.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/compose/MainMenu.kt
index 3221dddb98..f611287899 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/compose/MainMenu.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/compose/MainMenu.kt
@@ -470,7 +470,7 @@ private fun MenuDialogPreview() {
                 isDesktopMode = false,
                 isPdf = false,
                 isTranslationSupported = true,
-                isWebCompatReporterSupported = true,
+                isWebCompatReporterSupported = false,
                 showQuitMenu = true,
                 isExtensionsProcessDisabled = true,
                 extensionsMenuItemDescription = "No extensions enabled",
@@ -516,7 +516,7 @@ private fun MenuDialogPrivatePreview() {
                 isDesktopMode = false,
                 isPdf = false,
                 isTranslationSupported = true,
-                isWebCompatReporterSupported = true,
+                isWebCompatReporterSupported = false,
                 showQuitMenu = true,
                 isExtensionsProcessDisabled = false,
                 extensionsMenuItemDescription = "No extensions enabled",
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
index e7e19273f9..2e6139bd25 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
@@ -216,7 +216,7 @@ open class DefaultToolbarMenu(
         val url = store.state.selectedTab?.content?.url
         val isAboutUrl = url?.isAboutUrl() ?: false
         val isContentUrl = url?.isContentUrl() ?: false
-        return !isAboutUrl && !isContentUrl
+        return false
     }
     // End of predicates //
 
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt
index 13db9f3c83..094dd09a49 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt
@@ -50,7 +50,7 @@ import mozilla.components.feature.search.ext.waitForSelectedOrDefaultSearchEngin
 import mozilla.components.feature.syncedtabs.commands.GlobalSyncedTabsCommandsProvider
 import mozilla.components.feature.top.sites.TopSitesFrecencyConfig
 import mozilla.components.feature.top.sites.TopSitesProviderConfig
-import mozilla.components.feature.webcompat.reporter.WebCompatReporterFeature
+//import mozilla.components.feature.webcompat.reporter.WebCompatReporterFeature
 import mozilla.components.lib.crash.CrashReporter
 import mozilla.components.service.fxa.manager.SyncEnginesStorage
 import mozilla.components.service.sync.logins.LoginsApiException
@@ -699,7 +699,7 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
                     // TODO: Bug 1953359 - remove the code below in the next release.
                     if (Config.channel.isNightlyOrDebug || Config.channel.isBeta) {
                         logger.debug("Attempting to uninstall the WebCompat Reporter extension")
-                        WebCompatReporterFeature.uninstall(components.core.engine)
+//                        WebCompatReporterFeature.uninstall(components.core.engine)
                     }
                 },
                 onUpdatePermissionRequest = components.addonUpdater::onUpdatePermissionRequest,
