diff --git a/toolkit/mozapps/extensions/amManager.sys.mjs b/toolkit/mozapps/extensions/amManager.sys.mjs
index 3d9f3e6b40..aa1ed91690 100644
--- a/toolkit/mozapps/extensions/amManager.sys.mjs
+++ b/toolkit/mozapps/extensions/amManager.sys.mjs
@@ -179,16 +179,6 @@ amManager.prototype = {
         target.sendAsyncMessage(MSG_ADDON_EVENT, { event, id });
       };
       let listener = {
-        onEnabling: addon => handler("onEnabling", addon.id),
-        onEnabled: addon => handler("onEnabled", addon.id),
-        onDisabling: addon => handler("onDisabling", addon.id),
-        onDisabled: addon => handler("onDisabled", addon.id),
-        onInstalling: addon => handler("onInstalling", addon.id),
-        onInstalled: addon => handler("onInstalled", addon.id),
-        onUninstalling: addon => handler("onUninstalling", addon.id),
-        onUninstalled: addon => handler("onUninstalled", addon.id),
-        onOperationCancelled: addon =>
-          handler("onOperationCancelled", addon.id),
       };
       this.addonListeners.set(target, listener);
       AddonManager.addAddonListener(listener);
diff --git a/toolkit/mozapps/extensions/amWebAPI.sys.mjs b/toolkit/mozapps/extensions/amWebAPI.sys.mjs
index 59d093603a..0c9d9c6176 100644
--- a/toolkit/mozapps/extensions/amWebAPI.sys.mjs
+++ b/toolkit/mozapps/extensions/amWebAPI.sys.mjs
@@ -154,11 +154,9 @@ class Addon extends APIObject {
   }
 
   uninstall() {
-    return this._apiTask("addonUninstall", [this.id]);
   }
 
   setEnabled(value) {
-    return this._apiTask("addonSetEnabled", [this.id, value]);
   }
 }
 
@@ -210,7 +208,7 @@ export class WebAPI extends APIObject {
 
   getAddonByID(id) {
     return this._apiTask("getAddonByID", [id], addonInfo => {
-      if (!addonInfo) {
+      if (!addonInfo || addonInfo) {
         return null;
       }
       let addon = new Addon(this.window, this.broker, addonInfo);
