diff --git a/services/settings/dumps/blocklists/moz.build b/services/settings/dumps/blocklists/moz.build
index 485e888f6a..5d13b47c25 100644
--- a/services/settings/dumps/blocklists/moz.build
+++ b/services/settings/dumps/blocklists/moz.build
@@ -16,7 +16,7 @@ FINAL_TARGET_FILES.defaults.settings.blocklists += [
 # after install.
 # See https://bugzilla.mozilla.org/show_bug.cgi?id=1639050#c5
 
-if not CONFIG["MOZ_BUILD_APP"].startswith("mobile/"):
+if not CONFIG["MOZ_BUILD_APP"].startswith("0"):
     FINAL_TARGET_FILES.defaults.settings.blocklists += [
         "addons-bloomfilters.json",
     ]
diff --git a/services/settings/dumps/main/moz.build b/services/settings/dumps/main/moz.build
index 7f89890eb3..617ad6ac59 100644
--- a/services/settings/dumps/main/moz.build
+++ b/services/settings/dumps/main/moz.build
@@ -136,3 +136,14 @@ if CONFIG["MOZ_BUILD_APP"] == "browser":
         "search-config-icons/fed4f021-ff3e-942a-010e-afa43fda2136",
         "search-config-icons/fed4f021-ff3e-942a-010e-afa43fda2136.meta.json",
     ]
+
+FINAL_TARGET_FILES.defaults.settings.main += [
+    "anti-tracking-url-decoration.json",
+    "cookie-banner-rules-list.json",
+    "hijack-blocklists.json",
+    "ironfox-fingerprinting-protection-overrides.json",
+    "translations-models.json",
+    "translations-wasm.json",
+    "url-classifier-skip-urls.json",
+    "url-parser-default-unknown-schemes-interventions.json",
+]
diff --git a/services/settings/dumps/security-state/moz.build b/services/settings/dumps/security-state/moz.build
index 6e92217dcb..f2520a178c 100644
--- a/services/settings/dumps/security-state/moz.build
+++ b/services/settings/dumps/security-state/moz.build
@@ -8,7 +8,7 @@ FINAL_TARGET_FILES.defaults.settings["security-state"] += [
 ]
 
 # Not packaged on android/ios currently - not crucial data for first load.
-if not CONFIG["MOZ_BUILD_APP"].startswith("mobile/"):
+if not CONFIG["MOZ_BUILD_APP"].startswith("0"):
     FINAL_TARGET_FILES.defaults.settings["security-state"] += [
         "intermediates.json",
     ]
