diff --git a/services/settings/dumps/blocklists/moz.build b/services/settings/dumps/blocklists/moz.build
index 485e888f6a..27ca6589d6 100644
--- a/services/settings/dumps/blocklists/moz.build
+++ b/services/settings/dumps/blocklists/moz.build
@@ -16,14 +16,13 @@ FINAL_TARGET_FILES.defaults.settings.blocklists += [
 # after install.
 # See https://bugzilla.mozilla.org/show_bug.cgi?id=1639050#c5
 
-if not CONFIG["MOZ_BUILD_APP"].startswith("mobile/"):
-    FINAL_TARGET_FILES.defaults.settings.blocklists += [
-        "addons-bloomfilters.json",
-    ]
+FINAL_TARGET_FILES.defaults.settings.blocklists += [
+    "addons-bloomfilters.json",
+]
 
-    FINAL_TARGET_FILES.defaults.settings.blocklists["addons-bloomfilters"] += [
-        "addons-bloomfilters/addons-mlbf.bin",
-        "addons-bloomfilters/addons-mlbf.bin.meta.json",
-        "addons-bloomfilters/softblocks-addons-mlbf.bin",
-        "addons-bloomfilters/softblocks-addons-mlbf.bin.meta.json",
-    ]
+FINAL_TARGET_FILES.defaults.settings.blocklists["addons-bloomfilters"] += [
+    "addons-bloomfilters/addons-mlbf.bin",
+    "addons-bloomfilters/addons-mlbf.bin.meta.json",
+    "addons-bloomfilters/softblocks-addons-mlbf.bin",
+    "addons-bloomfilters/softblocks-addons-mlbf.bin.meta.json",
+]
diff --git a/services/settings/dumps/main/moz.build b/services/settings/dumps/main/moz.build
index 7f89890eb3..e94d348edf 100644
--- a/services/settings/dumps/main/moz.build
+++ b/services/settings/dumps/main/moz.build
@@ -37,7 +37,16 @@ if not CONFIG["MOZ_BUILD_APP"].startswith("mobile/"):
 
 # These collections are referenced in toolkit/ or other core code.
 FINAL_TARGET_FILES.defaults.settings.main += [
+    "anti-tracking-url-decoration.json",
+    "cookie-banner-rules-list.json",
+    "ironfox-fingerprinting-protection-overrides.json",
+    "moz-essential-domain-fallbacks.json",
     "password-recipes.json",
+    "remote-permissions.json",
+    "translations-models.json",
+    "translations-wasm.json",
+    "url-classifier-skip-urls.json",
+    "url-parser-default-unknown-schemes-interventions.json",
 ]
 
 # These search config icons only apply to desktop.
diff --git a/services/settings/dumps/security-state/moz.build b/services/settings/dumps/security-state/moz.build
index 6e92217dcb..d6c81e5643 100644
--- a/services/settings/dumps/security-state/moz.build
+++ b/services/settings/dumps/security-state/moz.build
@@ -8,7 +8,6 @@ FINAL_TARGET_FILES.defaults.settings["security-state"] += [
 ]
 
 # Not packaged on android/ios currently - not crucial data for first load.
-if not CONFIG["MOZ_BUILD_APP"].startswith("mobile/"):
-    FINAL_TARGET_FILES.defaults.settings["security-state"] += [
-        "intermediates.json",
-    ]
+FINAL_TARGET_FILES.defaults.settings["security-state"] += [
+    "intermediates.json",
+]
