diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 807c5c2..77abf7a 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -909,7 +909,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
 
     var shouldDeleteBrowsingDataOnQuit by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_delete_browsing_data_on_quit),
-        default = false,
+        default = true,
     )
 
     var deleteOpenTabs by booleanPreference(
@@ -924,7 +924,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
 
     var deleteCookies by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_delete_cookies_now),
-        default = true,
+        default = false,
     )
 
     var deleteCache by booleanPreference(
@@ -934,7 +934,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
 
     var deleteSitePermissions by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_delete_permissions_now),
-        default = true,
+        default = false,
     )
 
     var deleteDownloads by booleanPreference(
