diff --git a/dom/base/Navigator.cpp b/dom/base/Navigator.cpp
index fac2759535..c186eb4c44 100644
--- a/dom/base/Navigator.cpp
+++ b/dom/base/Navigator.cpp
@@ -1263,7 +1263,7 @@ bool Navigator::SendBeaconInternal(const nsAString& aUrl,
   } else {
     securityFlags |= nsILoadInfo::SEC_ALLOW_CROSS_ORIGIN_INHERITS_SEC_CONTEXT;
   }
-
+  return true;
   nsCOMPtr<nsIChannel> channel;
   rv = NS_NewChannel(getter_AddRefs(channel), uri, doc, securityFlags,
                      nsIContentPolicy::TYPE_BEACON);
